// flx-ui-datatable/esm5/lib/flx-ui-datatable.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { Validators, FormBuilder } from '@angular/forms';
import { FlxUiDatatableService } from './flx-ui-datatable.service';
import { BehaviorSubject } from 'rxjs';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/forms';
import * as ɵngcc2 from '@angular/common';
import * as ɵngcc3 from 'ngx-perfect-scrollbar';
import * as ɵngcc4 from './pipes/ceil.pipe';
import * as ɵngcc5 from './pipes/hasLimitText.pipe';
import * as ɵngcc6 from './pipes/limitText.pipe';

function FlxUiDatatableComponent_option_8_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "option", 25);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var paging_r244 = ctx.$implicit;
    ɵngcc0.ɵɵproperty("value", paging_r244 == null ? null : paging_r244.value);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(paging_r244 == null ? null : paging_r244.label);
} }
function FlxUiDatatableComponent_span_14_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "ceil");
    ɵngcc0.ɵɵpipe(3, "ceil");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r228 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate2("", ɵngcc0.ɵɵpipeBind2(2, 2, ctx_r228.service == null ? null : ctx_r228.service.dataOffset, ctx_r228.limit), " / ", ɵngcc0.ɵɵpipeBind2(3, 5, ctx_r228.service == null ? null : ctx_r228.service.totalItems, ctx_r228.limit), "");
} }
function FlxUiDatatableComponent_ng_template_15_Template(rf, ctx) { }
var _c0 = function (a0) { return { border: a0 }; };
function FlxUiDatatableComponent_input_22_Template(rf, ctx) { if (rf & 1) {
    var _r246 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "input", 26);
    ɵngcc0.ɵɵlistener("keyup", function FlxUiDatatableComponent_input_22_Template_input_keyup_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r246); var ctx_r245 = ɵngcc0.ɵɵnextContext(); return ctx_r245.filterData($event == null ? null : $event.target == null ? null : $event.target.value); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r231 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵstyleProp("background", ctx_r231.searchBar == null ? null : ctx_r231.searchBar.background, ɵngcc0.ɵɵdefaultStyleSanitizer)("color", ctx_r231.searchBar == null ? null : ctx_r231.searchBar.textColor);
    ɵngcc0.ɵɵproperty("ngStyle", ɵngcc0.ɵɵpureFunction1(6, _c0, (ctx_r231.searchBar == null ? null : ctx_r231.searchBar.borderSize) + " solid " + (ctx_r231.searchBar == null ? null : ctx_r231.searchBar.borderColor)))("placeholder", ctx_r231.searchPlaceholder);
} }
function FlxUiDatatableComponent_form_23_Template(rf, ctx) { if (rf & 1) {
    var _r249 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "form", 27, 28);
    ɵngcc0.ɵɵlistener("ngSubmit", function FlxUiDatatableComponent_form_23_Template_form_ngSubmit_0_listener() { ɵngcc0.ɵɵrestoreView(_r249); var _r247 = ɵngcc0.ɵɵreference(1); var ctx_r248 = ɵngcc0.ɵɵnextContext(); return ctx_r248.searchDataInApi(_r247 == null ? null : _r247.value, _r247); });
    ɵngcc0.ɵɵelementStart(2, "div", 29);
    ɵngcc0.ɵɵelement(3, "input", 30);
    ɵngcc0.ɵɵelementStart(4, "span", 31);
    ɵngcc0.ɵɵelementStart(5, "button", 32);
    ɵngcc0.ɵɵelement(6, "i", 33);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var _r247 = ɵngcc0.ɵɵreference(1);
    var ctx_r232 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵclassMapInterpolate1("form-control rmsh rmrd ", ctx_r232.classes == null ? null : ctx_r232.classes.searchbar, "");
    ɵngcc0.ɵɵstyleProp("background", ctx_r232.searchBar == null ? null : ctx_r232.searchBar.background, ɵngcc0.ɵɵdefaultStyleSanitizer)("color", ctx_r232.searchBar == null ? null : ctx_r232.searchBar.textColor);
    ɵngcc0.ɵɵproperty("ngStyle", ɵngcc0.ɵɵpureFunction1(10, _c0, (ctx_r232.searchBar == null ? null : ctx_r232.searchBar.borderSize) + " solid " + (ctx_r232.searchBar == null ? null : ctx_r232.searchBar.borderColor)))("placeholder", ctx_r232.searchPlaceholder);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("disabled", !(_r247 == null ? null : _r247.valid));
} }
function FlxUiDatatableComponent_div_24_Template(rf, ctx) { if (rf & 1) {
    var _r251 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 34);
    ɵngcc0.ɵɵelementStart(1, "button", 35);
    ɵngcc0.ɵɵlistener("click", function FlxUiDatatableComponent_div_24_Template_button_click_1_listener() { ɵngcc0.ɵɵrestoreView(_r251); var ctx_r250 = ɵngcc0.ɵɵnextContext(); return ctx_r250.addButtonClick(); });
    ɵngcc0.ɵɵelement(2, "span", 36);
    ɵngcc0.ɵɵtext(3, " Add");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r233 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵclassMap(ctx_r233.classes == null ? null : ctx_r233.classes.addButton);
    ɵngcc0.ɵɵstyleProp("background", ctx_r233.addButton == null ? null : ctx_r233.addButton.background, ɵngcc0.ɵɵdefaultStyleSanitizer)("border-color", ctx_r233.addButton == null ? null : ctx_r233.addButton.background)("color", ctx_r233.addButton == null ? null : ctx_r233.addButton.textColor);
} }
function FlxUiDatatableComponent_div_25_li_11_a_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "a", 51);
    ɵngcc0.ɵɵelement(1, "i", 52);
    ɵngcc0.ɵɵtext(2, " Print");
    ɵngcc0.ɵɵelementEnd();
} }
function FlxUiDatatableComponent_div_25_li_11_a_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "a", 53);
    ɵngcc0.ɵɵelement(1, "i", 54);
    ɵngcc0.ɵɵtext(2, " PDF");
    ɵngcc0.ɵɵelementEnd();
} }
function FlxUiDatatableComponent_div_25_li_11_a_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "a", 55);
    ɵngcc0.ɵɵelement(1, "i", 54);
    ɵngcc0.ɵɵtext(2, " Excel");
    ɵngcc0.ɵɵelementEnd();
} }
function FlxUiDatatableComponent_div_25_li_11_a_4_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "a", 56);
    ɵngcc0.ɵɵelement(1, "i", 54);
    ɵngcc0.ɵɵtext(2, " Word");
    ɵngcc0.ɵɵelementEnd();
} }
function FlxUiDatatableComponent_div_25_li_11_Template(rf, ctx) { if (rf & 1) {
    var _r261 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "li", 46);
    ɵngcc0.ɵɵlistener("click", function FlxUiDatatableComponent_div_25_li_11_Template_li_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r261); var export_r255 = ctx.$implicit; var ctx_r260 = ɵngcc0.ɵɵnextContext(2); return ctx_r260.exportDocumentsAs(export_r255); });
    ɵngcc0.ɵɵtemplate(1, FlxUiDatatableComponent_div_25_li_11_a_1_Template, 3, 0, "a", 47);
    ɵngcc0.ɵɵtemplate(2, FlxUiDatatableComponent_div_25_li_11_a_2_Template, 3, 0, "a", 48);
    ɵngcc0.ɵɵtemplate(3, FlxUiDatatableComponent_div_25_li_11_a_3_Template, 3, 0, "a", 49);
    ɵngcc0.ɵɵtemplate(4, FlxUiDatatableComponent_div_25_li_11_a_4_Template, 3, 0, "a", 50);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var export_r255 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", export_r255 == "print");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", export_r255 == "pdf");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", export_r255 == "excel");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", export_r255 == "word");
} }
function FlxUiDatatableComponent_div_25_span_14_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "async");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r253 = ɵngcc0.ɵɵnextContext(2);
    var tmp_0_0 = null;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate((tmp_0_0 = ɵngcc0.ɵɵpipeBind1(2, 1, ctx_r253.displayData)) == null ? null : tmp_0_0.length);
} }
function FlxUiDatatableComponent_div_25_span_15_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "async");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r254 = ɵngcc0.ɵɵnextContext(2);
    var tmp_0_0 = null;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate((tmp_0_0 = ɵngcc0.ɵɵpipeBind1(2, 1, ctx_r254.service == null ? null : ctx_r254.service.flxData)) == null ? null : tmp_0_0.length);
} }
function FlxUiDatatableComponent_div_25_Template(rf, ctx) { if (rf & 1) {
    var _r263 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 37);
    ɵngcc0.ɵɵelementStart(1, "div", 38);
    ɵngcc0.ɵɵelementStart(2, "button", 39);
    ɵngcc0.ɵɵelement(3, "i", 40);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(4, "ul");
    ɵngcc0.ɵɵelementStart(5, "li", 41);
    ɵngcc0.ɵɵtext(6);
    ɵngcc0.ɵɵelementStart(7, "input", 42);
    ɵngcc0.ɵɵlistener("change", function FlxUiDatatableComponent_div_25_Template_input_change_7_listener($event) { ɵngcc0.ɵɵrestoreView(_r263); var ctx_r262 = ɵngcc0.ɵɵnextContext(); return ctx_r262.checkToExportOption($event == null ? null : $event.target == null ? null : $event.target.checked); });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(8, "sup", 43);
    ɵngcc0.ɵɵtext(9, " All");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelement(10, "li", 44);
    ɵngcc0.ɵɵtemplate(11, FlxUiDatatableComponent_div_25_li_11_Template, 5, 4, "li", 45);
    ɵngcc0.ɵɵelement(12, "li", 44);
    ɵngcc0.ɵɵelementStart(13, "li", 41);
    ɵngcc0.ɵɵtemplate(14, FlxUiDatatableComponent_div_25_span_14_Template, 3, 3, "span", 20);
    ɵngcc0.ɵɵtemplate(15, FlxUiDatatableComponent_div_25_span_15_Template, 3, 3, "span", 20);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r234 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵclassMapInterpolate1("btn btn-default ", ctx_r234.classes == null ? null : ctx_r234.classes.exportButton, " dropdown-toggle");
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵclassMapInterpolate1("dropdown-menu dropdown-menu-export ", ctx_r234.classes == null ? null : ctx_r234.classes.exportDropdown, "");
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate1("", ctx_r234.dataExportsConfig == null ? null : ctx_r234.dataExportsConfig.title, ". ");
    ɵngcc0.ɵɵadvance(5);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r234.dataExportsConfig == null ? null : ctx_r234.dataExportsConfig.exportsTo);
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵproperty("ngIf", !ctx_r234.isExportAll);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r234.isExportAll);
} }
function FlxUiDatatableComponent_th_32_input_2_Template(rf, ctx) { if (rf & 1) {
    var _r267 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "input", 59);
    ɵngcc0.ɵɵlistener("change", function FlxUiDatatableComponent_th_32_input_2_Template_input_change_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r267); var ctx_r266 = ɵngcc0.ɵɵnextContext(2); return ctx_r266.addRemove($event == null ? null : $event.target == null ? null : $event.target.checked); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r264 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("checked", (ctx_r264.service == null ? null : ctx_r264.service.multipleDeletion == null ? null : ctx_r264.service.multipleDeletion.length) > 0);
} }
function FlxUiDatatableComponent_th_32_button_3_Template(rf, ctx) { if (rf & 1) {
    var _r269 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "button", 60);
    ɵngcc0.ɵɵlistener("click", function FlxUiDatatableComponent_th_32_button_3_Template_button_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r269); var ctx_r268 = ɵngcc0.ɵɵnextContext(2); return ctx_r268.confirmDelete(); });
    ɵngcc0.ɵɵelement(1, "span");
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r265 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵclassMap(ctx_r265.multipleSelectButton == null ? null : ctx_r265.multipleSelectButton.icon);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r265.multipleSelectButton == null ? null : ctx_r265.multipleSelectButton.text, "");
} }
function FlxUiDatatableComponent_th_32_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "th");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵtemplate(2, FlxUiDatatableComponent_th_32_input_2_Template, 1, 1, "input", 57);
    ɵngcc0.ɵɵtemplate(3, FlxUiDatatableComponent_th_32_button_3_Template, 3, 3, "button", 58);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r235 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1("", ctx_r235.actionHeader, " ");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r235.enableMultipleSelection);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r235.enableMultipleSelection && (ctx_r235.service == null ? null : ctx_r235.service.multipleDeletion == null ? null : ctx_r235.service.multipleDeletion.length) > 0);
} }
function FlxUiDatatableComponent_th_33_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "th");
    ɵngcc0.ɵɵtext(1, "N");
    ɵngcc0.ɵɵelementStart(2, "sup");
    ɵngcc0.ɵɵelementStart(3, "u");
    ɵngcc0.ɵɵtext(4, "o");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} }
function FlxUiDatatableComponent_th_34_input_2_Template(rf, ctx) { if (rf & 1) {
    var _r273 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "input", 59);
    ɵngcc0.ɵɵlistener("change", function FlxUiDatatableComponent_th_34_input_2_Template_input_change_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r273); var ctx_r272 = ɵngcc0.ɵɵnextContext(2); return ctx_r272.addRemove($event == null ? null : $event.target == null ? null : $event.target.checked); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r270 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("checked", (ctx_r270.service == null ? null : ctx_r270.service.multipleDeletion == null ? null : ctx_r270.service.multipleDeletion.length) > 0);
} }
function FlxUiDatatableComponent_th_34_button_3_Template(rf, ctx) { if (rf & 1) {
    var _r275 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "button", 60);
    ɵngcc0.ɵɵlistener("click", function FlxUiDatatableComponent_th_34_button_3_Template_button_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r275); var ctx_r274 = ɵngcc0.ɵɵnextContext(2); return ctx_r274.confirmDelete(); });
    ɵngcc0.ɵɵelement(1, "span");
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r271 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵclassMap(ctx_r271.multipleSelectButton == null ? null : ctx_r271.multipleSelectButton.icon);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r271.multipleSelectButton == null ? null : ctx_r271.multipleSelectButton.text, "");
} }
function FlxUiDatatableComponent_th_34_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "th");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵtemplate(2, FlxUiDatatableComponent_th_34_input_2_Template, 1, 1, "input", 57);
    ɵngcc0.ɵɵtemplate(3, FlxUiDatatableComponent_th_34_button_3_Template, 3, 3, "button", 58);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r237 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1("", ctx_r237.actionHeader, " ");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r237.enableMultipleSelection);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r237.enableMultipleSelection && (ctx_r237.service == null ? null : ctx_r237.service.multipleDeletion == null ? null : ctx_r237.service.multipleDeletion.length) > 0);
} }
function FlxUiDatatableComponent_th_35_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "th");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var header_r276 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(header_r276);
} }
function FlxUiDatatableComponent_th_36_input_2_Template(rf, ctx) { if (rf & 1) {
    var _r280 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "input", 59);
    ɵngcc0.ɵɵlistener("change", function FlxUiDatatableComponent_th_36_input_2_Template_input_change_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r280); var ctx_r279 = ɵngcc0.ɵɵnextContext(2); return ctx_r279.addRemove($event == null ? null : $event.target == null ? null : $event.target.checked); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r277 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("checked", (ctx_r277.service == null ? null : ctx_r277.service.multipleDeletion == null ? null : ctx_r277.service.multipleDeletion.length) > 0);
} }
function FlxUiDatatableComponent_th_36_button_3_Template(rf, ctx) { if (rf & 1) {
    var _r282 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "button", 60);
    ɵngcc0.ɵɵlistener("click", function FlxUiDatatableComponent_th_36_button_3_Template_button_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r282); var ctx_r281 = ɵngcc0.ɵɵnextContext(2); return ctx_r281.confirmDelete(); });
    ɵngcc0.ɵɵelement(1, "span");
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r278 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵclassMap(ctx_r278.multipleSelectButton == null ? null : ctx_r278.multipleSelectButton.icon);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r278.multipleSelectButton == null ? null : ctx_r278.multipleSelectButton.text, "");
} }
function FlxUiDatatableComponent_th_36_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "th");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵtemplate(2, FlxUiDatatableComponent_th_36_input_2_Template, 1, 1, "input", 57);
    ɵngcc0.ɵɵtemplate(3, FlxUiDatatableComponent_th_36_button_3_Template, 3, 3, "button", 58);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r239 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1("", ctx_r239.actionHeader, " ");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r239.enableMultipleSelection);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r239.enableMultipleSelection && (ctx_r239.service == null ? null : ctx_r239.service.multipleDeletion == null ? null : ctx_r239.service.multipleDeletion.length) > 0);
} }
function FlxUiDatatableComponent_tr_40_img_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "img", 63);
} if (rf & 2) {
    var ctx_r283 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵclassMap(ctx_r283.classes == null ? null : ctx_r283.classes.spinner);
    ɵngcc0.ɵɵproperty("src", ctx_r283.spinnerSrc, ɵngcc0.ɵɵsanitizeUrl);
} }
function FlxUiDatatableComponent_tr_40_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "tr");
    ɵngcc0.ɵɵelementStart(1, "td", 61);
    ɵngcc0.ɵɵtemplate(2, FlxUiDatatableComponent_tr_40_img_2_Template, 1, 3, "img", 62);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r240 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r240.spinnerSrc);
} }
var _c1 = function (a0, a1, a2) { return { "flx-tooltip-left": a0, "flx-tooltip-bottom": a1, "flx-tooltip-right": a2 }; };
function FlxUiDatatableComponent_tr_41_td_1_span_1_div_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 76);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var aButton_r293 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction3(2, _c1, (aButton_r293 == null ? null : aButton_r293.tooltipPosition) == "left", (aButton_r293 == null ? null : aButton_r293.tooltipPosition) == "bottom", (aButton_r293 == null ? null : aButton_r293.tooltipPosition) == "right"));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(aButton_r293 == null ? null : aButton_r293.tooltip);
} }
function FlxUiDatatableComponent_tr_41_td_1_span_1_span_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "span", 77);
} if (rf & 2) {
    var aButton_r293 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵclassMap(aButton_r293 == null ? null : aButton_r293.icon);
} }
function FlxUiDatatableComponent_tr_41_td_1_span_1_span_6_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "span");
} if (rf & 2) {
    var aButton_r293 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵclassMap(aButton_r293 == null ? null : aButton_r293.icon);
} }
function FlxUiDatatableComponent_tr_41_td_1_span_1_Template(rf, ctx) { if (rf & 1) {
    var _r303 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "span");
    ɵngcc0.ɵɵelementStart(1, "button", 71);
    ɵngcc0.ɵɵlistener("click", function FlxUiDatatableComponent_tr_41_td_1_span_1_Template_button_click_1_listener() { ɵngcc0.ɵɵrestoreView(_r303); var buttonIndex_r294 = ctx.index; var i_r285 = ɵngcc0.ɵɵnextContext(2).index; var ctx_r301 = ɵngcc0.ɵɵnextContext(); return ctx_r301.actionButtonClicked(i_r285, buttonIndex_r294); });
    ɵngcc0.ɵɵtemplate(2, FlxUiDatatableComponent_tr_41_td_1_span_1_div_2_Template, 2, 6, "div", 72);
    ɵngcc0.ɵɵtemplate(3, FlxUiDatatableComponent_tr_41_td_1_span_1_span_3_Template, 1, 2, "span", 73);
    ɵngcc0.ɵɵelementStart(4, "span", 74);
    ɵngcc0.ɵɵtext(5);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtemplate(6, FlxUiDatatableComponent_tr_41_td_1_span_1_span_6_Template, 1, 2, "span", 75);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var aButton_r293 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵclassMapInterpolate1("btn ", aButton_r293 == null ? null : aButton_r293.class, "");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", aButton_r293 == null ? null : aButton_r293.tooltip);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", !(aButton_r293 == null ? null : aButton_r293.iconPosition) || (aButton_r293 == null ? null : aButton_r293.iconPosition) != "right");
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate1(" ", aButton_r293 == null ? null : aButton_r293.text, " ");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", (aButton_r293 == null ? null : aButton_r293.iconPosition) == "right");
} }
function FlxUiDatatableComponent_tr_41_td_1_input_2_Template(rf, ctx) { if (rf & 1) {
    var _r306 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "input", 78);
    ɵngcc0.ɵɵlistener("change", function FlxUiDatatableComponent_tr_41_td_1_input_2_Template_input_change_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r306); var ctx_r305 = ɵngcc0.ɵɵnextContext(2); var data_r284 = ctx_r305.$implicit; var i_r285 = ctx_r305.index; var ctx_r304 = ɵngcc0.ɵɵnextContext(); return ctx_r304.addRemoveDeleteItem(data_r284[ctx_r304.multipleSelectKey], i_r285, $event == null ? null : $event.target == null ? null : $event.target.checked); });
    ɵngcc0.ɵɵelementEnd();
} }
function FlxUiDatatableComponent_tr_41_td_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "td", 69);
    ɵngcc0.ɵɵtemplate(1, FlxUiDatatableComponent_tr_41_td_1_span_1_Template, 7, 7, "span", 21);
    ɵngcc0.ɵɵtemplate(2, FlxUiDatatableComponent_tr_41_td_1_input_2_Template, 1, 0, "input", 70);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r286 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r286.actionButtons);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r286.enableMultipleSelection && (ctx_r286.service == null ? null : ctx_r286.service.multipleDeletion == null ? null : ctx_r286.service.multipleDeletion.length) > 0);
} }
function FlxUiDatatableComponent_tr_41_td_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "td", 79);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var i_r285 = ɵngcc0.ɵɵnextContext().index;
    var ctx_r287 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵclassMap(ctx_r287.classes == null ? null : ctx_r287.classes.tableData);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ctx_r287.offset + i_r285);
} }
function FlxUiDatatableComponent_tr_41_td_3_span_1_div_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 76);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var aButton_r310 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction3(2, _c1, (aButton_r310 == null ? null : aButton_r310.tooltipPosition) == "left", (aButton_r310 == null ? null : aButton_r310.tooltipPosition) == "bottom", (aButton_r310 == null ? null : aButton_r310.tooltipPosition) == "right"));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(aButton_r310 == null ? null : aButton_r310.tooltip);
} }
function FlxUiDatatableComponent_tr_41_td_3_span_1_span_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "span", 77);
} if (rf & 2) {
    var aButton_r310 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵclassMap(aButton_r310 == null ? null : aButton_r310.icon);
} }
function FlxUiDatatableComponent_tr_41_td_3_span_1_span_6_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "span");
} if (rf & 2) {
    var aButton_r310 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵclassMap(aButton_r310 == null ? null : aButton_r310.icon);
} }
function FlxUiDatatableComponent_tr_41_td_3_span_1_Template(rf, ctx) { if (rf & 1) {
    var _r320 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "span");
    ɵngcc0.ɵɵelementStart(1, "button", 71);
    ɵngcc0.ɵɵlistener("click", function FlxUiDatatableComponent_tr_41_td_3_span_1_Template_button_click_1_listener() { ɵngcc0.ɵɵrestoreView(_r320); var buttonIndex_r311 = ctx.index; var i_r285 = ɵngcc0.ɵɵnextContext(2).index; var ctx_r318 = ɵngcc0.ɵɵnextContext(); return ctx_r318.actionButtonClicked(i_r285, buttonIndex_r311); });
    ɵngcc0.ɵɵtemplate(2, FlxUiDatatableComponent_tr_41_td_3_span_1_div_2_Template, 2, 6, "div", 72);
    ɵngcc0.ɵɵtemplate(3, FlxUiDatatableComponent_tr_41_td_3_span_1_span_3_Template, 1, 2, "span", 73);
    ɵngcc0.ɵɵelementStart(4, "span", 74);
    ɵngcc0.ɵɵtext(5);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtemplate(6, FlxUiDatatableComponent_tr_41_td_3_span_1_span_6_Template, 1, 2, "span", 75);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var aButton_r310 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵclassMapInterpolate1("btn ", aButton_r310 == null ? null : aButton_r310.class, "");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", aButton_r310 == null ? null : aButton_r310.tooltip);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", !(aButton_r310 == null ? null : aButton_r310.iconPosition) || (aButton_r310 == null ? null : aButton_r310.iconPosition) != "right");
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate1(" ", aButton_r310 == null ? null : aButton_r310.text, " ");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", (aButton_r310 == null ? null : aButton_r310.iconPosition) == "right");
} }
function FlxUiDatatableComponent_tr_41_td_3_input_2_Template(rf, ctx) { if (rf & 1) {
    var _r323 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "input", 78);
    ɵngcc0.ɵɵlistener("change", function FlxUiDatatableComponent_tr_41_td_3_input_2_Template_input_change_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r323); var ctx_r322 = ɵngcc0.ɵɵnextContext(2); var data_r284 = ctx_r322.$implicit; var i_r285 = ctx_r322.index; var ctx_r321 = ɵngcc0.ɵɵnextContext(); return ctx_r321.addRemoveDeleteItem(data_r284[ctx_r321.multipleSelectKey], i_r285, $event == null ? null : $event.target == null ? null : $event.target.checked); });
    ɵngcc0.ɵɵelementEnd();
} }
function FlxUiDatatableComponent_tr_41_td_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "td", 69);
    ɵngcc0.ɵɵtemplate(1, FlxUiDatatableComponent_tr_41_td_3_span_1_Template, 7, 7, "span", 21);
    ɵngcc0.ɵɵtemplate(2, FlxUiDatatableComponent_tr_41_td_3_input_2_Template, 1, 0, "input", 70);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r288 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r288.actionButtons);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r288.enableMultipleSelection && (ctx_r288.service == null ? null : ctx_r288.service.multipleDeletion == null ? null : ctx_r288.service.multipleDeletion.length) > 0);
} }
function FlxUiDatatableComponent_tr_41_td_4_img_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "img", 82);
} if (rf & 2) {
    var dataKey_r324 = ɵngcc0.ɵɵnextContext().$implicit;
    var data_r284 = ɵngcc0.ɵɵnextContext().$implicit;
    var ctx_r326 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵclassMap("img-fall-back " + (ctx_r326.embedPictures == null ? null : ctx_r326.embedPictures.class));
    ɵngcc0.ɵɵproperty("src", (ctx_r326.embedPictures == null ? null : ctx_r326.embedPictures.rootFolder) + data_r284[dataKey_r324], ɵngcc0.ɵɵsanitizeUrl)("flx-ui-datatable-img-fallback", ctx_r326.embedPictures == null ? null : ctx_r326.embedPictures.fallbackUrl);
} }
function FlxUiDatatableComponent_tr_41_td_4_ng_template_2_section_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "section");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "limittext");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var dataKey_r324 = ɵngcc0.ɵɵnextContext(2).$implicit;
    var data_r284 = ɵngcc0.ɵɵnextContext().$implicit;
    var ctx_r331 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind2(2, 1, ctx_r331.textSubstring, data_r284[dataKey_r324]), " ");
} }
function FlxUiDatatableComponent_tr_41_td_4_ng_template_2_ng_template_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var dataKey_r324 = ɵngcc0.ɵɵnextContext(2).$implicit;
    var data_r284 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(data_r284[dataKey_r324]);
} }
function FlxUiDatatableComponent_tr_41_td_4_ng_template_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵtemplate(0, FlxUiDatatableComponent_tr_41_td_4_ng_template_2_section_0_Template, 3, 4, "section", 11);
    ɵngcc0.ɵɵpipe(1, "haslimittext");
    ɵngcc0.ɵɵtemplate(2, FlxUiDatatableComponent_tr_41_td_4_ng_template_2_ng_template_2_Template, 2, 1, "ng-template", null, 83, ɵngcc0.ɵɵtemplateRefExtractor);
} if (rf & 2) {
    var _r332 = ɵngcc0.ɵɵreference(3);
    var x_r325 = ɵngcc0.ɵɵnextContext().index;
    var ctx_r328 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("ngIf", (ctx_r328.textSubstring == null ? null : ctx_r328.textSubstring.atIndexes) && ɵngcc0.ɵɵpipeBind2(1, 2, ctx_r328.textSubstring == null ? null : ctx_r328.textSubstring.atIndexes, x_r325))("ngIfElse", _r332);
} }
function FlxUiDatatableComponent_tr_41_td_4_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "td");
    ɵngcc0.ɵɵtemplate(1, FlxUiDatatableComponent_tr_41_td_4_img_1_Template, 1, 4, "img", 80);
    ɵngcc0.ɵɵtemplate(2, FlxUiDatatableComponent_tr_41_td_4_ng_template_2_Template, 4, 5, "ng-template", null, 81, ɵngcc0.ɵɵtemplateRefExtractor);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var x_r325 = ctx.index;
    var _r327 = ɵngcc0.ɵɵreference(3);
    var ctx_r289 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵclassMap(ctx_r289.classes == null ? null : ctx_r289.classes.tableData);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r289.hasImageEmbeded() && x_r325 == (ctx_r289.embedPictures == null ? null : ctx_r289.embedPictures.index))("ngIfElse", _r327);
} }
function FlxUiDatatableComponent_tr_41_td_5_span_1_div_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 76);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var aButton_r341 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction3(2, _c1, (aButton_r341 == null ? null : aButton_r341.tooltipPosition) == "left", (aButton_r341 == null ? null : aButton_r341.tooltipPosition) == "bottom", (aButton_r341 == null ? null : aButton_r341.tooltipPosition) == "right"));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(aButton_r341 == null ? null : aButton_r341.tooltip);
} }
function FlxUiDatatableComponent_tr_41_td_5_span_1_span_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "span", 77);
} if (rf & 2) {
    var aButton_r341 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵclassMap(aButton_r341 == null ? null : aButton_r341.icon);
} }
function FlxUiDatatableComponent_tr_41_td_5_span_1_span_6_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "span");
} if (rf & 2) {
    var aButton_r341 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵclassMap(aButton_r341 == null ? null : aButton_r341.icon);
} }
function FlxUiDatatableComponent_tr_41_td_5_span_1_Template(rf, ctx) { if (rf & 1) {
    var _r351 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "span");
    ɵngcc0.ɵɵelementStart(1, "button", 71);
    ɵngcc0.ɵɵlistener("click", function FlxUiDatatableComponent_tr_41_td_5_span_1_Template_button_click_1_listener() { ɵngcc0.ɵɵrestoreView(_r351); var buttonIndex_r342 = ctx.index; var i_r285 = ɵngcc0.ɵɵnextContext(2).index; var ctx_r349 = ɵngcc0.ɵɵnextContext(); return ctx_r349.actionButtonClicked(i_r285, buttonIndex_r342); });
    ɵngcc0.ɵɵtemplate(2, FlxUiDatatableComponent_tr_41_td_5_span_1_div_2_Template, 2, 6, "div", 72);
    ɵngcc0.ɵɵtemplate(3, FlxUiDatatableComponent_tr_41_td_5_span_1_span_3_Template, 1, 2, "span", 73);
    ɵngcc0.ɵɵelementStart(4, "span", 74);
    ɵngcc0.ɵɵtext(5);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtemplate(6, FlxUiDatatableComponent_tr_41_td_5_span_1_span_6_Template, 1, 2, "span", 75);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var aButton_r341 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵclassMapInterpolate1("btn ", aButton_r341 == null ? null : aButton_r341.class, "");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", aButton_r341 == null ? null : aButton_r341.tooltip);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", !(aButton_r341 == null ? null : aButton_r341.iconPosition) || (aButton_r341 == null ? null : aButton_r341.iconPosition) != "right");
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate1(" ", aButton_r341 == null ? null : aButton_r341.text, " ");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", (aButton_r341 == null ? null : aButton_r341.iconPosition) == "right");
} }
function FlxUiDatatableComponent_tr_41_td_5_input_2_Template(rf, ctx) { if (rf & 1) {
    var _r354 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "input", 78);
    ɵngcc0.ɵɵlistener("change", function FlxUiDatatableComponent_tr_41_td_5_input_2_Template_input_change_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r354); var ctx_r353 = ɵngcc0.ɵɵnextContext(2); var data_r284 = ctx_r353.$implicit; var i_r285 = ctx_r353.index; var ctx_r352 = ɵngcc0.ɵɵnextContext(); return ctx_r352.addRemoveDeleteItem(data_r284[ctx_r352.multipleSelectKey], i_r285, $event == null ? null : $event.target == null ? null : $event.target.checked); });
    ɵngcc0.ɵɵelementEnd();
} }
function FlxUiDatatableComponent_tr_41_td_5_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "td", 84);
    ɵngcc0.ɵɵtemplate(1, FlxUiDatatableComponent_tr_41_td_5_span_1_Template, 7, 7, "span", 21);
    ɵngcc0.ɵɵtemplate(2, FlxUiDatatableComponent_tr_41_td_5_input_2_Template, 1, 0, "input", 70);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r290 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r290.actionButtons);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r290.enableMultipleSelection && (ctx_r290.service == null ? null : ctx_r290.service.multipleDeletion == null ? null : ctx_r290.service.multipleDeletion.length) > 0);
} }
function FlxUiDatatableComponent_tr_41_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "tr", 64);
    ɵngcc0.ɵɵtemplate(1, FlxUiDatatableComponent_tr_41_td_1_Template, 3, 2, "td", 65);
    ɵngcc0.ɵɵtemplate(2, FlxUiDatatableComponent_tr_41_td_2_Template, 2, 4, "td", 66);
    ɵngcc0.ɵɵtemplate(3, FlxUiDatatableComponent_tr_41_td_3_Template, 3, 2, "td", 65);
    ɵngcc0.ɵɵtemplate(4, FlxUiDatatableComponent_tr_41_td_4_Template, 4, 5, "td", 67);
    ɵngcc0.ɵɵtemplate(5, FlxUiDatatableComponent_tr_41_td_5_Template, 3, 2, "td", 68);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r241 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r241.hasActionButtons && ((ctx_r241.actionButtonsConfig == null ? null : ctx_r241.actionButtonsConfig.position) == "left" && (ctx_r241.actionButtonsConfig == null ? null : ctx_r241.actionButtonsConfig.beforeNumbers)));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", !ctx_r241.hideNumbers);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r241.hasActionButtons && ((ctx_r241.actionButtonsConfig == null ? null : ctx_r241.actionButtonsConfig.position) == "left" && !(ctx_r241.actionButtonsConfig == null ? null : ctx_r241.actionButtonsConfig.beforeNumbers)));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r241.dataKeys);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r241.hasActionButtons && (ctx_r241.actionButtonsConfig == null ? null : ctx_r241.actionButtonsConfig.position) != "left");
} }
function FlxUiDatatableComponent_tr_43_td_1_Template(rf, ctx) { if (rf & 1) {
    var _r357 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "td", 86);
    ɵngcc0.ɵɵelementStart(1, "span", 87);
    ɵngcc0.ɵɵtext(2, "No data found");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelement(3, "br");
    ɵngcc0.ɵɵelementStart(4, "button", 88);
    ɵngcc0.ɵɵlistener("click", function FlxUiDatatableComponent_tr_43_td_1_Template_button_click_4_listener() { ɵngcc0.ɵɵrestoreView(_r357); var ctx_r356 = ɵngcc0.ɵɵnextContext(2); return ctx_r356.reload(); });
    ɵngcc0.ɵɵelement(5, "span", 89);
    ɵngcc0.ɵɵtext(6, " Reload");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r355 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(4);
    ɵngcc0.ɵɵclassMapInterpolate1("btn btn-default ", ctx_r355.classes == null ? null : ctx_r355.classes.reloadbutton, "");
} }
function FlxUiDatatableComponent_tr_43_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "tr");
    ɵngcc0.ɵɵtemplate(1, FlxUiDatatableComponent_tr_43_td_1_Template, 7, 3, "td", 85);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r242 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r242.service == null ? null : ctx_r242.service.loadFinish);
} }
function FlxUiDatatableComponent_div_45_div_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 3);
    ɵngcc0.ɵɵelementStart(1, "div", 93);
    ɵngcc0.ɵɵtext(2, "Total pagination: ");
    ɵngcc0.ɵɵelementStart(3, "span");
    ɵngcc0.ɵɵtext(4);
    ɵngcc0.ɵɵpipe(5, "ceil");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(6, "div", 94);
    ɵngcc0.ɵɵtext(7, "# of items per pagination: ");
    ɵngcc0.ɵɵelementStart(8, "span");
    ɵngcc0.ɵɵtext(9);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(10, "div", 95);
    ɵngcc0.ɵɵtext(11, "Total items: ");
    ɵngcc0.ɵɵelementStart(12, "span");
    ɵngcc0.ɵɵtext(13);
    ɵngcc0.ɵɵpipe(14, "async");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r358 = ɵngcc0.ɵɵnextContext(2);
    var tmp_2_0 = null;
    ɵngcc0.ɵɵadvance(4);
    ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind2(5, 3, ctx_r358.service == null ? null : ctx_r358.service.totalItems, ctx_r358.limit), "");
    ɵngcc0.ɵɵadvance(5);
    ɵngcc0.ɵɵtextInterpolate(ctx_r358.limit);
    ɵngcc0.ɵɵadvance(4);
    ɵngcc0.ɵɵtextInterpolate((tmp_2_0 = ɵngcc0.ɵɵpipeBind1(14, 6, ctx_r358.service == null ? null : ctx_r358.service.flxData)) == null ? null : tmp_2_0.length);
} }
function FlxUiDatatableComponent_div_45_ng_template_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 3);
    ɵngcc0.ɵɵelementStart(1, "div", 96);
    ɵngcc0.ɵɵtext(2, "Total search result: ");
    ɵngcc0.ɵɵelementStart(3, "span");
    ɵngcc0.ɵɵtext(4);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r360 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(4);
    ɵngcc0.ɵɵtextInterpolate(ctx_r360.searchResultFound == null ? null : ctx_r360.searchResultFound.length);
} }
function FlxUiDatatableComponent_div_45_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 90);
    ɵngcc0.ɵɵtemplate(1, FlxUiDatatableComponent_div_45_div_1_Template, 15, 8, "div", 91);
    ɵngcc0.ɵɵtemplate(2, FlxUiDatatableComponent_div_45_ng_template_2_Template, 5, 1, "ng-template", null, 92, ɵngcc0.ɵɵtemplateRefExtractor);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var _r359 = ɵngcc0.ɵɵreference(3);
    var ctx_r243 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", !ctx_r243.isSearching)("ngIfElse", _r359);
} }
var _c2 = function (a0) { return { "flx-pagination-end": a0 }; };
var FlxUiDatatableComponent = /** @class */ (function () {
    function FlxUiDatatableComponent(__form, service) {
        this.__form = __form;
        this.service = service;
        this.classes = {};
        this.headers = [];
        this.lazyloadingConfig = {};
        this.embedPictures = {};
        this.dataKeys = [];
        this.enableDataExports = false;
        this.dataExportsConfig = {};
        this.searchKeys = [];
        this.dataSrcKey = '';
        this.hasActionButtons = false;
        this.hideNumbers = false;
        this.enableMultipleSelection = false;
        this.multipleSelectKey = '';
        this.hasAddButton = false;
        this.dataUrl = '';
        this.actionButtonStart = false;
        this.multipleSelectButton = { text: 'Selected', icon: '' };
        this.searchPlaceholder = 'Enter name to search';
        this.actionHeader = 'Actions';
        this.limit = 20;
        this.spinnerSrc = '';
        this.actionButtons = [];
        this.paginationButtons = { background: '#dddddd', textColor: '#335599' };
        this.tableHeader = { background: '#ffffff', textColor: '#335599' };
        this.searchButton = { background: '#cccccc', textColor: '#335599' };
        this.addButton = {};
        this.searchBar = { borderSize: '1px', borderColor: '#ccc', background: '#ffffff', textColor: '#000000' };
        this.actionButtonsConfig = { position: 'right', beforeNumbers: true };
        this.firstActionButtonClicked = new EventEmitter();
        this.secondActionButtonClicked = new EventEmitter();
        this.thirdActionButtonClicked = new EventEmitter();
        this.multipleSelectClicked = new EventEmitter();
        this.addButtonClicked = new EventEmitter();
        /**
         * 18/12/2018
         */
        this.textSubstring = {};
        this.conditionManipulator = {};
        this.isExportAll = false;
        this.tData = [];
        this.behavior = new BehaviorSubject([]);
        this.searchDataTempOffset = [];
        this.displayData = this.behavior.asObservable();
        this.offset = 1;
        this.searchResultFound = [];
    }
    /**
     * @return {?}
     */
    FlxUiDatatableComponent.prototype.reload = /**
     * @return {?}
     */
    function () {
        this.service.loadFlxDataTableData(this.reloadUrl, true);
    };
    /**
     *
     * @param checked Export all selection
     */
    /**
     *
     * @param {?} checked Export all selection
     * @return {?}
     */
    FlxUiDatatableComponent.prototype.checkToExportOption = /**
     *
     * @param {?} checked Export all selection
     * @return {?}
     */
    function (checked) {
        this.isExportAll = checked;
    };
    /**
     *
     * @param exportType Export type: print|pdf|excel|word
     */
    /**
     *
     * @param {?} exportType Export type: print|pdf|excel|word
     * @return {?}
     */
    FlxUiDatatableComponent.prototype.exportDocumentsAs = /**
     *
     * @param {?} exportType Export type: print|pdf|excel|word
     * @return {?}
     */
    function (exportType) {
        var _this = this;
        var /** @type {?} */ loading = /** @type {?} */ (document.getElementById("export_loading"));
        loading.style.display = 'block';
        var /** @type {?} */ headers = (!this.dataExportsConfig.dataColumns || this.dataExportsConfig.dataColumns.length < 1) ? this.dataKeys : this.dataExportsConfig.dataColumns;
        var /** @type {?} */ dataToExport = (!this.isExportAll) ? this.displayData : this.service.flxData;
        //Subscribe to data
        dataToExport.subscribe(function (data) {
            var /** @type {?} */ arrayObj = [];
            try {
                //Loop and push data
                for (var data_1 = tslib_1.__values(data), data_1_1 = data_1.next(); !data_1_1.done; data_1_1 = data_1.next()) {
                    var d = data_1_1.value;
                    var /** @type {?} */ obj = {};
                    for (var /** @type {?} */ h = 0; h < headers.length; h++) {
                        obj[headers[h]] = d[headers[h]];
                    }
                    arrayObj.push(obj);
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (data_1_1 && !data_1_1.done && (_a = data_1.return)) _a.call(data_1);
                }
                finally { if (e_1) throw e_1.error; }
            }
            if (exportType == 'print') {
                try {
                    printJS({ printable: arrayObj, properties: headers, type: 'json' });
                    loading.style.display = 'none';
                }
                catch (/** @type {?} */ e) {
                    loading.style.display = 'none';
                    // console.log('PrintJS not found. Add `https://printjs-4de6.kxcdn.com/print.min.js` to your index.html or add as part of angular.json script') ;
                }
            }
            else {
                var /** @type {?} */ extension_1 = (exportType == 'pdf') ? 'pdf' : (exportType == 'excel') ? 'xlsx' : 'docx';
                var /** @type {?} */ pageId = (exportType == 'pdf') ? 3 : (exportType == 'excel') ? 5 : 4;
                var /** @type {?} */ requestData = { "data": JSON.stringify(arrayObj) };
                _this.service.postData('http://exporter.azurewebsites.net/api/export/ExportFromJSON/', pageId, requestData).subscribe(function (resp) {
                    var /** @type {?} */ download = 'http://exporter.azurewebsites.net/api/export/GetFile/' + resp;
                    download += "?fileName=andrei&extension=" + extension_1;
                    window.location.href = download;
                    loading.style.display = 'none';
                }, (function (e) {
                    //console.log('file export error',e) ;
                }));
            }
            var e_1, _a;
        }).unsubscribe();
    };
    /**
     * @return {?}
     */
    FlxUiDatatableComponent.prototype.hasImageEmbeded = /**
     * @return {?}
     */
    function () {
        return this.embedPictures.hasOwnProperty("index");
    };
    /**
     * @return {?}
     */
    FlxUiDatatableComponent.prototype.getImage = /**
     * @return {?}
     */
    function () {
        console.log('eoeoe');
        //   let img = new Image() ;
        //   img.src = imageSrc ;
        //   img.onload = ((e)=>{
        //       return imageSrc ;
        //   }) ;
        //   img.onerror = ((e)=>{
        //     return this.embedPictures.fallbackUrl ;
        //   })
    };
    /**
     * @param {?} JSONData
     * @param {?} ReportTitle
     * @param {?} ShowLabel
     * @return {?}
     */
    FlxUiDatatableComponent.prototype.JSONToCSVConvertor = /**
     * @param {?} JSONData
     * @param {?} ReportTitle
     * @param {?} ShowLabel
     * @return {?}
     */
    function (JSONData, ReportTitle, ShowLabel) {
        //If JSONData is not an object then JSON.parse will parse the JSON string in an Object
        var /** @type {?} */ arrData = typeof JSONData != 'object' ? JSON.parse(JSONData) : JSONData;
        var /** @type {?} */ CSV = '';
        //Set Report title in first row or line
        CSV += ReportTitle + '\r\n\n';
        //This condition will generate the Label/Header
        if (ShowLabel) {
            var /** @type {?} */ row = "";
            //This loop will extract the label from 1st index of on array
            for (var /** @type {?} */ index in arrData[0]) {
                //Now convert each value to string and comma-seprated
                row += index + ',';
            }
            row = row.slice(0, -1);
            //append Label row with line break
            CSV += row + '\r\n';
        }
        //1st loop is to extract each row
        for (var /** @type {?} */ i = 0; i < arrData.length; i++) {
            var /** @type {?} */ row = "";
            //2nd loop will extract each column and convert it in string comma-seprated
            for (var /** @type {?} */ index in arrData[i]) {
                row += '"' + arrData[i][index] + '",';
            }
            row.slice(0, row.length - 1);
            //add a line break after each row
            CSV += row + '\r\n';
        }
        if (CSV == '') {
            alert("Invalid data");
            return;
        }
        //Generate a file name
        var /** @type {?} */ fileName = "MyReport_";
        //this will remove the blank-spaces from the title and replace it with an underscore
        fileName += ReportTitle.replace(/ /g, "_");
        //Initialize file format you want csv or xls
        var /** @type {?} */ uri = 'data:text/csv;charset=utf-8,' + escape(CSV);
        // Now the little tricky part.
        // you can use either>> window.open(uri);
        // but this will not work in some browsers
        // or you will not get the correct file extension
        //this trick will generate a temp <a /> tag
        var /** @type {?} */ link = document.createElement("a");
        link.href = uri;
        //set the visibility hidden so it will not effect on your web-layout
        link.style = "visibility:hidden";
        link.download = fileName + ".csv";
        //this part will append the anchor tag and remove it after automatic click
        document.body.appendChild(link);
        link.click();
        document.body.removeChild(link);
    };
    /**
     *
     * @param newData
     */
    /**
     *
     * @param {?} newData
     * @return {?}
     */
    FlxUiDatatableComponent.prototype.changeDisplayData = /**
     *
     * @param {?} newData
     * @return {?}
     */
    function (newData) {
        this.behavior.next(newData);
        this.setColumnWidths();
    };
    /**
     * @return {?}
     */
    FlxUiDatatableComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        var _this = this;
        $(window).resize(function () {
            /* Same as before */
        }).resize();
        if (this.isLazyloadingEnabled()) {
            this.reloadUrl = this.dataUrl + '&' + this.lazyloadingConfig.apiOffsetKey + '=0&' + this.lazyloadingConfig.apiSearchKey + '=';
        }
        else {
            this.reloadUrl = this.dataUrl;
        }
        this.searchForm = this.__form.group({
            searchString: ['', Validators.required]
        });
        this.searchForm = this.__form.group({
            searchString: ['', Validators.required]
        });
        this.service.limit = this.limit;
        this.service.setLazyloadingConfig(this.lazyloadingConfig);
        this.service.setDataUrl(this.dataUrl);
        this.service.setDataSrcKey(this.dataSrcKey);
        var /** @type {?} */ url = (this.isLazyloadingEnabled()) ? this.dataUrl + '&' + this.lazyloadingConfig.apiOffsetKey + '=0&' + this.lazyloadingConfig.apiSearchKey + '=' : this.dataUrl;
        this.service.loadFlxDataTableData(url);
        this.service.flxData.subscribe(function (resp) {
            _this.tData = resp;
            var /** @type {?} */ obj = [];
            if (_this.tData.length > _this.limit) {
                for (var /** @type {?} */ i = 0; i < _this.limit; i++) {
                    obj.push(_this.tData[i]);
                }
                // this.service.dataOffset = this.limit;
            }
            else {
                var /** @type {?} */ counter = 0;
                for (var /** @type {?} */ i = 0; i < _this.tData.length; i++) {
                    obj.push(_this.tData[i]);
                    counter++;
                }
                // this.service.dataOffset = obj.length;
            }
            _this.searchDataTempOffset = obj;
            _this.changeDisplayData(obj);
        });
    };
    /**
     * @param {?} values
     * @param {?} form
     * @return {?}
     */
    FlxUiDatatableComponent.prototype.searchDataInApi = /**
     * @param {?} values
     * @param {?} form
     * @return {?}
     */
    function (values, form) {
        this.service.chageDataTable([]);
        this.service.loadFlxDataTableData(this.dataUrl + '&' + this.lazyloadingConfig.apiOffsetKey + '=0&' + this.lazyloadingConfig.apiSearchKey + '=' + values.searchString);
    };
    /**
     * @return {?}
     */
    FlxUiDatatableComponent.prototype.ngAfterViewInit = /**
     * @return {?}
     */
    function () {
        var /** @type {?} */ eachColumnWidth = this.returColumnswidth();
        $(window).ready(function () {
            $(".flx-table-thead tr th").css({ 'width': eachColumnWidth + '%' });
        });
    };
    /**
     * @return {?}
     */
    FlxUiDatatableComponent.prototype.setColumnWidths = /**
     * @return {?}
     */
    function () {
        var /** @type {?} */ eachColumnWidth = this.returColumnswidth();
        $(window).ready(function () {
            $(".flx-table-data tbody tr td").css({ 'width': eachColumnWidth + '%' });
        });
    };
    /**
     * @return {?}
     */
    FlxUiDatatableComponent.prototype.returColumnswidth = /**
     * @return {?}
     */
    function () {
        var /** @type {?} */ hasNumbers = (!this.hideNumbers) ? 0 : 1;
        var /** @type {?} */ hasActionButtons = (!this.hasActionButtons) ? 0 : 1;
        var /** @type {?} */ numberOfColumns = (this.headers.length + 1) + hasActionButtons + hasNumbers;
        console.log('number_of_columns', numberOfColumns);
        var /** @type {?} */ eachColumnWidth = 100 / numberOfColumns;
        return eachColumnWidth;
    };
    /**
     * @param {?} index
     * @param {?} buttonIndex
     * @return {?}
     */
    FlxUiDatatableComponent.prototype.actionButtonClicked = /**
     * @param {?} index
     * @param {?} buttonIndex
     * @return {?}
     */
    function (index, buttonIndex) {
        var _this = this;
        this.displayData.subscribe(function (data) {
            if (buttonIndex == 0) {
                return _this.firstActionButtonClicked.emit({ index: index, data: data[index] });
            }
            else if (buttonIndex == 1) {
                return _this.secondActionButtonClicked.emit({ index: index, data: data[index] });
            }
            else {
                _this.thirdActionButtonClicked.emit({ index: index, data: data[index] });
            }
        }).unsubscribe();
    };
    /**
     * @return {?}
     */
    FlxUiDatatableComponent.prototype.addButtonClick = /**
     * @return {?}
     */
    function () {
        this.addButtonClicked.emit();
    };
    /**
     * @return {?}
     */
    FlxUiDatatableComponent.prototype.confirmDelete = /**
     * @return {?}
     */
    function () {
        return this.multipleSelectClicked.emit(this.service.multipleDeletion);
    };
    /**
     * @param {?} checked
     * @return {?}
     */
    FlxUiDatatableComponent.prototype.addRemove = /**
     * @param {?} checked
     * @return {?}
     */
    function (checked) {
        var _this = this;
        if (checked) {
            this.displayData.subscribe(function (data) {
                var /** @type {?} */ counter = 0;
                try {
                    for (var data_2 = tslib_1.__values(data), data_2_1 = data_2.next(); !data_2_1.done; data_2_1 = data_2.next()) {
                        var i = data_2_1.value;
                        try {
                            _this.service.multipleDeletion.push(i[_this.multipleSelectKey]);
                        }
                        catch (/** @type {?} */ e) { }
                    }
                }
                catch (e_2_1) { e_2 = { error: e_2_1 }; }
                finally {
                    try {
                        if (data_2_1 && !data_2_1.done && (_a = data_2.return)) _a.call(data_2);
                    }
                    finally { if (e_2) throw e_2.error; }
                }
                // console.log(this.service.multipleDeletion) ;
                var e_2, _a;
            }).unsubscribe();
        }
        else {
            this.service.multipleDeletion = [];
        }
    };
    /**
     * @param {?} dataKeyvalue
     * @param {?} index
     * @param {?} selected
     * @return {?}
     */
    FlxUiDatatableComponent.prototype.addRemoveDeleteItem = /**
     * @param {?} dataKeyvalue
     * @param {?} index
     * @param {?} selected
     * @return {?}
     */
    function (dataKeyvalue, index, selected) {
        var _this = this;
        if (!selected) {
            for (var /** @type {?} */ i = 0; i < this.service.multipleDeletion.length; i++) {
                if (dataKeyvalue == this.service.multipleDeletion[i]) {
                    this.service.multipleDeletion.splice(i, 1);
                    break;
                }
            }
        }
        else {
            this.displayData.subscribe(function (resp) {
                _this.service.multipleDeletion.push(resp[index][_this.multipleSelectKey]);
            }).unsubscribe();
        }
        // console.log('left '+dataKeyvalue,this.service.multipleDeletion) ;
    };
    /**
     * @return {?}
     */
    FlxUiDatatableComponent.prototype.getSearchColumns = /**
     * @return {?}
     */
    function () {
        return (this.hasAddButton) ? (this.enableDataExports) ? 'col-md-6 search-container' : 'col-md-7 search-container' :
            (this.enableDataExports) ? 'col-md-7 search-container' : 'col-md-8 search-container';
    };
    /**
     * @return {?}
     */
    FlxUiDatatableComponent.prototype.disablePrevtButton = /**
     * @return {?}
     */
    function () {
        return Math.ceil(this.service.dataOffset / this.limit) <= 1;
    };
    /**
     * @return {?}
     */
    FlxUiDatatableComponent.prototype.disableNextButton = /**
     * @return {?}
     */
    function () {
        return Math.ceil(this.service.dataOffset / this.limit) == Math.ceil(this.service.totalItems / this.limit);
    };
    /**
     * @return {?}
     */
    FlxUiDatatableComponent.prototype.isLazyloadingEnabled = /**
     * @return {?}
     */
    function () {
        return this.lazyloadingConfig.hasOwnProperty("apiOffsetKey") && this.lazyloadingConfig.apiOffsetKey;
    };
    /**
     * @param {?} type
     * @return {?}
     */
    FlxUiDatatableComponent.prototype.nextPrevItem = /**
     * @param {?} type
     * @return {?}
     */
    function (type) {
        var _this = this;
        if (this.isLazyloadingEnabled()) {
            this.service.loadFinish = false;
            this.service.getDataLength().then(function (dataLength) {
                _this.service.chageDataTable([]);
                _this.service.dataOffset = (type == 'prev') ? ((_this.service.dataOffset - _this.limit) - _this.limit) : _this.service.dataOffset;
                var /** @type {?} */ url = (_this.isLazyloadingEnabled()) ? _this.dataUrl + '&' + _this.lazyloadingConfig.apiOffsetKey + '=' + _this.service.dataOffset + '&' + _this.lazyloadingConfig.apiSearchKey + '=' : _this.dataUrl;
                _this.service.loadFlxDataTableData(url);
            }).catch(function (e) {
                //console.log('error',e) ;
            });
            return;
        }
        // Paginate if lazyloading is disabled
        if (type == 'prev') {
            this.paginateDatatableRecord((this.service.dataOffset - this.limit) - this.limit);
        }
        else {
            if (this.service.dataOffset < this.limit) {
                this.paginateDatatableRecord(this.service.dataOffset + (this.limit - 1));
            }
            else {
                this.paginateDatatableRecord(this.service.dataOffset);
            }
        }
    };
    /**
     * @param {?=} searchString
     * @return {?}
     */
    FlxUiDatatableComponent.prototype.filterData = /**
     * @param {?=} searchString
     * @return {?}
     */
    function (searchString) {
        var _this = this;
        if (searchString === void 0) { searchString = ''; }
        if (searchString == '') {
            this.isSearching = false;
        }
        else {
            this.isSearching = true;
        }
        this.changeDisplayData([]);
        this.service.flxData.subscribe(function (data) {
            var /** @type {?} */ searchResults = [];
            //If no string provided. Register all the previous data to the dataset
            if (searchString.trim() == '') {
                _this.changeDisplayData(_this.searchDataTempOffset);
                return;
            }
            //Check if searchKeys are set else use dataKeys as searchKeys
            var /** @type {?} */ searchKeys = (_this.searchKeys.length < 1) ? _this.dataKeys : _this.searchKeys;
            for (var /** @type {?} */ i = 0; i < data.length; i++) {
                //Variable to check if data is found
                var /** @type {?} */ found = -1;
                for (var /** @type {?} */ x = 0; x < searchKeys.length; x++) {
                    try {
                        if (data[i][String(searchKeys[x])].toLowerCase().indexOf(searchString.toLocaleLowerCase()) !== -1) {
                            found = i;
                            break;
                        }
                    }
                    catch (/** @type {?} */ e) { }
                }
                //If found push the index of the data to the searchResults variable
                if (found > -1) {
                    searchResults.push(data[found]);
                }
            }
            _this.searchResultFound = searchResults;
            //Register the results to the dataset
            //Register the results to the dataset
            _this.changeDisplayData(searchResults);
            _this.setColumnWidths();
        }).unsubscribe();
    };
    /**
     * @param value offset value
     */
    /**
     * @param {?} value offset value
     * @return {?}
     */
    FlxUiDatatableComponent.prototype.paginateDatatable = /**
     * @param {?} value offset value
     * @return {?}
     */
    function (value) {
        var _this = this;
        // Check if lazy loading is enabled
        if (this.isLazyloadingEnabled()) {
            this.service.loadFinish = false;
            // Subscribe to get the data length
            this.service.getDataLength().then(function () {
                _this.service.chageDataTable([]);
                // Check if all is selected to prevent NAN value
                if (value != 'all') {
                    _this.service.dataOffset = parseInt(value);
                }
                // setup url
                var /** @type {?} */ url = (_this.isLazyloadingEnabled()) ? _this.dataUrl + '&' + _this.lazyloadingConfig['apiOffsetKey'] + '=' + value + '&' + _this.lazyloadingConfig.apiSearchKey + '=' : _this.dataUrl;
                // paginate
                // paginate
                _this.service.loadFlxDataTableData(url, false);
            }).catch(function (e) {
                // console.log('error',e) ;
            });
            return;
        }
        this.paginateDatatableRecord(value);
    };
    /**
     *
     * @param value pagination number
     * Perform pagination to the dataset
     * @return
     */
    /**
     *
     * @param {?} value pagination number
     * Perform pagination to the dataset
     * @return {?}
     */
    FlxUiDatatableComponent.prototype.paginateDatatableRecord = /**
     *
     * @param {?} value pagination number
     * Perform pagination to the dataset
     * @return {?}
     */
    function (value) {
        var _this = this;
        if (this.lazyloadingConfig.hasOwnProperty("apiOffsetKey") && this.lazyloadingConfig['apiOffsetKey']) {
            this.service.loadFinish = false;
            this.service.getDataLength().then(function (dataLength) {
                _this.service.chageDataTable([]);
                _this.service.dataOffset = parseInt(value) + _this.limit;
                _this.service.loadFlxDataTableData(_this.dataUrl + '&' + _this.lazyloadingConfig.apiOffsetKey + '=' + value + '&' + _this.lazyloadingConfig.apiSearchKey + '=');
                _this.setColumnWidths();
            }).catch(function (e) {
                // console.log('error',e) ;
            });
            return;
        }
        var /** @type {?} */ num = parseInt(value);
        if (num <= 0) {
            this.offset = 1;
            this.service.dataOffset = this.limit;
        }
        else {
            if (value != 'all') {
                this.offset = num + 1;
                this.service.dataOffset = num + this.limit;
            }
            else {
                this.offset = 1;
            }
        }
        this.service.flxData.subscribe(function (data) {
            if (value !== 'all') {
                var /** @type {?} */ paginateResult = [];
                for (var /** @type {?} */ i = value; i < (_this.limit + parseInt(value)); i++) {
                    if (data[i]) {
                        paginateResult.push(data[i]);
                    }
                }
                if (paginateResult.length > 0) {
                    _this.changeDisplayData(paginateResult);
                }
            }
            else {
                _this.changeDisplayData(data);
                _this.searchDataTempOffset = data;
            }
            _this.setColumnWidths();
        }).unsubscribe();
    };
    /** @nocollapse */
    FlxUiDatatableComponent.ctorParameters = function () { return [
        { type: FormBuilder },
        { type: FlxUiDatatableService }
    ]; };
    FlxUiDatatableComponent.propDecorators = {
        classes: [{ type: Input }],
        headers: [{ type: Input }],
        lazyloadingConfig: [{ type: Input }],
        embedPictures: [{ type: Input }],
        dataKeys: [{ type: Input }],
        enableDataExports: [{ type: Input }],
        dataExportsConfig: [{ type: Input }],
        showBottomInfo: [{ type: Input }],
        searchKeys: [{ type: Input }],
        dataSrcKey: [{ type: Input }],
        hasActionButtons: [{ type: Input }],
        hideNumbers: [{ type: Input }],
        enableMultipleSelection: [{ type: Input }],
        multipleSelectKey: [{ type: Input }],
        hasAddButton: [{ type: Input }],
        dataUrl: [{ type: Input }],
        actionButtonStart: [{ type: Input }],
        multipleSelectButton: [{ type: Input }],
        searchPlaceholder: [{ type: Input }],
        actionHeader: [{ type: Input }],
        limit: [{ type: Input }],
        spinnerSrc: [{ type: Input }],
        actionButtons: [{ type: Input }],
        paginationButtons: [{ type: Input }],
        tableHeader: [{ type: Input }],
        searchButton: [{ type: Input }],
        addButton: [{ type: Input }],
        searchBar: [{ type: Input }],
        actionButtonsConfig: [{ type: Input }],
        firstActionButtonClicked: [{ type: Output }],
        secondActionButtonClicked: [{ type: Output }],
        thirdActionButtonClicked: [{ type: Output }],
        multipleSelectClicked: [{ type: Output }],
        addButtonClicked: [{ type: Output }],
        textSubstring: [{ type: Input }],
        conditionManipulator: [{ type: Input }]
    };
FlxUiDatatableComponent.ɵfac = function FlxUiDatatableComponent_Factory(t) { return new (t || FlxUiDatatableComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.FormBuilder), ɵngcc0.ɵɵdirectiveInject(FlxUiDatatableService)); };
FlxUiDatatableComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: FlxUiDatatableComponent, selectors: [["flx-ui-datatable"]], inputs: { classes: "classes", headers: "headers", lazyloadingConfig: "lazyloadingConfig", embedPictures: "embedPictures", dataKeys: "dataKeys", enableDataExports: "enableDataExports", dataExportsConfig: "dataExportsConfig", searchKeys: "searchKeys", dataSrcKey: "dataSrcKey", hasActionButtons: "hasActionButtons", hideNumbers: "hideNumbers", enableMultipleSelection: "enableMultipleSelection", multipleSelectKey: "multipleSelectKey", hasAddButton: "hasAddButton", dataUrl: "dataUrl", actionButtonStart: "actionButtonStart", multipleSelectButton: "multipleSelectButton", searchPlaceholder: "searchPlaceholder", actionHeader: "actionHeader", limit: "limit", spinnerSrc: "spinnerSrc", actionButtons: "actionButtons", paginationButtons: "paginationButtons", tableHeader: "tableHeader", searchButton: "searchButton", addButton: "addButton", searchBar: "searchBar", actionButtonsConfig: "actionButtonsConfig", textSubstring: "textSubstring", conditionManipulator: "conditionManipulator", showBottomInfo: "showBottomInfo" }, outputs: { firstActionButtonClicked: "firstActionButtonClicked", secondActionButtonClicked: "secondActionButtonClicked", thirdActionButtonClicked: "thirdActionButtonClicked", multipleSelectClicked: "multipleSelectClicked", addButtonClicked: "addButtonClicked" }, decls: 46, vars: 58, consts: [["id", "export_loading", 1, "col-md-12", "text-center", 2, "display", "none", "margin-bottom", "0.5em", "color", "#dddddd", "font-size", "13px", "font-weight", "bold"], [1, "col-md-12", "flx-ui-datatable-header"], [1, "container-fluid"], [1, "row"], [1, "col-xs-3", "col-sm-3", "pagination-select", "col-md-2", 2, "position", "relative", "z-index", "1"], [3, "change"], [3, "value", 4, "ngFor", "ngForOf"], [1, "col-xs-5", "col-sm-5", "col-md-2", "text-center", "flx-datatable-pagination", "rmpd", 2, "position", "relative", "z-index", "2"], ["mat-icon-button", "", 3, "ngClass", "disabled", "click"], [1, "fa", "fa-angle-double-left", "fa-1x"], [1, "flx-datatable-tooltip-prev", 3, "ngClass"], [4, "ngIf", "ngIfElse"], ["searching", ""], [1, "fa", "fa-angle-double-right", "fa-1x"], [1, "flx-datatable-tooltip-next", 3, "ngClass"], ["type", "text", "class", "form-control rmsh rmrd customclass", 3, "background", "color", "ngStyle", "placeholder", "keyup", 4, "ngIf"], [3, "ngSubmit", 4, "ngIf"], ["class", "col-md-1 text-right rmpd", 4, "ngIf"], ["class", "col-md-1 text-right rmpd export-cnt", 4, "ngIf"], [1, "col-md-12", "rmpd", "table-responsive", "flx-tabl-container"], [4, "ngIf"], [4, "ngFor", "ngForOf"], [1, "flxdatatablebodyscrollbar"], ["class", "flxuidatatablerow", "id", "flxdatatable_singlerow", 4, "ngFor", "ngForOf"], ["class", "col-md-12 rmpd flx-total-data", 4, "ngIf"], [3, "value"], ["type", "text", 1, "form-control", "rmsh", "rmrd", "customclass", 3, "ngStyle", "placeholder", "keyup"], [3, "ngSubmit"], ["srch", "ngForm"], [1, "input-group"], ["type", "text", "required", "", "name", "searchString", "ngModel", "", 3, "ngStyle", "placeholder"], [1, "input-group-addon"], ["type", "submit", 1, "btn", "btn-default", "btn-clear", "btn-md", 3, "disabled"], [1, "fa", "fa-search"], [1, "col-md-1", "text-right", "rmpd"], [3, "click"], [1, "fa", "fa-plus"], [1, "col-md-1", "text-right", "rmpd", "export-cnt"], [1, "dropdown"], ["type", "button", "data-toggle", "dropdown"], [1, "fa", "fa-angle-down"], [1, "dropdown-header"], ["type", "checkbox", 2, "position", "relative", "top", "0.3em", 3, "change"], [2, "font-size", "10px", "color", "#999"], [1, "divider"], ["class", "dropdown-submenu", 3, "click", 4, "ngFor", "ngForOf"], [1, "dropdown-submenu", 3, "click"], ["href", "javascript:void(0)", "class", "btn btn-default btn-sm form-control", 4, "ngIf"], ["href", "javascript:void(0)", "class", "btn btn-default btn-sm form-control", "style", "color:#ff0000", 4, "ngIf"], ["href", "javascript:void(0)", "class", "btn btn-default btn-sm form-control", "style", "color:#009900;", 4, "ngIf"], ["href", "javascript:void(0)", "class", "btn btn-default btn-sm form-control", "style", "color:#335599;", 4, "ngIf"], ["href", "javascript:void(0)", 1, "btn", "btn-default", "btn-sm", "form-control"], [1, "glyphicon", "glyphicon-print"], ["href", "javascript:void(0)", 1, "btn", "btn-default", "btn-sm", "form-control", 2, "color", "#ff0000"], [1, "glyphicon", "glyphicon-file"], ["href", "javascript:void(0)", 1, "btn", "btn-default", "btn-sm", "form-control", 2, "color", "#009900"], ["href", "javascript:void(0)", 1, "btn", "btn-default", "btn-sm", "form-control", 2, "color", "#335599"], ["type", "checkbox", 3, "checked", "change", 4, "ngIf"], ["class", "btn btn-danger btn-xs flx-multiple-deletion-button", 3, "click", 4, "ngIf"], ["type", "checkbox", 3, "checked", "change"], [1, "btn", "btn-danger", "btn-xs", "flx-multiple-deletion-button", 3, "click"], ["colspan", "20", 1, "text-center"], ["style", "margin-top:1em;margin-bottom:1em;", 3, "class", "src", 4, "ngIf"], [2, "margin-top", "1em", "margin-bottom", "1em", 3, "src"], ["id", "flxdatatable_singlerow", 1, "flxuidatatablerow"], ["class", "table-buttons", "scope", "row", 4, "ngIf"], ["style", "color: #999;", 3, "class", 4, "ngIf"], [3, "class", 4, "ngFor", "ngForOf"], ["class", "table-buttons", 4, "ngIf"], ["scope", "row", 1, "table-buttons"], ["type", "checkbox", "checked", "", 3, "change", 4, "ngIf"], ["type", "button", 3, "click"], ["class", "toltip", "class", "flx-tooltip", 3, "ngClass", 4, "ngIf"], ["class", "action-button-icon-left", 3, "class", 4, "ngIf"], [1, "button-text"], [3, "class", 4, "ngIf"], [1, "flx-tooltip", 3, "ngClass"], [1, "action-button-icon-left"], ["type", "checkbox", "checked", "", 3, "change"], [2, "color", "#999"], [3, "class", "src", "flx-ui-datatable-img-fallback", 4, "ngIf", "ngIfElse"], ["nopicture", ""], [3, "src", "flx-ui-datatable-img-fallback"], ["nolimittext", ""], [1, "table-buttons"], ["colspan", "10", "class", "text-center", 4, "ngIf"], ["colspan", "10", 1, "text-center"], [2, "color", "#ff0000", "font-size", "13px"], ["type", "button", "color", "primary", 2, "margin-top", "1em", 3, "click"], [1, "fa", "fa-refresh"], [1, "col-md-12", "rmpd", "flx-total-data"], ["class", "row", 4, "ngIf", "ngIfElse"], ["bottomInfoSearching", ""], [1, "col-md-4", "text-left", "rmpd"], [1, "col-md-4", "text-center", "rmpd"], [1, "col-md-4", "text-right", "rmpd"], [1, "col-md-12", "text-right", "rmpd"]], template: function FlxUiDatatableComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div");
        ɵngcc0.ɵɵelementStart(1, "div", 0);
        ɵngcc0.ɵɵtext(2, "Exporting...");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(3, "div", 1);
        ɵngcc0.ɵɵelementStart(4, "div", 2);
        ɵngcc0.ɵɵelementStart(5, "div", 3);
        ɵngcc0.ɵɵelementStart(6, "div", 4);
        ɵngcc0.ɵɵelementStart(7, "select", 5);
        ɵngcc0.ɵɵlistener("change", function FlxUiDatatableComponent_Template_select_change_7_listener($event) { return ctx.paginateDatatable($event == null ? null : $event.target == null ? null : $event.target.value); });
        ɵngcc0.ɵɵtemplate(8, FlxUiDatatableComponent_option_8_Template, 2, 2, "option", 6);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(9, "div", 7);
        ɵngcc0.ɵɵelementStart(10, "button", 8);
        ɵngcc0.ɵɵlistener("click", function FlxUiDatatableComponent_Template_button_click_10_listener() { return ctx.nextPrevItem("prev"); });
        ɵngcc0.ɵɵelement(11, "span", 9);
        ɵngcc0.ɵɵelementStart(12, "span", 10);
        ɵngcc0.ɵɵtext(13, "Previous");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(14, FlxUiDatatableComponent_span_14_Template, 4, 8, "span", 11);
        ɵngcc0.ɵɵtemplate(15, FlxUiDatatableComponent_ng_template_15_Template, 0, 0, "ng-template", null, 12, ɵngcc0.ɵɵtemplateRefExtractor);
        ɵngcc0.ɵɵelementStart(17, "button", 8);
        ɵngcc0.ɵɵlistener("click", function FlxUiDatatableComponent_Template_button_click_17_listener() { return ctx.nextPrevItem("next"); });
        ɵngcc0.ɵɵelement(18, "span", 13);
        ɵngcc0.ɵɵelementStart(19, "span", 14);
        ɵngcc0.ɵɵtext(20, "Next");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(21, "div");
        ɵngcc0.ɵɵtemplate(22, FlxUiDatatableComponent_input_22_Template, 1, 8, "input", 15);
        ɵngcc0.ɵɵtemplate(23, FlxUiDatatableComponent_form_23_Template, 7, 12, "form", 16);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(24, FlxUiDatatableComponent_div_24_Template, 4, 9, "div", 17);
        ɵngcc0.ɵɵtemplate(25, FlxUiDatatableComponent_div_25_Template, 16, 10, "div", 18);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(26, "div", 2);
        ɵngcc0.ɵɵelementStart(27, "div", 3);
        ɵngcc0.ɵɵelementStart(28, "div", 19);
        ɵngcc0.ɵɵelementStart(29, "table");
        ɵngcc0.ɵɵelementStart(30, "thead");
        ɵngcc0.ɵɵelementStart(31, "tr");
        ɵngcc0.ɵɵtemplate(32, FlxUiDatatableComponent_th_32_Template, 4, 3, "th", 20);
        ɵngcc0.ɵɵtemplate(33, FlxUiDatatableComponent_th_33_Template, 5, 0, "th", 20);
        ɵngcc0.ɵɵtemplate(34, FlxUiDatatableComponent_th_34_Template, 4, 3, "th", 20);
        ɵngcc0.ɵɵtemplate(35, FlxUiDatatableComponent_th_35_Template, 2, 1, "th", 21);
        ɵngcc0.ɵɵtemplate(36, FlxUiDatatableComponent_th_36_Template, 4, 3, "th", 20);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(37, "perfect-scrollbar");
        ɵngcc0.ɵɵelementStart(38, "table");
        ɵngcc0.ɵɵelementStart(39, "tbody", 22);
        ɵngcc0.ɵɵtemplate(40, FlxUiDatatableComponent_tr_40_Template, 3, 1, "tr", 20);
        ɵngcc0.ɵɵtemplate(41, FlxUiDatatableComponent_tr_41_Template, 6, 5, "tr", 23);
        ɵngcc0.ɵɵpipe(42, "async");
        ɵngcc0.ɵɵtemplate(43, FlxUiDatatableComponent_tr_43_Template, 2, 1, "tr", 20);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(44, "div", 3);
        ɵngcc0.ɵɵtemplate(45, FlxUiDatatableComponent_div_45_Template, 4, 2, "div", 24);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        var _r229 = ɵngcc0.ɵɵreference(16);
        ɵngcc0.ɵɵclassMapInterpolate1("col-md-12 flx-ui-datatable-main ", ctx.classes == null ? null : ctx.classes.maincontainer, "");
        ɵngcc0.ɵɵadvance(7);
        ɵngcc0.ɵɵclassMapInterpolate1("form-control rmsh rmrd ", ctx.classes == null ? null : ctx.classes.paginationselect, "");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.service == null ? null : ctx.service.pagination);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵclassMapInterpolate1("flx-ui-datatable-pagination-buttons ", ctx.classes == null ? null : ctx.classes.paginationButton, "");
        ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(50, _c2, ctx.disablePrevtButton()))("disabled", ctx.disablePrevtButton());
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(52, _c2, ctx.disablePrevtButton()));
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.isSearching)("ngIfElse", _r229);
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵclassMapInterpolate1("flx-ui-datatable-pagination-buttons ", ctx.classes == null ? null : ctx.classes.paginationButton, "");
        ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(54, _c2, ctx.disableNextButton()))("disabled", ctx.disableNextButton());
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(56, _c2, ctx.disableNextButton()));
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵclassMap("search-bar " + ctx.getSearchColumns());
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.isLazyloadingEnabled());
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.isLazyloadingEnabled());
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.hasAddButton);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.enableDataExports);
        ɵngcc0.ɵɵadvance(4);
        ɵngcc0.ɵɵclassMapInterpolate1("table ", ctx.classes == null ? null : ctx.classes.tableType, " th-table");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵclassMapInterpolate1("", ctx.classes == null ? null : ctx.classes.tableHeader, " flx-table-thead");
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.hasActionButtons && ((ctx.actionButtonsConfig == null ? null : ctx.actionButtonsConfig.position) == "left" && (ctx.actionButtonsConfig == null ? null : ctx.actionButtonsConfig.beforeNumbers)));
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.hideNumbers);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.hasActionButtons && ((ctx.actionButtonsConfig == null ? null : ctx.actionButtonsConfig.position) == "left" && !(ctx.actionButtonsConfig == null ? null : ctx.actionButtonsConfig.beforeNumbers)));
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.headers);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.hasActionButtons && (ctx.actionButtonsConfig == null ? null : ctx.actionButtonsConfig.position) == "right");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵclassMapInterpolate1("data-scroll-content ", ctx.classes == null ? null : ctx.classes.dataScrollContent, "");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵclassMapInterpolate1("table ", ctx.classes == null ? null : ctx.classes.tableType, " flx-table-data");
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", !(ctx.service == null ? null : ctx.service.loadFinish));
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngForOf", ɵngcc0.ɵɵpipeBind1(42, 48, ctx.displayData));
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", (ctx.tData == null ? null : ctx.tData.length) < 1);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.showBottomInfo);
    } }, directives: function () { return [ɵngcc2.NgForOf, ɵngcc2.NgClass, ɵngcc2.NgIf, ɵngcc3.PerfectScrollbarComponent, ɵngcc1.NgSelectOption, ɵngcc1.ɵangular_packages_forms_forms_x, ɵngcc2.NgStyle, ɵngcc1.ɵangular_packages_forms_forms_y, ɵngcc1.NgControlStatusGroup, ɵngcc1.NgForm, ɵngcc1.DefaultValueAccessor, ɵngcc1.RequiredValidator, ɵngcc1.NgControlStatus, ɵngcc1.NgModel, ImageFallBack]; }, pipes: function () { return [ɵngcc2.AsyncPipe, ɵngcc4.Ceil, ɵngcc5.HasLimitText, ɵngcc6.LimitText]; }, styles: [".flx-ui-datatable-main[_ngcontent-%COMP%]{background-color:#fff;padding-top:1em;padding-bottom:1em}.flx-ui-datatable-main[_ngcontent-%COMP%]   .btn-danger[_ngcontent-%COMP%]{background-color:#f50057;border:1px solid #f50057;box-shadow:0 3px 5px 1px #ddd;-moz-box-shadow:0 3px 5px 1px #ddd;-webkit-box-shadow:0 3px 5px 1px #ddd;-o-box-shadow:0 3px 5px 1px #ddd;-ms-box-shadow:0 3px 5px 1px #ddd;border-radius:3px;margin-left:.3em}.flx-ui-datatable-main[_ngcontent-%COMP%]   #flxdatatablebodyscrollbar[_ngcontent-%COMP%], .flx-ui-datatable-main[_ngcontent-%COMP%]   .table-responsive[_ngcontent-%COMP%], .flx-ui-datatable-main[_ngcontent-%COMP%]   .table-responsive[_ngcontent-%COMP%]   table[_ngcontent-%COMP%]{width:100%}.flx-ui-datatable-main[_ngcontent-%COMP%]   img.img-fall-back[_ngcontent-%COMP%]{width:30px;height:30px}.flx-ui-datatable-main[_ngcontent-%COMP%]   .pagination-select[_ngcontent-%COMP%]   input[type=text][_ngcontent-%COMP%], .flx-ui-datatable-main[_ngcontent-%COMP%]   .pagination-select[_ngcontent-%COMP%]   select[_ngcontent-%COMP%], .flx-ui-datatable-main[_ngcontent-%COMP%]   .search-bar[_ngcontent-%COMP%]   input[type=text][_ngcontent-%COMP%], .flx-ui-datatable-main[_ngcontent-%COMP%]   .search-bar[_ngcontent-%COMP%]   select[_ngcontent-%COMP%]{border-top:none!important;border-left:none!important;border-right:none!important;border-bottom:2px solid #ddd!important;border-radius:0!important;box-shadow:0 0 0 0 transparent!important;-moz-box-shadow:0 0 0 0 transparent!important;-webkit-box-shadow:0 0 0 0 transparent!important;-o-box-shadow:0 0 0 0 transparent!important;-ms-box-shadow:0 0 0 0 transparent!important}.flx-ui-datatable-main[_ngcontent-%COMP%]   .pagination-select[_ngcontent-%COMP%]   input[type=text][_ngcontent-%COMP%]:focus, .flx-ui-datatable-main[_ngcontent-%COMP%]   .pagination-select[_ngcontent-%COMP%]   select[_ngcontent-%COMP%]:focus, .flx-ui-datatable-main[_ngcontent-%COMP%]   .search-bar[_ngcontent-%COMP%]   input[type=text][_ngcontent-%COMP%]:focus, .flx-ui-datatable-main[_ngcontent-%COMP%]   .search-bar[_ngcontent-%COMP%]   select[_ngcontent-%COMP%]:focus{border-bottom-color:#359!important;transition:.5s;-moz-transition:.5s;-webkit-transition:.5s;-o-transition:.5s;-ms-transition:.5s}.flx-ui-datatable-main[_ngcontent-%COMP%]   .pagination-select[_ngcontent-%COMP%]   select[_ngcontent-%COMP%], .flx-ui-datatable-main[_ngcontent-%COMP%]   .search-bar[_ngcontent-%COMP%]   select[_ngcontent-%COMP%]{-webkit-appearance:none;appearance:none;-moz-appearance:none}.flx-ui-datatable-main[_ngcontent-%COMP%]   .dropdown-toggle[_ngcontent-%COMP%]::after{display:none}.flx-ui-datatable-main[_ngcontent-%COMP%]   .dropdown-menu-export[_ngcontent-%COMP%]{margin-left:-130px;padding-left:1em;padding-right:1em;width:100px!important;margin-top:1em;border:1px solid #ddd!important}.flx-ui-datatable-main[_ngcontent-%COMP%]   .dropdown-menu-export[_ngcontent-%COMP%]::before{content:\"\";width:auto;height:auto;position:absolute;top:-20px;right:3px;border:10px solid;border-color:transparent transparent #ddd}.flx-ui-datatable-main[_ngcontent-%COMP%]   .flx-datatable-pagination[_ngcontent-%COMP%]{padding-top:.5em}.flx-ui-datatable-main[_ngcontent-%COMP%]   .flx-datatable-pagination[_ngcontent-%COMP%]   button[_ngcontent-%COMP%]{width:35px!important;height:35px!important;border-radius:50em!important;border:none!important;box-shadow:0 3px 10px 0 #b3c4e6;-moz-box-shadow:0 3px 10px 0 #b3c4e6;-webkit-box-shadow:0 3px 10px 0 #b3c4e6;-o-box-shadow:0 3px 10px 0 #b3c4e6;-ms-box-shadow:0 3px 10px 0 #b3c4e6;background-color:#359;color:#fff;font-size:23px;font-weight:700;position:absolute;top:0}.flx-ui-datatable-main[_ngcontent-%COMP%]   .flx-datatable-pagination[_ngcontent-%COMP%]   button[_ngcontent-%COMP%]:first-child{left:0}.flx-ui-datatable-main[_ngcontent-%COMP%]   .flx-datatable-pagination[_ngcontent-%COMP%]   button[_ngcontent-%COMP%]:first-child   .flx-datatable-tooltip-prev[_ngcontent-%COMP%]{position:absolute;left:0;font-size:13px;font-weight:400;color:#fff;background-color:#359;padding-left:.3em;padding-right:.3em;border-radius:3px;margin-left:-57px;margin-top:.3em;box-shadow:0 3px 10px 0 #b3c4e6!important;-moz-box-shadow:0 3px 10px 0 #b3c4e6!important;-webkit-box-shadow:0 3px 10px 0 #b3c4e6!important;-o-box-shadow:0 3px 10px 0 #b3c4e6!important;-ms-box-shadow:0 3px 10px 0 #b3c4e6!important;visibility:hidden}.flx-ui-datatable-main[_ngcontent-%COMP%]   .flx-datatable-pagination[_ngcontent-%COMP%]   button[_ngcontent-%COMP%]:hover > span.flx-datatable-tooltip-next[_ngcontent-%COMP%], .flx-ui-datatable-main[_ngcontent-%COMP%]   .flx-datatable-pagination[_ngcontent-%COMP%]   button[_ngcontent-%COMP%]:hover > span.flx-datatable-tooltip-prev[_ngcontent-%COMP%]{visibility:visible}.flx-ui-datatable-main[_ngcontent-%COMP%]   .flx-datatable-pagination[_ngcontent-%COMP%]   .flx-pagination-end[_ngcontent-%COMP%]{background-color:#f50057!important;box-shadow:0 3px 10px 0 #ffc2d8!important;-moz-box-shadow:0 3px 10px 0 #ffc2d8!important;-webkit-box-shadow:0 3px 10px 0 #ffc2d8!important;-o-box-shadow:0 3px 10px 0 #ffc2d8!important;-ms-box-shadow:0 3px 10px 0 #ffc2d8!important;cursor:not-allowed}.flx-ui-datatable-main[_ngcontent-%COMP%]   .flx-datatable-pagination[_ngcontent-%COMP%]   .flx-pagination-end[_ngcontent-%COMP%]:hover > span.flx-datatable-tooltip-next[_ngcontent-%COMP%], .flx-ui-datatable-main[_ngcontent-%COMP%]   .flx-datatable-pagination[_ngcontent-%COMP%]   .flx-pagination-end[_ngcontent-%COMP%]:hover > span.flx-datatable-tooltip-prev[_ngcontent-%COMP%]{visibility:hidden}.flx-ui-datatable-main[_ngcontent-%COMP%]   .flx-datatable-pagination[_ngcontent-%COMP%]   button[_ngcontent-%COMP%]:last-child{right:0}.flx-ui-datatable-main[_ngcontent-%COMP%]   .flx-datatable-pagination[_ngcontent-%COMP%]   button[_ngcontent-%COMP%]:last-child   .flx-datatable-tooltip-next[_ngcontent-%COMP%]{position:absolute;left:0;font-size:13px;font-weight:400;color:#fff;background-color:#359;padding-left:.3em;padding-right:.3em;border-radius:3px;margin-left:35px;margin-top:.5em;box-shadow:0 3px 10px 0 #b3c4e6!important;-moz-box-shadow:0 3px 10px 0 #b3c4e6!important;-webkit-box-shadow:0 3px 10px 0 #b3c4e6!important;-o-box-shadow:0 3px 10px 0 #b3c4e6!important;-ms-box-shadow:0 3px 10px 0 #b3c4e6!important;visibility:hidden}.flx-ui-datatable-main[_ngcontent-%COMP%]   .export-cnt[_ngcontent-%COMP%]   button[_ngcontent-%COMP%]{border-radius:50em!important}.flx-ui-datatable-main[_ngcontent-%COMP%]   .flx-tabl-container[_ngcontent-%COMP%]   table.th-table[_ngcontent-%COMP%]{margin-top:.5em}.flx-ui-datatable-main[_ngcontent-%COMP%]   table[_ngcontent-%COMP%]   tbody[_ngcontent-%COMP%]   tr[_ngcontent-%COMP%]{padding-top:0!important}.flx-ui-datatable-main[_ngcontent-%COMP%]   table[_ngcontent-%COMP%]   tbody[_ngcontent-%COMP%]   tr[_ngcontent-%COMP%]   td[_ngcontent-%COMP%]{padding-top:.5em;border-top:1px solid #f0f0f0;border-bottom:1px solid #f0f0f0}.flx-ui-datatable-main[_ngcontent-%COMP%]   table[_ngcontent-%COMP%]   tbody[_ngcontent-%COMP%]   tr[_ngcontent-%COMP%]   td[_ngcontent-%COMP%]   button[_ngcontent-%COMP%]{margin-right:.3em;margin-left:0}.flx-ui-datatable-main[_ngcontent-%COMP%]   table[_ngcontent-%COMP%]   tbody[_ngcontent-%COMP%]   tr[_ngcontent-%COMP%]   td[_ngcontent-%COMP%]   button[_ngcontent-%COMP%]   div.flx-tooltip[_ngcontent-%COMP%]{position:absolute;background-color:rgba(32,27,27,.808);text-align:center;font-size:13px;color:#fff;border-radius:3px;box-shadow:0 3px 20px 0 #4b4949;-moz-box-shadow:0 3px 20px 0 #4b4949;-webkit-box-shadow:0 3px 20px 0 #4b4949;-o-box-shadow:0 3px 20px 0 #4b4949;-ms-box-shadow:0 3px 20px 0 #4b4949;margin-left:-2.5em;margin-top:-2.8em;visibility:hidden;width:80px;padding:.3em .5em}.flx-ui-datatable-main[_ngcontent-%COMP%]   table[_ngcontent-%COMP%]   tbody[_ngcontent-%COMP%]   tr[_ngcontent-%COMP%]   td[_ngcontent-%COMP%]   button[_ngcontent-%COMP%]   .flx-tooltip-left[_ngcontent-%COMP%]{margin-left:-95px!important;margin-top:-.3em!important}.flx-ui-datatable-main[_ngcontent-%COMP%]   table[_ngcontent-%COMP%]   tbody[_ngcontent-%COMP%]   tr[_ngcontent-%COMP%]   td[_ngcontent-%COMP%]   button[_ngcontent-%COMP%]   .flx-tooltip-bottom[_ngcontent-%COMP%]{margin-top:2.3em!important}.flx-ui-datatable-main[_ngcontent-%COMP%]   table[_ngcontent-%COMP%]   tbody[_ngcontent-%COMP%]   tr[_ngcontent-%COMP%]   td[_ngcontent-%COMP%]   button[_ngcontent-%COMP%]   .flx-tooltip-right[_ngcontent-%COMP%]{margin-left:28px!important;margin-top:-.3em!important}.flx-ui-datatable-main[_ngcontent-%COMP%]   table[_ngcontent-%COMP%]   tbody[_ngcontent-%COMP%]   tr[_ngcontent-%COMP%]   td[_ngcontent-%COMP%]   button[_ngcontent-%COMP%]:hover > div.flx-tooltip[_ngcontent-%COMP%]{transition:.3s;visibility:visible}.flx-ui-datatable-main[_ngcontent-%COMP%]   table[_ngcontent-%COMP%]   tbody[_ngcontent-%COMP%]   tr[_ngcontent-%COMP%]   td.table-buttons[_ngcontent-%COMP%]{padding-top:.2em;padding-bottom:.2em}.flx-ui-datatable-main[_ngcontent-%COMP%]   table[_ngcontent-%COMP%]   tbody[_ngcontent-%COMP%]   tr[_ngcontent-%COMP%]:nth-of-type(even){background-color:#f8f9fa}.flx-ui-datatable-main[_ngcontent-%COMP%]   table[_ngcontent-%COMP%]   tbody[_ngcontent-%COMP%]   tr[_ngcontent-%COMP%]:nth-of-type(odd){background-color:#fff}.flx-ui-datatable-main[_ngcontent-%COMP%]   .btn-danger[_ngcontent-%COMP%]:hover{background-color:#ff146b;border:1px solid #ff146b;box-shadow:0 3px 10px 1px #ff5fb6;-moz-box-shadow:0 3px 10px 1px #ff5fb6;-webkit-box-shadow:0 3px 10px 1px #ff5fb6;-o-box-shadow:0 3px 10px 1px #ff5fb6;-ms-box-shadow:0 3px 10px 1px #ff5fb6;transition:.5s;-moz-transition:.5s;-webkit-transition:.5s;-o-transition:.5s;-ms-transition:.5s}.flx-ui-datatable-main[_ngcontent-%COMP%]   .btn-danger[_ngcontent-%COMP%]:focus{background-color:#f50057;border:1px solid #f50057}.flx-ui-datatable-main[_ngcontent-%COMP%]   .btn-white[_ngcontent-%COMP%]{background-color:#fff}.flx-ui-datatable-main[_ngcontent-%COMP%]   .btn-dark[_ngcontent-%COMP%]{background-color:#222!important}.flx-ui-datatable-main[_ngcontent-%COMP%]   .btn-primary[_ngcontent-%COMP%]{background-color:#359;color:#fff;border:1px solid #359;box-shadow:0 3px 5px 1px #ddd;-moz-box-shadow:0 3px 5px 1px #ddd;-webkit-box-shadow:0 3px 5px 1px #ddd;-o-box-shadow:0 3px 5px 1px #ddd;-ms-box-shadow:0 3px 5px 1px #ddd;border-radius:3px}.flx-ui-datatable-main[_ngcontent-%COMP%]   .btn-primary[_ngcontent-%COMP%]:hover{background-color:#4769ad;border:1px solid #4769ad;box-shadow:0 3px 10px 0 #b3c4e6;-moz-box-shadow:0 3px 10px 0 #b3c4e6;-webkit-box-shadow:0 3px 10px 0 #b3c4e6;-o-box-shadow:0 3px 10px 0 #b3c4e6;-ms-box-shadow:0 3px 10px 0 #b3c4e6;transition:.5s;-moz-transition:.5s;-webkit-transition:.5s;-o-transition:.5s;-ms-transition:.5s}.flx-ui-datatable-main[_ngcontent-%COMP%]   .btn-primary[_ngcontent-%COMP%]:focus{background-color:#359;border:1px solid #359}.flx-ui-datatable-main[_ngcontent-%COMP%]   .btn-large[_ngcontent-%COMP%]{padding-top:1em!important;padding-bottom:1em!important}.flx-ui-datatable-main[_ngcontent-%COMP%]   .btn-medium[_ngcontent-%COMP%]{padding-top:.7em!important;padding-bottom:.7em!important}.flx-ui-datatable-main[_ngcontent-%COMP%]   .btn-success[_ngcontent-%COMP%]{box-shadow:0 3px 5px 1px #ddd;-moz-box-shadow:0 3px 5px 1px #ddd;-webkit-box-shadow:0 3px 5px 1px #ddd;-o-box-shadow:0 3px 5px 1px #ddd;-ms-box-shadow:0 3px 5px 1px #ddd;border-radius:3px;background-color:#5cb85c;border:1px solid #5cb85c}.flx-ui-datatable-main[_ngcontent-%COMP%]   .btn-success[_ngcontent-%COMP%]:hover{background-color:#70cc70;border:1px solid #70cc70;box-shadow:0 3px 10px 1px #9df99d;-moz-box-shadow:0 3px 10px 1px #9df99d;-webkit-box-shadow:0 3px 10px 1px #9df99d;-o-box-shadow:0 3px 10px 1px #9df99d;-ms-box-shadow:0 3px 10px 1px #9df99d;transition:.5s;-moz-transition:.5s;-webkit-transition:.5s;-o-transition:.5s;-ms-transition:.5s}.flx-ui-datatable-main[_ngcontent-%COMP%]   .btn-success[_ngcontent-%COMP%]:focus{background-color:#5cb85c;border:1px solid #5cb85c}.flx-ui-datatable-main[_ngcontent-%COMP%]   .btn-default[_ngcontent-%COMP%]{background-color:#fff;box-shadow:0 3px 5px 1px #eee;-moz-box-shadow:0 3px 5px 1px #eee;-webkit-box-shadow:0 3px 5px 1px #eee;-o-box-shadow:0 3px 5px 1px #eee;-ms-box-shadow:0 3px 5px 1px #eee;border-radius:3px;border:1px solid #ddd}.flx-ui-datatable-main[_ngcontent-%COMP%]   .btn-default[_ngcontent-%COMP%]:hover{background-color:#fff;border:1px solid #e7e7e7;box-shadow:0 3px 10px 1px #e2e2e2;-moz-box-shadow:0 3px 10px 1px #e2e2e2;-webkit-box-shadow:0 3px 10px 1px #e2e2e2;-o-box-shadow:0 3px 10px 1px #e2e2e2;-ms-box-shadow:0 3px 10px 1px #e2e2e2;transition:.5s;-moz-transition:.5s;-webkit-transition:.5s;-o-transition:.5s;-ms-transition:.5s}.flx-ui-datatable-main[_ngcontent-%COMP%]   .btn-default[_ngcontent-%COMP%]:focus{background-color:#fff;border:1px solid #ddd}.flx-ui-datatable-main[_ngcontent-%COMP%]   .btn-secondary[_ngcontent-%COMP%]{box-shadow:0 3px 5px 1px #ddd;-moz-box-shadow:0 3px 5px 1px #ddd;-webkit-box-shadow:0 3px 5px 1px #ddd;-o-box-shadow:0 3px 5px 1px #ddd;-ms-box-shadow:0 3px 5px 1px #ddd;border-radius:3px;background-color:#1e88e5;color:#fff}.flx-ui-datatable-main[_ngcontent-%COMP%]   .btn-secondary[_ngcontent-%COMP%]:hover{color:#fff;background-color:#2892ef;border:1px solid #2892ef;box-shadow:0 3px 10px 1px #55bfff;-moz-box-shadow:0 3px 10px 1px #55bfff;-webkit-box-shadow:0 3px 10px 1px #55bfff;-o-box-shadow:0 3px 10px 1px #55bfff;-ms-box-shadow:0 3px 10px 1px #55bfff;transition:.5s;-moz-transition:.5s;-webkit-transition:.5s;-o-transition:.5s;-ms-transition:.5s}.flx-ui-datatable-main[_ngcontent-%COMP%]   .btn-secondary[_ngcontent-%COMP%]:focus{color:#fff}.flx-ui-datatable-main[_ngcontent-%COMP%]   .pagination-button[_ngcontent-%COMP%]{background-color:#359;color:#fff}.flx-ui-datatable-main[_ngcontent-%COMP%]   .table-font[_ngcontent-%COMP%]{font-family:Khula,sans-serif!important}.flx-ui-datatable-main[_ngcontent-%COMP%]   .table-header-icon[_ngcontent-%COMP%]{position:absolute;right:.2em;width:80px;height:80px;font-size:50px;margin-top:-30px;border-radius:5px!important}.flx-ui-datatable-main[_ngcontent-%COMP%]   .table-title[_ngcontent-%COMP%]{background-color:#359;color:#fff;border-radius:2px;padding:1em;font-size:15px;font-weight:700;margin-bottom:1.5em;font-family:Roboto,sans-serif;box-shadow:0 1px 5px 1px #ddd;-moz-box-shadow:0 1px 5px 1px #ddd;-webkit-box-shadow:0 1px 5px 1px #ddd;-o-box-shadow:0 1px 5px 1px #ddd;-ms-box-shadow:0 1px 5px 1px #ddd}.flx-ui-datatable-main[_ngcontent-%COMP%]   .danger[_ngcontent-%COMP%]{background-color:#f50057;color:#fff}.flx-ui-datatable-main[_ngcontent-%COMP%]   .primary[_ngcontent-%COMP%]{background-color:#359;color:#fff}.flx-ui-datatable-main[_ngcontent-%COMP%]   .success[_ngcontent-%COMP%]{background-color:#5cb85c;color:#fff}.flx-ui-datatable-main[_ngcontent-%COMP%]   .default[_ngcontent-%COMP%]{background-color:#fff;color:#000}.flx-ui-datatable-main[_ngcontent-%COMP%]   .secondary[_ngcontent-%COMP%]{background-color:#1e88e5;color:#fff}.flx-ui-datatable-main[_ngcontent-%COMP%]   .btn-clear[_ngcontent-%COMP%]{border:none!important;box-shadow:none!important}.flx-ui-datatable-main[_ngcontent-%COMP%]   .input-group[_ngcontent-%COMP%], .flx-ui-datatable-main[_ngcontent-%COMP%]   .input-group[_ngcontent-%COMP%]   input[_ngcontent-%COMP%]{background-color:transparent!important}.flx-ui-datatable-main[_ngcontent-%COMP%]   .input-group-addon[_ngcontent-%COMP%]{border:none!important;padding:0!important;box-shadow:none!important;background-color:transparent!important}.flx-ui-datatable-main[_ngcontent-%COMP%]   .input-group-addon[_ngcontent-%COMP%]   button[_ngcontent-%COMP%]{border:1px solid transparent!important;box-shadow:none!important;border-top:none!important;border-bottom:none!important;background-color:transparent!important;border-radius:50em!important;color:#359;width:30px;height:30px}.flx-ui-datatable-main[_ngcontent-%COMP%]   .input-group-addon[_ngcontent-%COMP%]   button[_ngcontent-%COMP%]   i[_ngcontent-%COMP%]{font-size:18px}.flx-ui-datatable-main[_ngcontent-%COMP%]   .input-group-addon[_ngcontent-%COMP%]   button[_ngcontent-%COMP%]:disabled{background-color:transparent!important}.flx-ui-datatable-main[_ngcontent-%COMP%]   .input-group-addon[_ngcontent-%COMP%]   button[_ngcontent-%COMP%]:disabled   i[_ngcontent-%COMP%]{color:#f50057}.flx-ui-datatable-main[_ngcontent-%COMP%]   .data-scroll-content[_ngcontent-%COMP%]{max-width:100%;max-height:400px;margin-top:-1em}.flx-ui-datatable-main[_ngcontent-%COMP%]   .data-scroll-content[_ngcontent-%COMP%]   table[_ngcontent-%COMP%]   tbody[_ngcontent-%COMP%]   tr[_ngcontent-%COMP%]:first-child   td[_ngcontent-%COMP%], .flx-ui-datatable-main[_ngcontent-%COMP%]   .flx-table-thead[_ngcontent-%COMP%]   tr[_ngcontent-%COMP%]   th[_ngcontent-%COMP%]{border-top:none!important}"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FlxUiDatatableComponent, [{
        type: Component,
        args: [{
                selector: 'flx-ui-datatable',
                template: "<div class=\"col-md-12 flx-ui-datatable-main {{ classes?.maincontainer }}\">\n    <div id=\"export_loading\" class=\"col-md-12 text-center\" style=\"display: none;margin-bottom:0.5em;color:#dddddd;font-size:13px;font-weight:bold;\">Exporting...</div>\n    <div class=\"col-md-12 flx-ui-datatable-header\">\n        <div class=\"container-fluid\">\n            <div class=\"row\">\n                <div class=\"col-xs-3 col-sm-3 pagination-select col-md-2\" style=\"position:relative;z-index: 1;\">\n                    <select class=\"form-control rmsh rmrd {{ classes?.paginationselect }}\" (change)=\"paginateDatatable($event?.target?.value)\">\n                        <option *ngFor=\"let paging of service?.pagination\" [value]=\"paging?.value\">{{ paging?.label }}</option>\n                    </select>\n                </div>\n                <div class=\"col-xs-5 col-sm-5 col-md-2 text-center flx-datatable-pagination rmpd\" style=\"position:relative;z-index: 2;\">\n                    <button mat-icon-button [ngClass]=\"{'flx-pagination-end': disablePrevtButton()}\" (click)=\"nextPrevItem('prev')\" [disabled]=\"disablePrevtButton()\" class=\"flx-ui-datatable-pagination-buttons {{ classes?.paginationButton }}\"><span class=\"fa fa-angle-double-left fa-1x\"></span> <span class=\"flx-datatable-tooltip-prev\" [ngClass]=\"{'flx-pagination-end': disablePrevtButton()}\">Previous</span> </button>\n                        <span *ngIf=\"!isSearching;else searching\">{{ service?.dataOffset | ceil: limit }} / {{ service?.totalItems | ceil: limit }}</span>\n                        <ng-template #searching>\n                            \n                        </ng-template>\n                    <button mat-icon-button [ngClass]=\"{'flx-pagination-end': disableNextButton()}\" (click)=\"nextPrevItem('next')\" [disabled]=\"disableNextButton()\" class=\"flx-ui-datatable-pagination-buttons {{ classes?.paginationButton }}\"><span class=\"fa fa-angle-double-right fa-1x\"></span> <span class=\"flx-datatable-tooltip-next\" [ngClass]=\"{'flx-pagination-end': disableNextButton()}\">Next</span></button>\n                </div>\n                <div [class]=\"'search-bar '+getSearchColumns()\">\n                    <input type=\"text\" *ngIf=\"!isLazyloadingEnabled()\" [style.background]=\"searchBar?.background\" [style.color]=\"searchBar?.textColor\" [ngStyle]=\"{border:searchBar?.borderSize +' solid '+ searchBar?.borderColor} \" (keyup)=\"filterData($event?.target?.value)\" class=\"form-control rmsh rmrd customclass\" [placeholder]=\"searchPlaceholder\">\n                    <form (ngSubmit)=\"searchDataInApi(srch?.value,srch)\" #srch=\"ngForm\" *ngIf=\"isLazyloadingEnabled()\">\n                        <div class=\"input-group\">\n                            <input type=\"text\" required name=\"searchString\" ngModel [style.background]=\"searchBar?.background\" [style.color]=\"searchBar?.textColor\" [ngStyle]=\"{border:searchBar?.borderSize +' solid '+ searchBar?.borderColor} \" class=\"form-control rmsh rmrd {{ classes?.searchbar }}\" [placeholder]=\"searchPlaceholder\">\n                            <span class=\"input-group-addon\">\n                                <button [disabled]=\"!srch?.valid\" type=\"submit\" class=\"btn btn-default btn-clear btn-md\">\n                                    <i class=\"fa fa-search\"></i>\n                                </button>\n                            </span>\n                        </div>\n                    </form>\n                </div>\n                <div class=\"col-md-1 text-right rmpd\" *ngIf=\"hasAddButton\">\n                    <button (click)=\"addButtonClick()\" class=\"{{ classes?.addButton }}\" [style.background]=\"addButton?.background\" [style.borderColor]=\"addButton?.background\" [style.color]=\"addButton?.textColor\"><span class=\"fa fa-plus\"></span> Add</button>\n                </div>\n                <div class=\"col-md-1 text-right rmpd export-cnt\" *ngIf=\"enableDataExports\">\n                    <div class=\"dropdown\">\n                        <button class=\"btn btn-default {{ classes?.exportButton }} dropdown-toggle\" type=\"button\" data-toggle=\"dropdown\">\n                            <i class=\"fa fa-angle-down\"></i>\n                        </button>\n                        <ul class=\"dropdown-menu dropdown-menu-export {{ classes?.exportDropdown }}\">\n                            <li class=\"dropdown-header\">{{ dataExportsConfig?.title }}. <input type=\"checkbox\" (change)=\"checkToExportOption($event?.target?.checked)\" style=\"position: relative;top:0.3em;\"> <sup style=\"font-size:10px;color:#999;\"> All</sup></li> \n                            <li class=\"divider\"></li>\n                            <li class=\"dropdown-submenu\" *ngFor=\"let export of dataExportsConfig?.exportsTo\" (click)=\"exportDocumentsAs(export)\">\n                                <a href=\"javascript:void(0)\" class=\"btn btn-default btn-sm form-control\" *ngIf=\"export=='print'\"><i class=\"glyphicon glyphicon-print\"></i> Print</a>\n                                <a href=\"javascript:void(0)\" class=\"btn btn-default btn-sm form-control\" *ngIf=\"export=='pdf'\" style=\"color:#ff0000\"><i class=\"glyphicon glyphicon-file\"></i> PDF</a>\n                                <a href=\"javascript:void(0)\" class=\"btn btn-default btn-sm form-control\" *ngIf=\"export=='excel'\" style=\"color:#009900;\"><i class=\"glyphicon glyphicon-file\"></i> Excel</a>            \n                                <a href=\"javascript:void(0)\" class=\"btn btn-default btn-sm form-control\" *ngIf=\"export=='word'\" style=\"color:#335599;\"><i class=\"glyphicon glyphicon-file\"></i> Word</a>                        \n                            </li>\n                            <li class=\"divider\"></li>\n                            <li class=\"dropdown-header\">\n                                <span  *ngIf=\"!isExportAll\">{{ (displayData | async)?.length }}</span>\n                                <span  *ngIf=\"isExportAll\">{{ (service?.flxData | async)?.length }}</span> \n                            </li>\n                        </ul>\n                    </div>\n                </div>\n            </div>\n        </div>\n    </div>\n    <div class=\"container-fluid\">\n        <div class=\"row\">\n            <div class=\"col-md-12 rmpd table-responsive flx-tabl-container\">\n                <table class=\"table {{ classes?.tableType }} th-table\">\n                    <thead class=\"{{ classes?.tableHeader }} flx-table-thead\">\n                        <tr>\n                            <!-- Actions buttons header left before numbers -->\n                            <th *ngIf=\"hasActionButtons && (actionButtonsConfig?.position=='left' && actionButtonsConfig?.beforeNumbers)\">{{ actionHeader }} \n                                <input type=\"checkbox\" [checked]=\"service?.multipleDeletion?.length>0\" (change)=\"addRemove($event?.target?.checked)\" *ngIf=\"enableMultipleSelection\">\n                                <button class=\"btn btn-danger btn-xs flx-multiple-deletion-button\" *ngIf=\"enableMultipleSelection && service?.multipleDeletion?.length>0\" (click)=\"confirmDelete()\"><span [class]=\"multipleSelectButton?.icon\"></span> {{ multipleSelectButton?.text }}</button>\n                            </th>\n\n\n                            <th *ngIf=\"!hideNumbers\">N<sup><u>o</u></sup></th>\n\n                            <!-- Actions buttons header left after numbers -->\n                            <th *ngIf=\"hasActionButtons && (actionButtonsConfig?.position=='left' && !actionButtonsConfig?.beforeNumbers)\">{{ actionHeader }} \n                                <input type=\"checkbox\" [checked]=\"service?.multipleDeletion?.length>0\" (change)=\"addRemove($event?.target?.checked)\" *ngIf=\"enableMultipleSelection\">\n                                <button class=\"btn btn-danger btn-xs flx-multiple-deletion-button\" *ngIf=\"enableMultipleSelection && service?.multipleDeletion?.length>0\" (click)=\"confirmDelete()\"><span [class]=\"multipleSelectButton?.icon\"></span> {{ multipleSelectButton?.text }}</button>\n                            </th>\n\n\n                            <th *ngFor=\"let header of headers\">{{ header }}</th>\n\n                            <!-- Actions buttons header right -->\n                            <th *ngIf=\"hasActionButtons  && (actionButtonsConfig?.position=='right')\">{{ actionHeader }} \n                                <input type=\"checkbox\" [checked]=\"service?.multipleDeletion?.length>0\" (change)=\"addRemove($event?.target?.checked)\" *ngIf=\"enableMultipleSelection\">\n                                <button class=\"btn btn-danger btn-xs flx-multiple-deletion-button\" *ngIf=\"enableMultipleSelection && service?.multipleDeletion?.length>0\" (click)=\"confirmDelete()\"><span [class]=\"multipleSelectButton?.icon\"></span> {{ multipleSelectButton?.text }}</button>\n                            </th>\n                        </tr>\n                    </thead>\n                </table>\n                <perfect-scrollbar class=\"data-scroll-content {{ classes?.dataScrollContent }}\">\n                    <table class=\"table {{ classes?.tableType }} flx-table-data\">                        \n                        <tbody class=\"flxdatatablebodyscrollbar\">\n                            <tr *ngIf=\"!service?.loadFinish\">                    \n                                <td colspan=\"20\" class=\"text-center\">\n                                    <img *ngIf=\"spinnerSrc\" [class]=\"classes?.spinner\" [src]=\"spinnerSrc\" style=\"margin-top:1em;margin-bottom:1em;\">\n                                </td>\n                            </tr>\n                            <tr class=\"flxuidatatablerow\" id=\"flxdatatable_singlerow\" *ngFor=\"let data of displayData | async;let i=index\">\n                                \n                                <!-- Buttons Left Before Numbers -->\n                                <td class=\"table-buttons\" *ngIf=\"hasActionButtons && (actionButtonsConfig?.position=='left' && actionButtonsConfig?.beforeNumbers)\" scope=\"row\">\n                                    <span *ngFor=\"let aButton of actionButtons;let buttonIndex=index\">\n                                        <button type=\"button\" (click)=\"actionButtonClicked(i,buttonIndex)\" class=\"btn {{ aButton?.class }}\">\n                                            <div class=\"toltip\" class=\"flx-tooltip\" [ngClass]=\"{'flx-tooltip-left':aButton?.tooltipPosition=='left','flx-tooltip-bottom':aButton?.tooltipPosition=='bottom','flx-tooltip-right':aButton?.tooltipPosition=='right'}\" *ngIf=\"aButton?.tooltip\">{{ aButton?.tooltip }}</div>\n                                            <span class=\"action-button-icon-left\" [class]=\"aButton?.icon\" *ngIf=\"!aButton?.iconPosition || aButton?.iconPosition!='right'\"></span>\n                                            <span class=\"button-text\"> {{ aButton?.text }} </span>\n                                            <span [class]=\"aButton?.icon\" *ngIf=\"aButton?.iconPosition=='right'\"></span>\n                                        </button>\n                                    </span> \n                                    <input type=\"checkbox\" checked (change)=\"addRemoveDeleteItem(data[multipleSelectKey],i,$event?.target?.checked)\" *ngIf=\"enableMultipleSelection && service?.multipleDeletion?.length>0\">\n                                </td>\n                                \n                                <!-- Numbers -->\n                                <td class=\"{{ classes?.tableData }}\" *ngIf=\"!hideNumbers\" style=\"color: #999;\">{{ offset+i }}</td>\n                                \n                                <!-- Buttons Left After Numbers -->\n                                <td class=\"table-buttons\" *ngIf=\"hasActionButtons && (actionButtonsConfig?.position=='left' && !actionButtonsConfig?.beforeNumbers)\" scope=\"row\">\n                                    <span *ngFor=\"let aButton of actionButtons;let buttonIndex=index\">\n                                        <button type=\"button\" (click)=\"actionButtonClicked(i,buttonIndex)\" class=\"btn {{ aButton?.class }}\">\n                                            <div class=\"toltip\" class=\"flx-tooltip\" [ngClass]=\"{'flx-tooltip-left':aButton?.tooltipPosition=='left','flx-tooltip-bottom':aButton?.tooltipPosition=='bottom','flx-tooltip-right':aButton?.tooltipPosition=='right'}\" *ngIf=\"aButton?.tooltip\">{{ aButton?.tooltip }}</div>\n                                            <span class=\"action-button-icon-left\" [class]=\"aButton?.icon\" *ngIf=\"!aButton?.iconPosition || aButton?.iconPosition!='right'\"></span>\n                                            <span class=\"button-text\"> {{ aButton?.text }} </span>\n                                            <span [class]=\"aButton?.icon\" *ngIf=\"aButton?.iconPosition=='right'\"></span>\n                                        </button>\n                                    </span> \n                                    <input type=\"checkbox\" checked (change)=\"addRemoveDeleteItem(data[multipleSelectKey],i,$event?.target?.checked)\" *ngIf=\"enableMultipleSelection && service?.multipleDeletion?.length>0\">\n                                </td>\n                                \n                                <!-- Main -->\n                                <td class=\"{{ classes?.tableData }}\" *ngFor=\"let dataKey of dataKeys;let x=index\">\n                                    <img *ngIf=\"hasImageEmbeded() && x==embedPictures?.index; else nopicture\" [class]=\"'img-fall-back ' +embedPictures?.class\" [src]=\"embedPictures?.rootFolder+data[dataKey]\" [flx-ui-datatable-img-fallback]=\"embedPictures?.fallbackUrl\" >\n                                    <ng-template #nopicture>\n                                        <section *ngIf=\"textSubstring?.atIndexes && (textSubstring?.atIndexes | haslimittext:x);else nolimittext\">\n                                            {{ textSubstring | limittext: data[dataKey] }}\n                                        </section>\n                                        <ng-template #nolimittext>\n                                            <span>{{ data[dataKey] }}</span>\n                                        </ng-template>\n                                    </ng-template> \n                                </td>\n                                <!-- Buttons -->\n                                <td class=\"table-buttons\" *ngIf=\"hasActionButtons && actionButtonsConfig?.position!='left'\">\n                                    <span *ngFor=\"let aButton of actionButtons;let buttonIndex=index\">\n                                        <button type=\"button\" (click)=\"actionButtonClicked(i,buttonIndex)\" class=\"btn {{ aButton?.class }}\">\n                                            <div class=\"toltip\" class=\"flx-tooltip\" [ngClass]=\"{'flx-tooltip-left':aButton?.tooltipPosition=='left','flx-tooltip-bottom':aButton?.tooltipPosition=='bottom','flx-tooltip-right':aButton?.tooltipPosition=='right'}\" *ngIf=\"aButton?.tooltip\">{{ aButton?.tooltip }}</div>\n                                            <span class=\"action-button-icon-left\" [class]=\"aButton?.icon\" *ngIf=\"!aButton?.iconPosition || aButton?.iconPosition!='right'\"></span>\n                                            <span class=\"button-text\"> {{ aButton?.text }} </span>\n                                            <span [class]=\"aButton?.icon\" *ngIf=\"aButton?.iconPosition=='right'\"></span>\n                                        </button>\n                                    </span> \n                                    <input type=\"checkbox\" checked (change)=\"addRemoveDeleteItem(data[multipleSelectKey],i,$event?.target?.checked)\" *ngIf=\"enableMultipleSelection && service?.multipleDeletion?.length>0\">\n                                </td>\n                            </tr>\n                            <tr *ngIf=\"tData?.length<1\">\n                                <td colspan=\"10\" class=\"text-center\" *ngIf=\"service?.loadFinish\">\n                                    <span style=\"color:#ff0000;font-size:13px;\">No data found</span> <br>\n                                    <button type=\"button\" style=\"margin-top:1em;\" (click)=\"reload()\" class=\"btn btn-default {{ classes?.reloadbutton }}\" color=\"primary\"><span class=\"fa fa-refresh\"></span> Reload</button>\n                                </td>\n                            </tr>\n                        </tbody>\n                    </table>\n                </perfect-scrollbar>\n                <div class=\"row\">\n                    <div class=\"col-md-12 rmpd flx-total-data\" *ngIf=\"showBottomInfo\">\n                        <div class=\"row\" *ngIf=\"!isSearching;else bottomInfoSearching\">\n                            <div class=\"col-md-4 text-left rmpd\">Total pagination: <span> {{ service?.totalItems | ceil: limit }}</span></div>\n                            <div class=\"col-md-4 text-center rmpd\"># of items per pagination: <span>{{ limit }}</span></div>\n                            <div class=\"col-md-4 text-right rmpd\">Total items: <span>{{ (service?.flxData | async)?.length }}</span></div>\n                        </div>\n                        <ng-template #bottomInfoSearching>\n                            <div class=\"row\">\n                                <div class=\"col-md-12 text-right rmpd\">Total search result: <span>{{ searchResultFound?.length }}</span></div>\n                            </div>\n                        </ng-template>\n                    </div>\n                </div>\n            </div>\n        </div>\n    </div>\n</div>",
                styles: [".flx-ui-datatable-main{background-color:#fff;padding-top:1em;padding-bottom:1em}.flx-ui-datatable-main .btn-danger{background-color:#f50057;border:1px solid #f50057;box-shadow:0 3px 5px 1px #ddd;-moz-box-shadow:0 3px 5px 1px #ddd;-webkit-box-shadow:0 3px 5px 1px #ddd;-o-box-shadow:0 3px 5px 1px #ddd;-ms-box-shadow:0 3px 5px 1px #ddd;border-radius:3px;margin-left:.3em}.flx-ui-datatable-main #flxdatatablebodyscrollbar,.flx-ui-datatable-main .table-responsive,.flx-ui-datatable-main .table-responsive table{width:100%}.flx-ui-datatable-main img.img-fall-back{width:30px;height:30px}.flx-ui-datatable-main .pagination-select input[type=text],.flx-ui-datatable-main .pagination-select select,.flx-ui-datatable-main .search-bar input[type=text],.flx-ui-datatable-main .search-bar select{border-top:none!important;border-left:none!important;border-right:none!important;border-bottom:2px solid #ddd!important;border-radius:0!important;box-shadow:0 0 0 0 transparent!important;-moz-box-shadow:0 0 0 0 transparent!important;-webkit-box-shadow:0 0 0 0 transparent!important;-o-box-shadow:0 0 0 0 transparent!important;-ms-box-shadow:0 0 0 0 transparent!important}.flx-ui-datatable-main .pagination-select input[type=text]:focus,.flx-ui-datatable-main .pagination-select select:focus,.flx-ui-datatable-main .search-bar input[type=text]:focus,.flx-ui-datatable-main .search-bar select:focus{border-bottom-color:#359!important;transition:.5s;-moz-transition:.5s;-webkit-transition:.5s;-o-transition:.5s;-ms-transition:.5s}.flx-ui-datatable-main .pagination-select select,.flx-ui-datatable-main .search-bar select{-webkit-appearance:none;appearance:none;-moz-appearance:none}.flx-ui-datatable-main .dropdown-toggle::after{display:none}.flx-ui-datatable-main .dropdown-menu-export{margin-left:-130px;padding-left:1em;padding-right:1em;width:100px!important;margin-top:1em;border:1px solid #ddd!important}.flx-ui-datatable-main .dropdown-menu-export::before{content:\"\";width:auto;height:auto;position:absolute;top:-20px;right:3px;border:10px solid;border-color:transparent transparent #ddd}.flx-ui-datatable-main .flx-datatable-pagination{padding-top:.5em}.flx-ui-datatable-main .flx-datatable-pagination button{width:35px!important;height:35px!important;border-radius:50em!important;border:none!important;box-shadow:0 3px 10px 0 #b3c4e6;-moz-box-shadow:0 3px 10px 0 #b3c4e6;-webkit-box-shadow:0 3px 10px 0 #b3c4e6;-o-box-shadow:0 3px 10px 0 #b3c4e6;-ms-box-shadow:0 3px 10px 0 #b3c4e6;background-color:#359;color:#fff;font-size:23px;font-weight:700;position:absolute;top:0}.flx-ui-datatable-main .flx-datatable-pagination button:first-child{left:0}.flx-ui-datatable-main .flx-datatable-pagination button:first-child .flx-datatable-tooltip-prev{position:absolute;left:0;font-size:13px;font-weight:400;color:#fff;background-color:#359;padding-left:.3em;padding-right:.3em;border-radius:3px;margin-left:-57px;margin-top:.3em;box-shadow:0 3px 10px 0 #b3c4e6!important;-moz-box-shadow:0 3px 10px 0 #b3c4e6!important;-webkit-box-shadow:0 3px 10px 0 #b3c4e6!important;-o-box-shadow:0 3px 10px 0 #b3c4e6!important;-ms-box-shadow:0 3px 10px 0 #b3c4e6!important;visibility:hidden}.flx-ui-datatable-main .flx-datatable-pagination button:hover>span.flx-datatable-tooltip-next,.flx-ui-datatable-main .flx-datatable-pagination button:hover>span.flx-datatable-tooltip-prev{visibility:visible}.flx-ui-datatable-main .flx-datatable-pagination .flx-pagination-end{background-color:#f50057!important;box-shadow:0 3px 10px 0 #ffc2d8!important;-moz-box-shadow:0 3px 10px 0 #ffc2d8!important;-webkit-box-shadow:0 3px 10px 0 #ffc2d8!important;-o-box-shadow:0 3px 10px 0 #ffc2d8!important;-ms-box-shadow:0 3px 10px 0 #ffc2d8!important;cursor:not-allowed}.flx-ui-datatable-main .flx-datatable-pagination .flx-pagination-end:hover>span.flx-datatable-tooltip-next,.flx-ui-datatable-main .flx-datatable-pagination .flx-pagination-end:hover>span.flx-datatable-tooltip-prev{visibility:hidden}.flx-ui-datatable-main .flx-datatable-pagination button:last-child{right:0}.flx-ui-datatable-main .flx-datatable-pagination button:last-child .flx-datatable-tooltip-next{position:absolute;left:0;font-size:13px;font-weight:400;color:#fff;background-color:#359;padding-left:.3em;padding-right:.3em;border-radius:3px;margin-left:35px;margin-top:.5em;box-shadow:0 3px 10px 0 #b3c4e6!important;-moz-box-shadow:0 3px 10px 0 #b3c4e6!important;-webkit-box-shadow:0 3px 10px 0 #b3c4e6!important;-o-box-shadow:0 3px 10px 0 #b3c4e6!important;-ms-box-shadow:0 3px 10px 0 #b3c4e6!important;visibility:hidden}.flx-ui-datatable-main .export-cnt button{border-radius:50em!important}.flx-ui-datatable-main .flx-tabl-container table.th-table{margin-top:.5em}.flx-ui-datatable-main table tbody tr{padding-top:0!important}.flx-ui-datatable-main table tbody tr td{padding-top:.5em;border-top:1px solid #f0f0f0;border-bottom:1px solid #f0f0f0}.flx-ui-datatable-main table tbody tr td button{margin-right:.3em;margin-left:0}.flx-ui-datatable-main table tbody tr td button div.flx-tooltip{position:absolute;background-color:rgba(32,27,27,.808);text-align:center;font-size:13px;color:#fff;border-radius:3px;box-shadow:0 3px 20px 0 #4b4949;-moz-box-shadow:0 3px 20px 0 #4b4949;-webkit-box-shadow:0 3px 20px 0 #4b4949;-o-box-shadow:0 3px 20px 0 #4b4949;-ms-box-shadow:0 3px 20px 0 #4b4949;margin-left:-2.5em;margin-top:-2.8em;visibility:hidden;width:80px;padding:.3em .5em}.flx-ui-datatable-main table tbody tr td button .flx-tooltip-left{margin-left:-95px!important;margin-top:-.3em!important}.flx-ui-datatable-main table tbody tr td button .flx-tooltip-bottom{margin-top:2.3em!important}.flx-ui-datatable-main table tbody tr td button .flx-tooltip-right{margin-left:28px!important;margin-top:-.3em!important}.flx-ui-datatable-main table tbody tr td button:hover>div.flx-tooltip{transition:.3s;visibility:visible}.flx-ui-datatable-main table tbody tr td.table-buttons{padding-top:.2em;padding-bottom:.2em}.flx-ui-datatable-main table tbody tr:nth-of-type(even){background-color:#f8f9fa}.flx-ui-datatable-main table tbody tr:nth-of-type(odd){background-color:#fff}.flx-ui-datatable-main .btn-danger:hover{background-color:#ff146b;border:1px solid #ff146b;box-shadow:0 3px 10px 1px #ff5fb6;-moz-box-shadow:0 3px 10px 1px #ff5fb6;-webkit-box-shadow:0 3px 10px 1px #ff5fb6;-o-box-shadow:0 3px 10px 1px #ff5fb6;-ms-box-shadow:0 3px 10px 1px #ff5fb6;transition:.5s;-moz-transition:.5s;-webkit-transition:.5s;-o-transition:.5s;-ms-transition:.5s}.flx-ui-datatable-main .btn-danger:focus{background-color:#f50057;border:1px solid #f50057}.flx-ui-datatable-main .btn-white{background-color:#fff}.flx-ui-datatable-main .btn-dark{background-color:#222!important}.flx-ui-datatable-main .btn-primary{background-color:#359;color:#fff;border:1px solid #359;box-shadow:0 3px 5px 1px #ddd;-moz-box-shadow:0 3px 5px 1px #ddd;-webkit-box-shadow:0 3px 5px 1px #ddd;-o-box-shadow:0 3px 5px 1px #ddd;-ms-box-shadow:0 3px 5px 1px #ddd;border-radius:3px}.flx-ui-datatable-main .btn-primary:hover{background-color:#4769ad;border:1px solid #4769ad;box-shadow:0 3px 10px 0 #b3c4e6;-moz-box-shadow:0 3px 10px 0 #b3c4e6;-webkit-box-shadow:0 3px 10px 0 #b3c4e6;-o-box-shadow:0 3px 10px 0 #b3c4e6;-ms-box-shadow:0 3px 10px 0 #b3c4e6;transition:.5s;-moz-transition:.5s;-webkit-transition:.5s;-o-transition:.5s;-ms-transition:.5s}.flx-ui-datatable-main .btn-primary:focus{background-color:#359;border:1px solid #359}.flx-ui-datatable-main .btn-large{padding-top:1em!important;padding-bottom:1em!important}.flx-ui-datatable-main .btn-medium{padding-top:.7em!important;padding-bottom:.7em!important}.flx-ui-datatable-main .btn-success{box-shadow:0 3px 5px 1px #ddd;-moz-box-shadow:0 3px 5px 1px #ddd;-webkit-box-shadow:0 3px 5px 1px #ddd;-o-box-shadow:0 3px 5px 1px #ddd;-ms-box-shadow:0 3px 5px 1px #ddd;border-radius:3px;background-color:#5cb85c;border:1px solid #5cb85c}.flx-ui-datatable-main .btn-success:hover{background-color:#70cc70;border:1px solid #70cc70;box-shadow:0 3px 10px 1px #9df99d;-moz-box-shadow:0 3px 10px 1px #9df99d;-webkit-box-shadow:0 3px 10px 1px #9df99d;-o-box-shadow:0 3px 10px 1px #9df99d;-ms-box-shadow:0 3px 10px 1px #9df99d;transition:.5s;-moz-transition:.5s;-webkit-transition:.5s;-o-transition:.5s;-ms-transition:.5s}.flx-ui-datatable-main .btn-success:focus{background-color:#5cb85c;border:1px solid #5cb85c}.flx-ui-datatable-main .btn-default{background-color:#fff;box-shadow:0 3px 5px 1px #eee;-moz-box-shadow:0 3px 5px 1px #eee;-webkit-box-shadow:0 3px 5px 1px #eee;-o-box-shadow:0 3px 5px 1px #eee;-ms-box-shadow:0 3px 5px 1px #eee;border-radius:3px;border:1px solid #ddd}.flx-ui-datatable-main .btn-default:hover{background-color:#fff;border:1px solid #e7e7e7;box-shadow:0 3px 10px 1px #e2e2e2;-moz-box-shadow:0 3px 10px 1px #e2e2e2;-webkit-box-shadow:0 3px 10px 1px #e2e2e2;-o-box-shadow:0 3px 10px 1px #e2e2e2;-ms-box-shadow:0 3px 10px 1px #e2e2e2;transition:.5s;-moz-transition:.5s;-webkit-transition:.5s;-o-transition:.5s;-ms-transition:.5s}.flx-ui-datatable-main .btn-default:focus{background-color:#fff;border:1px solid #ddd}.flx-ui-datatable-main .btn-secondary{box-shadow:0 3px 5px 1px #ddd;-moz-box-shadow:0 3px 5px 1px #ddd;-webkit-box-shadow:0 3px 5px 1px #ddd;-o-box-shadow:0 3px 5px 1px #ddd;-ms-box-shadow:0 3px 5px 1px #ddd;border-radius:3px;background-color:#1e88e5;color:#fff}.flx-ui-datatable-main .btn-secondary:hover{color:#fff;background-color:#2892ef;border:1px solid #2892ef;box-shadow:0 3px 10px 1px #55bfff;-moz-box-shadow:0 3px 10px 1px #55bfff;-webkit-box-shadow:0 3px 10px 1px #55bfff;-o-box-shadow:0 3px 10px 1px #55bfff;-ms-box-shadow:0 3px 10px 1px #55bfff;transition:.5s;-moz-transition:.5s;-webkit-transition:.5s;-o-transition:.5s;-ms-transition:.5s}.flx-ui-datatable-main .btn-secondary:focus{color:#fff}.flx-ui-datatable-main .pagination-button{background-color:#359;color:#fff}.flx-ui-datatable-main .table-font{font-family:Khula,sans-serif!important}.flx-ui-datatable-main .table-header-icon{position:absolute;right:.2em;width:80px;height:80px;font-size:50px;margin-top:-30px;border-radius:5px!important}.flx-ui-datatable-main .table-title{background-color:#359;color:#fff;border-radius:2px;padding:1em;font-size:15px;font-weight:700;margin-bottom:1.5em;font-family:Roboto,sans-serif;box-shadow:0 1px 5px 1px #ddd;-moz-box-shadow:0 1px 5px 1px #ddd;-webkit-box-shadow:0 1px 5px 1px #ddd;-o-box-shadow:0 1px 5px 1px #ddd;-ms-box-shadow:0 1px 5px 1px #ddd}.flx-ui-datatable-main .danger{background-color:#f50057;color:#fff}.flx-ui-datatable-main .primary{background-color:#359;color:#fff}.flx-ui-datatable-main .success{background-color:#5cb85c;color:#fff}.flx-ui-datatable-main .default{background-color:#fff;color:#000}.flx-ui-datatable-main .secondary{background-color:#1e88e5;color:#fff}.flx-ui-datatable-main .btn-clear{border:none!important;box-shadow:none!important}.flx-ui-datatable-main .input-group,.flx-ui-datatable-main .input-group input{background-color:transparent!important}.flx-ui-datatable-main .input-group-addon{border:none!important;padding:0!important;box-shadow:none!important;background-color:transparent!important}.flx-ui-datatable-main .input-group-addon button{border:1px solid transparent!important;box-shadow:none!important;border-top:none!important;border-bottom:none!important;background-color:transparent!important;border-radius:50em!important;color:#359;width:30px;height:30px}.flx-ui-datatable-main .input-group-addon button i{font-size:18px}.flx-ui-datatable-main .input-group-addon button:disabled{background-color:transparent!important}.flx-ui-datatable-main .input-group-addon button:disabled i{color:#f50057}.flx-ui-datatable-main .data-scroll-content{max-width:100%;max-height:400px;margin-top:-1em}.flx-ui-datatable-main .data-scroll-content table tbody tr:first-child td,.flx-ui-datatable-main .flx-table-thead tr th{border-top:none!important}"]
            }]
    }], function () { return [{ type: ɵngcc1.FormBuilder }, { type: FlxUiDatatableService }]; }, { classes: [{
            type: Input
        }], headers: [{
            type: Input
        }], lazyloadingConfig: [{
            type: Input
        }], embedPictures: [{
            type: Input
        }], dataKeys: [{
            type: Input
        }], enableDataExports: [{
            type: Input
        }], dataExportsConfig: [{
            type: Input
        }], searchKeys: [{
            type: Input
        }], dataSrcKey: [{
            type: Input
        }], hasActionButtons: [{
            type: Input
        }], hideNumbers: [{
            type: Input
        }], enableMultipleSelection: [{
            type: Input
        }], multipleSelectKey: [{
            type: Input
        }], hasAddButton: [{
            type: Input
        }], dataUrl: [{
            type: Input
        }], actionButtonStart: [{
            type: Input
        }], multipleSelectButton: [{
            type: Input
        }], searchPlaceholder: [{
            type: Input
        }], actionHeader: [{
            type: Input
        }], limit: [{
            type: Input
        }], spinnerSrc: [{
            type: Input
        }], actionButtons: [{
            type: Input
        }], paginationButtons: [{
            type: Input
        }], tableHeader: [{
            type: Input
        }], searchButton: [{
            type: Input
        }], addButton: [{
            type: Input
        }], searchBar: [{
            type: Input
        }], actionButtonsConfig: [{
            type: Input
        }], firstActionButtonClicked: [{
            type: Output
        }], secondActionButtonClicked: [{
            type: Output
        }], thirdActionButtonClicked: [{
            type: Output
        }], multipleSelectClicked: [{
            type: Output
        }], addButtonClicked: [{
            type: Output
        }], textSubstring: [{
            type: Input
        }], conditionManipulator: [{
            type: Input
        }], showBottomInfo: [{
            type: Input
        }] }); })();
    return FlxUiDatatableComponent;
}());
export { FlxUiDatatableComponent };
function FlxUiDatatableComponent_tsickle_Closure_declarations() {
    /** @type {?} */
    FlxUiDatatableComponent.prototype.classes;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.headers;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.lazyloadingConfig;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.embedPictures;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.dataKeys;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.enableDataExports;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.dataExportsConfig;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.showBottomInfo;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.searchKeys;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.dataSrcKey;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.hasActionButtons;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.hideNumbers;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.enableMultipleSelection;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.multipleSelectKey;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.hasAddButton;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.dataUrl;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.actionButtonStart;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.multipleSelectButton;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.searchPlaceholder;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.actionHeader;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.limit;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.spinnerSrc;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.actionButtons;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.paginationButtons;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.tableHeader;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.searchButton;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.addButton;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.searchBar;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.actionButtonsConfig;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.firstActionButtonClicked;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.secondActionButtonClicked;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.thirdActionButtonClicked;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.multipleSelectClicked;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.addButtonClicked;
    /**
     * 18/12/2018
     * @type {?}
     */
    FlxUiDatatableComponent.prototype.textSubstring;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.conditionManipulator;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.isExportAll;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.searchForm;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.tData;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.behavior;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.searchDataTempOffset;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.displayData;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.offset;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.reloadUrl;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.isSearching;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.searchResultFound;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.__form;
    /** @type {?} */
    FlxUiDatatableComponent.prototype.service;
}
import { Directive, ElementRef } from '@angular/core';
var ImageFallBack = /** @class */ (function () {
    function ImageFallBack(el) {
        this.isApplied = false;
        this.EVENT_TYPE = 'error';
        this.el = el.nativeElement;
        this.el.addEventListener(this.EVENT_TYPE, this.onError.bind(this));
    }
    /**
     * @return {?}
     */
    ImageFallBack.prototype.onError = /**
     * @return {?}
     */
    function () {
        this.removeEvents();
        if (!this.isApplied) {
            this.isApplied = true;
            this.el.setAttribute('src', this.imgSrc);
        }
    };
    /**
     * @return {?}
     */
    ImageFallBack.prototype.removeEvents = /**
     * @return {?}
     */
    function () {
        this.el.removeEventListener(this.EVENT_TYPE, this.onError);
    };
    /**
     * @return {?}
     */
    ImageFallBack.prototype.ngOnDestroy = /**
     * @return {?}
     */
    function () {
        this.removeEvents();
    };
    /** @nocollapse */
    ImageFallBack.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    ImageFallBack.propDecorators = {
        imgSrc: [{ type: Input, args: ['flx-ui-datatable-img-fallback',] }]
    };
ImageFallBack.ɵfac = function ImageFallBack_Factory(t) { return new (t || ImageFallBack)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef)); };
ImageFallBack.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: ImageFallBack, selectors: [["", "flx-ui-datatable-img-fallback", ""]], inputs: { imgSrc: ["flx-ui-datatable-img-fallback", "imgSrc"] } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ImageFallBack, [{
        type: Directive,
        args: [{
                selector: '[flx-ui-datatable-img-fallback]'
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }]; }, { imgSrc: [{
            type: Input,
            args: ['flx-ui-datatable-img-fallback']
        }] }); })();
    return ImageFallBack;
}());
export { ImageFallBack };
function ImageFallBack_tsickle_Closure_declarations() {
    /** @type {?} */
    ImageFallBack.prototype.imgSrc;
    /** @type {?} */
    ImageFallBack.prototype.el;
    /** @type {?} */
    ImageFallBack.prototype.isApplied;
    /** @type {?} */
    ImageFallBack.prototype.EVENT_TYPE;
}
