// flx-ui-datatable/esm5/lib/flx-ui-datatable.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ReactiveFormsModule, FormsModule } from '@angular/forms';
import { FlxUiDatatableComponent, ImageFallBack } from './flx-ui-datatable.component';
import { HttpModule } from '@angular/http';
import { FlxUiDatatableService } from './flx-ui-datatable.service';
import { Ceil } from './pipes/ceil.pipe';
import { HasLimitText } from './pipes/hasLimitText.pipe';
import { LimitText } from './pipes/limitText.pipe';
import { ProcessedData } from './pipes/processeddata.pipe';
import { PerfectScrollbarModule } from 'ngx-perfect-scrollbar';
import { PERFECT_SCROLLBAR_CONFIG } from 'ngx-perfect-scrollbar';
import * as ɵngcc0 from '@angular/core';
var /** @type {?} */ DEFAULT_PERFECT_SCROLLBAR_CONFIG = {
    suppressScrollX: true
};
var ɵ0 = DEFAULT_PERFECT_SCROLLBAR_CONFIG;
var FlxUiDatatableModule = /** @class */ (function () {
    function FlxUiDatatableModule() {
    }
FlxUiDatatableModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: FlxUiDatatableModule });
FlxUiDatatableModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function FlxUiDatatableModule_Factory(t) { return new (t || FlxUiDatatableModule)(); }, providers: [FlxUiDatatableService,
        {
            provide: PERFECT_SCROLLBAR_CONFIG,
            useValue: ɵ0
        }], imports: [[
            CommonModule, ReactiveFormsModule, FormsModule, HttpModule, PerfectScrollbarModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(FlxUiDatatableModule, { declarations: function () { return [FlxUiDatatableComponent, Ceil, ImageFallBack, HasLimitText, LimitText, ProcessedData]; }, imports: function () { return [CommonModule, ReactiveFormsModule, FormsModule, HttpModule, PerfectScrollbarModule]; }, exports: function () { return [FlxUiDatatableComponent, Ceil]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FlxUiDatatableModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule, ReactiveFormsModule, FormsModule, HttpModule, PerfectScrollbarModule
                ],
                declarations: [
                    FlxUiDatatableComponent, Ceil, ImageFallBack, HasLimitText, LimitText, ProcessedData
                ],
                exports: [FlxUiDatatableComponent, Ceil],
                providers: [FlxUiDatatableService,
                    {
                        provide: PERFECT_SCROLLBAR_CONFIG,
                        useValue: ɵ0
                    }]
            }]
    }], function () { return []; }, null); })();
    return FlxUiDatatableModule;
}());
export { FlxUiDatatableModule };
export { ɵ0 };
