// flx-ui-datatable/fesm2015/flx-ui-datatable.js
import { Injectable, Component, Input, Output, EventEmitter, Directive, ElementRef, Pipe, NgModule } from '@angular/core';
import { Http, Headers, HttpModule } from '@angular/http';
import { map, retry } from 'rxjs/operators';
import { BehaviorSubject } from 'rxjs';
import { Validators, FormBuilder, ReactiveFormsModule, FormsModule } from '@angular/forms';
import { CommonModule } from '@angular/common';
import { PerfectScrollbarModule, PERFECT_SCROLLBAR_CONFIG } from 'ngx-perfect-scrollbar';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/http';
import * as ɵngcc2 from '@angular/forms';
import * as ɵngcc3 from '@angular/common';
import * as ɵngcc4 from 'ngx-perfect-scrollbar';

function FlxUiDatatableComponent_option_8_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "option", 25);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const paging_r56 = ctx.$implicit;
    ɵngcc0.ɵɵproperty("value", paging_r56 == null ? null : paging_r56.value);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(paging_r56 == null ? null : paging_r56.label);
} }
function FlxUiDatatableComponent_span_14_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "ceil");
    ɵngcc0.ɵɵpipe(3, "ceil");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r40 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate2("", ɵngcc0.ɵɵpipeBind2(2, 2, ctx_r40.service == null ? null : ctx_r40.service.dataOffset, ctx_r40.limit), " / ", ɵngcc0.ɵɵpipeBind2(3, 5, ctx_r40.service == null ? null : ctx_r40.service.totalItems, ctx_r40.limit), "");
} }
function FlxUiDatatableComponent_ng_template_15_Template(rf, ctx) { }
const _c0 = function (a0) { return { border: a0 }; };
function FlxUiDatatableComponent_input_22_Template(rf, ctx) { if (rf & 1) {
    const _r58 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "input", 26);
    ɵngcc0.ɵɵlistener("keyup", function FlxUiDatatableComponent_input_22_Template_input_keyup_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r58); const ctx_r57 = ɵngcc0.ɵɵnextContext(); return ctx_r57.filterData($event == null ? null : $event.target == null ? null : $event.target.value); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r43 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵstyleProp("background", ctx_r43.searchBar == null ? null : ctx_r43.searchBar.background, ɵngcc0.ɵɵdefaultStyleSanitizer)("color", ctx_r43.searchBar == null ? null : ctx_r43.searchBar.textColor);
    ɵngcc0.ɵɵproperty("ngStyle", ɵngcc0.ɵɵpureFunction1(6, _c0, (ctx_r43.searchBar == null ? null : ctx_r43.searchBar.borderSize) + " solid " + (ctx_r43.searchBar == null ? null : ctx_r43.searchBar.borderColor)))("placeholder", ctx_r43.searchPlaceholder);
} }
function FlxUiDatatableComponent_form_23_Template(rf, ctx) { if (rf & 1) {
    const _r61 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "form", 27, 28);
    ɵngcc0.ɵɵlistener("ngSubmit", function FlxUiDatatableComponent_form_23_Template_form_ngSubmit_0_listener() { ɵngcc0.ɵɵrestoreView(_r61); const _r59 = ɵngcc0.ɵɵreference(1); const ctx_r60 = ɵngcc0.ɵɵnextContext(); return ctx_r60.searchDataInApi(_r59 == null ? null : _r59.value, _r59); });
    ɵngcc0.ɵɵelementStart(2, "div", 29);
    ɵngcc0.ɵɵelement(3, "input", 30);
    ɵngcc0.ɵɵelementStart(4, "span", 31);
    ɵngcc0.ɵɵelementStart(5, "button", 32);
    ɵngcc0.ɵɵelement(6, "i", 33);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const _r59 = ɵngcc0.ɵɵreference(1);
    const ctx_r44 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵclassMapInterpolate1("form-control rmsh rmrd ", ctx_r44.classes == null ? null : ctx_r44.classes.searchbar, "");
    ɵngcc0.ɵɵstyleProp("background", ctx_r44.searchBar == null ? null : ctx_r44.searchBar.background, ɵngcc0.ɵɵdefaultStyleSanitizer)("color", ctx_r44.searchBar == null ? null : ctx_r44.searchBar.textColor);
    ɵngcc0.ɵɵproperty("ngStyle", ɵngcc0.ɵɵpureFunction1(10, _c0, (ctx_r44.searchBar == null ? null : ctx_r44.searchBar.borderSize) + " solid " + (ctx_r44.searchBar == null ? null : ctx_r44.searchBar.borderColor)))("placeholder", ctx_r44.searchPlaceholder);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("disabled", !(_r59 == null ? null : _r59.valid));
} }
function FlxUiDatatableComponent_div_24_Template(rf, ctx) { if (rf & 1) {
    const _r63 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 34);
    ɵngcc0.ɵɵelementStart(1, "button", 35);
    ɵngcc0.ɵɵlistener("click", function FlxUiDatatableComponent_div_24_Template_button_click_1_listener() { ɵngcc0.ɵɵrestoreView(_r63); const ctx_r62 = ɵngcc0.ɵɵnextContext(); return ctx_r62.addButtonClick(); });
    ɵngcc0.ɵɵelement(2, "span", 36);
    ɵngcc0.ɵɵtext(3, " Add");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r45 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵclassMap(ctx_r45.classes == null ? null : ctx_r45.classes.addButton);
    ɵngcc0.ɵɵstyleProp("background", ctx_r45.addButton == null ? null : ctx_r45.addButton.background, ɵngcc0.ɵɵdefaultStyleSanitizer)("border-color", ctx_r45.addButton == null ? null : ctx_r45.addButton.background)("color", ctx_r45.addButton == null ? null : ctx_r45.addButton.textColor);
} }
function FlxUiDatatableComponent_div_25_li_11_a_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "a", 51);
    ɵngcc0.ɵɵelement(1, "i", 52);
    ɵngcc0.ɵɵtext(2, " Print");
    ɵngcc0.ɵɵelementEnd();
} }
function FlxUiDatatableComponent_div_25_li_11_a_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "a", 53);
    ɵngcc0.ɵɵelement(1, "i", 54);
    ɵngcc0.ɵɵtext(2, " PDF");
    ɵngcc0.ɵɵelementEnd();
} }
function FlxUiDatatableComponent_div_25_li_11_a_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "a", 55);
    ɵngcc0.ɵɵelement(1, "i", 54);
    ɵngcc0.ɵɵtext(2, " Excel");
    ɵngcc0.ɵɵelementEnd();
} }
function FlxUiDatatableComponent_div_25_li_11_a_4_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "a", 56);
    ɵngcc0.ɵɵelement(1, "i", 54);
    ɵngcc0.ɵɵtext(2, " Word");
    ɵngcc0.ɵɵelementEnd();
} }
function FlxUiDatatableComponent_div_25_li_11_Template(rf, ctx) { if (rf & 1) {
    const _r73 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "li", 46);
    ɵngcc0.ɵɵlistener("click", function FlxUiDatatableComponent_div_25_li_11_Template_li_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r73); const export_r67 = ctx.$implicit; const ctx_r72 = ɵngcc0.ɵɵnextContext(2); return ctx_r72.exportDocumentsAs(export_r67); });
    ɵngcc0.ɵɵtemplate(1, FlxUiDatatableComponent_div_25_li_11_a_1_Template, 3, 0, "a", 47);
    ɵngcc0.ɵɵtemplate(2, FlxUiDatatableComponent_div_25_li_11_a_2_Template, 3, 0, "a", 48);
    ɵngcc0.ɵɵtemplate(3, FlxUiDatatableComponent_div_25_li_11_a_3_Template, 3, 0, "a", 49);
    ɵngcc0.ɵɵtemplate(4, FlxUiDatatableComponent_div_25_li_11_a_4_Template, 3, 0, "a", 50);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const export_r67 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", export_r67 == "print");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", export_r67 == "pdf");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", export_r67 == "excel");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", export_r67 == "word");
} }
function FlxUiDatatableComponent_div_25_span_14_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "async");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r65 = ɵngcc0.ɵɵnextContext(2);
    var tmp_0_0 = null;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate((tmp_0_0 = ɵngcc0.ɵɵpipeBind1(2, 1, ctx_r65.displayData)) == null ? null : tmp_0_0.length);
} }
function FlxUiDatatableComponent_div_25_span_15_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "async");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r66 = ɵngcc0.ɵɵnextContext(2);
    var tmp_0_0 = null;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate((tmp_0_0 = ɵngcc0.ɵɵpipeBind1(2, 1, ctx_r66.service == null ? null : ctx_r66.service.flxData)) == null ? null : tmp_0_0.length);
} }
function FlxUiDatatableComponent_div_25_Template(rf, ctx) { if (rf & 1) {
    const _r75 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 37);
    ɵngcc0.ɵɵelementStart(1, "div", 38);
    ɵngcc0.ɵɵelementStart(2, "button", 39);
    ɵngcc0.ɵɵelement(3, "i", 40);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(4, "ul");
    ɵngcc0.ɵɵelementStart(5, "li", 41);
    ɵngcc0.ɵɵtext(6);
    ɵngcc0.ɵɵelementStart(7, "input", 42);
    ɵngcc0.ɵɵlistener("change", function FlxUiDatatableComponent_div_25_Template_input_change_7_listener($event) { ɵngcc0.ɵɵrestoreView(_r75); const ctx_r74 = ɵngcc0.ɵɵnextContext(); return ctx_r74.checkToExportOption($event == null ? null : $event.target == null ? null : $event.target.checked); });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(8, "sup", 43);
    ɵngcc0.ɵɵtext(9, " All");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelement(10, "li", 44);
    ɵngcc0.ɵɵtemplate(11, FlxUiDatatableComponent_div_25_li_11_Template, 5, 4, "li", 45);
    ɵngcc0.ɵɵelement(12, "li", 44);
    ɵngcc0.ɵɵelementStart(13, "li", 41);
    ɵngcc0.ɵɵtemplate(14, FlxUiDatatableComponent_div_25_span_14_Template, 3, 3, "span", 20);
    ɵngcc0.ɵɵtemplate(15, FlxUiDatatableComponent_div_25_span_15_Template, 3, 3, "span", 20);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r46 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵclassMapInterpolate1("btn btn-default ", ctx_r46.classes == null ? null : ctx_r46.classes.exportButton, " dropdown-toggle");
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵclassMapInterpolate1("dropdown-menu dropdown-menu-export ", ctx_r46.classes == null ? null : ctx_r46.classes.exportDropdown, "");
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate1("", ctx_r46.dataExportsConfig == null ? null : ctx_r46.dataExportsConfig.title, ". ");
    ɵngcc0.ɵɵadvance(5);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r46.dataExportsConfig == null ? null : ctx_r46.dataExportsConfig.exportsTo);
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵproperty("ngIf", !ctx_r46.isExportAll);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r46.isExportAll);
} }
function FlxUiDatatableComponent_th_32_input_2_Template(rf, ctx) { if (rf & 1) {
    const _r79 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "input", 59);
    ɵngcc0.ɵɵlistener("change", function FlxUiDatatableComponent_th_32_input_2_Template_input_change_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r79); const ctx_r78 = ɵngcc0.ɵɵnextContext(2); return ctx_r78.addRemove($event == null ? null : $event.target == null ? null : $event.target.checked); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r76 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("checked", (ctx_r76.service == null ? null : ctx_r76.service.multipleDeletion == null ? null : ctx_r76.service.multipleDeletion.length) > 0);
} }
function FlxUiDatatableComponent_th_32_button_3_Template(rf, ctx) { if (rf & 1) {
    const _r81 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "button", 60);
    ɵngcc0.ɵɵlistener("click", function FlxUiDatatableComponent_th_32_button_3_Template_button_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r81); const ctx_r80 = ɵngcc0.ɵɵnextContext(2); return ctx_r80.confirmDelete(); });
    ɵngcc0.ɵɵelement(1, "span");
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r77 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵclassMap(ctx_r77.multipleSelectButton == null ? null : ctx_r77.multipleSelectButton.icon);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r77.multipleSelectButton == null ? null : ctx_r77.multipleSelectButton.text, "");
} }
function FlxUiDatatableComponent_th_32_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "th");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵtemplate(2, FlxUiDatatableComponent_th_32_input_2_Template, 1, 1, "input", 57);
    ɵngcc0.ɵɵtemplate(3, FlxUiDatatableComponent_th_32_button_3_Template, 3, 3, "button", 58);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r47 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1("", ctx_r47.actionHeader, " ");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r47.enableMultipleSelection);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r47.enableMultipleSelection && (ctx_r47.service == null ? null : ctx_r47.service.multipleDeletion == null ? null : ctx_r47.service.multipleDeletion.length) > 0);
} }
function FlxUiDatatableComponent_th_33_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "th");
    ɵngcc0.ɵɵtext(1, "N");
    ɵngcc0.ɵɵelementStart(2, "sup");
    ɵngcc0.ɵɵelementStart(3, "u");
    ɵngcc0.ɵɵtext(4, "o");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} }
function FlxUiDatatableComponent_th_34_input_2_Template(rf, ctx) { if (rf & 1) {
    const _r85 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "input", 59);
    ɵngcc0.ɵɵlistener("change", function FlxUiDatatableComponent_th_34_input_2_Template_input_change_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r85); const ctx_r84 = ɵngcc0.ɵɵnextContext(2); return ctx_r84.addRemove($event == null ? null : $event.target == null ? null : $event.target.checked); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r82 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("checked", (ctx_r82.service == null ? null : ctx_r82.service.multipleDeletion == null ? null : ctx_r82.service.multipleDeletion.length) > 0);
} }
function FlxUiDatatableComponent_th_34_button_3_Template(rf, ctx) { if (rf & 1) {
    const _r87 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "button", 60);
    ɵngcc0.ɵɵlistener("click", function FlxUiDatatableComponent_th_34_button_3_Template_button_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r87); const ctx_r86 = ɵngcc0.ɵɵnextContext(2); return ctx_r86.confirmDelete(); });
    ɵngcc0.ɵɵelement(1, "span");
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r83 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵclassMap(ctx_r83.multipleSelectButton == null ? null : ctx_r83.multipleSelectButton.icon);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r83.multipleSelectButton == null ? null : ctx_r83.multipleSelectButton.text, "");
} }
function FlxUiDatatableComponent_th_34_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "th");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵtemplate(2, FlxUiDatatableComponent_th_34_input_2_Template, 1, 1, "input", 57);
    ɵngcc0.ɵɵtemplate(3, FlxUiDatatableComponent_th_34_button_3_Template, 3, 3, "button", 58);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r49 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1("", ctx_r49.actionHeader, " ");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r49.enableMultipleSelection);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r49.enableMultipleSelection && (ctx_r49.service == null ? null : ctx_r49.service.multipleDeletion == null ? null : ctx_r49.service.multipleDeletion.length) > 0);
} }
function FlxUiDatatableComponent_th_35_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "th");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const header_r88 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(header_r88);
} }
function FlxUiDatatableComponent_th_36_input_2_Template(rf, ctx) { if (rf & 1) {
    const _r92 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "input", 59);
    ɵngcc0.ɵɵlistener("change", function FlxUiDatatableComponent_th_36_input_2_Template_input_change_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r92); const ctx_r91 = ɵngcc0.ɵɵnextContext(2); return ctx_r91.addRemove($event == null ? null : $event.target == null ? null : $event.target.checked); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r89 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("checked", (ctx_r89.service == null ? null : ctx_r89.service.multipleDeletion == null ? null : ctx_r89.service.multipleDeletion.length) > 0);
} }
function FlxUiDatatableComponent_th_36_button_3_Template(rf, ctx) { if (rf & 1) {
    const _r94 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "button", 60);
    ɵngcc0.ɵɵlistener("click", function FlxUiDatatableComponent_th_36_button_3_Template_button_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r94); const ctx_r93 = ɵngcc0.ɵɵnextContext(2); return ctx_r93.confirmDelete(); });
    ɵngcc0.ɵɵelement(1, "span");
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r90 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵclassMap(ctx_r90.multipleSelectButton == null ? null : ctx_r90.multipleSelectButton.icon);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r90.multipleSelectButton == null ? null : ctx_r90.multipleSelectButton.text, "");
} }
function FlxUiDatatableComponent_th_36_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "th");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵtemplate(2, FlxUiDatatableComponent_th_36_input_2_Template, 1, 1, "input", 57);
    ɵngcc0.ɵɵtemplate(3, FlxUiDatatableComponent_th_36_button_3_Template, 3, 3, "button", 58);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r51 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1("", ctx_r51.actionHeader, " ");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r51.enableMultipleSelection);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r51.enableMultipleSelection && (ctx_r51.service == null ? null : ctx_r51.service.multipleDeletion == null ? null : ctx_r51.service.multipleDeletion.length) > 0);
} }
function FlxUiDatatableComponent_tr_40_img_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "img", 63);
} if (rf & 2) {
    const ctx_r95 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵclassMap(ctx_r95.classes == null ? null : ctx_r95.classes.spinner);
    ɵngcc0.ɵɵproperty("src", ctx_r95.spinnerSrc, ɵngcc0.ɵɵsanitizeUrl);
} }
function FlxUiDatatableComponent_tr_40_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "tr");
    ɵngcc0.ɵɵelementStart(1, "td", 61);
    ɵngcc0.ɵɵtemplate(2, FlxUiDatatableComponent_tr_40_img_2_Template, 1, 3, "img", 62);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r52 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r52.spinnerSrc);
} }
const _c1 = function (a0, a1, a2) { return { "flx-tooltip-left": a0, "flx-tooltip-bottom": a1, "flx-tooltip-right": a2 }; };
function FlxUiDatatableComponent_tr_41_td_1_span_1_div_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 76);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const aButton_r105 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction3(2, _c1, (aButton_r105 == null ? null : aButton_r105.tooltipPosition) == "left", (aButton_r105 == null ? null : aButton_r105.tooltipPosition) == "bottom", (aButton_r105 == null ? null : aButton_r105.tooltipPosition) == "right"));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(aButton_r105 == null ? null : aButton_r105.tooltip);
} }
function FlxUiDatatableComponent_tr_41_td_1_span_1_span_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "span", 77);
} if (rf & 2) {
    const aButton_r105 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵclassMap(aButton_r105 == null ? null : aButton_r105.icon);
} }
function FlxUiDatatableComponent_tr_41_td_1_span_1_span_6_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "span");
} if (rf & 2) {
    const aButton_r105 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵclassMap(aButton_r105 == null ? null : aButton_r105.icon);
} }
function FlxUiDatatableComponent_tr_41_td_1_span_1_Template(rf, ctx) { if (rf & 1) {
    const _r115 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "span");
    ɵngcc0.ɵɵelementStart(1, "button", 71);
    ɵngcc0.ɵɵlistener("click", function FlxUiDatatableComponent_tr_41_td_1_span_1_Template_button_click_1_listener() { ɵngcc0.ɵɵrestoreView(_r115); const buttonIndex_r106 = ctx.index; const i_r97 = ɵngcc0.ɵɵnextContext(2).index; const ctx_r113 = ɵngcc0.ɵɵnextContext(); return ctx_r113.actionButtonClicked(i_r97, buttonIndex_r106); });
    ɵngcc0.ɵɵtemplate(2, FlxUiDatatableComponent_tr_41_td_1_span_1_div_2_Template, 2, 6, "div", 72);
    ɵngcc0.ɵɵtemplate(3, FlxUiDatatableComponent_tr_41_td_1_span_1_span_3_Template, 1, 2, "span", 73);
    ɵngcc0.ɵɵelementStart(4, "span", 74);
    ɵngcc0.ɵɵtext(5);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtemplate(6, FlxUiDatatableComponent_tr_41_td_1_span_1_span_6_Template, 1, 2, "span", 75);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const aButton_r105 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵclassMapInterpolate1("btn ", aButton_r105 == null ? null : aButton_r105.class, "");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", aButton_r105 == null ? null : aButton_r105.tooltip);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", !(aButton_r105 == null ? null : aButton_r105.iconPosition) || (aButton_r105 == null ? null : aButton_r105.iconPosition) != "right");
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate1(" ", aButton_r105 == null ? null : aButton_r105.text, " ");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", (aButton_r105 == null ? null : aButton_r105.iconPosition) == "right");
} }
function FlxUiDatatableComponent_tr_41_td_1_input_2_Template(rf, ctx) { if (rf & 1) {
    const _r118 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "input", 78);
    ɵngcc0.ɵɵlistener("change", function FlxUiDatatableComponent_tr_41_td_1_input_2_Template_input_change_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r118); const ctx_r117 = ɵngcc0.ɵɵnextContext(2); const data_r96 = ctx_r117.$implicit; const i_r97 = ctx_r117.index; const ctx_r116 = ɵngcc0.ɵɵnextContext(); return ctx_r116.addRemoveDeleteItem(data_r96[ctx_r116.multipleSelectKey], i_r97, $event == null ? null : $event.target == null ? null : $event.target.checked); });
    ɵngcc0.ɵɵelementEnd();
} }
function FlxUiDatatableComponent_tr_41_td_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "td", 69);
    ɵngcc0.ɵɵtemplate(1, FlxUiDatatableComponent_tr_41_td_1_span_1_Template, 7, 7, "span", 21);
    ɵngcc0.ɵɵtemplate(2, FlxUiDatatableComponent_tr_41_td_1_input_2_Template, 1, 0, "input", 70);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r98 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r98.actionButtons);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r98.enableMultipleSelection && (ctx_r98.service == null ? null : ctx_r98.service.multipleDeletion == null ? null : ctx_r98.service.multipleDeletion.length) > 0);
} }
function FlxUiDatatableComponent_tr_41_td_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "td", 79);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const i_r97 = ɵngcc0.ɵɵnextContext().index;
    const ctx_r99 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵclassMap(ctx_r99.classes == null ? null : ctx_r99.classes.tableData);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ctx_r99.offset + i_r97);
} }
function FlxUiDatatableComponent_tr_41_td_3_span_1_div_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 76);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const aButton_r122 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction3(2, _c1, (aButton_r122 == null ? null : aButton_r122.tooltipPosition) == "left", (aButton_r122 == null ? null : aButton_r122.tooltipPosition) == "bottom", (aButton_r122 == null ? null : aButton_r122.tooltipPosition) == "right"));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(aButton_r122 == null ? null : aButton_r122.tooltip);
} }
function FlxUiDatatableComponent_tr_41_td_3_span_1_span_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "span", 77);
} if (rf & 2) {
    const aButton_r122 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵclassMap(aButton_r122 == null ? null : aButton_r122.icon);
} }
function FlxUiDatatableComponent_tr_41_td_3_span_1_span_6_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "span");
} if (rf & 2) {
    const aButton_r122 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵclassMap(aButton_r122 == null ? null : aButton_r122.icon);
} }
function FlxUiDatatableComponent_tr_41_td_3_span_1_Template(rf, ctx) { if (rf & 1) {
    const _r132 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "span");
    ɵngcc0.ɵɵelementStart(1, "button", 71);
    ɵngcc0.ɵɵlistener("click", function FlxUiDatatableComponent_tr_41_td_3_span_1_Template_button_click_1_listener() { ɵngcc0.ɵɵrestoreView(_r132); const buttonIndex_r123 = ctx.index; const i_r97 = ɵngcc0.ɵɵnextContext(2).index; const ctx_r130 = ɵngcc0.ɵɵnextContext(); return ctx_r130.actionButtonClicked(i_r97, buttonIndex_r123); });
    ɵngcc0.ɵɵtemplate(2, FlxUiDatatableComponent_tr_41_td_3_span_1_div_2_Template, 2, 6, "div", 72);
    ɵngcc0.ɵɵtemplate(3, FlxUiDatatableComponent_tr_41_td_3_span_1_span_3_Template, 1, 2, "span", 73);
    ɵngcc0.ɵɵelementStart(4, "span", 74);
    ɵngcc0.ɵɵtext(5);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtemplate(6, FlxUiDatatableComponent_tr_41_td_3_span_1_span_6_Template, 1, 2, "span", 75);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const aButton_r122 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵclassMapInterpolate1("btn ", aButton_r122 == null ? null : aButton_r122.class, "");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", aButton_r122 == null ? null : aButton_r122.tooltip);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", !(aButton_r122 == null ? null : aButton_r122.iconPosition) || (aButton_r122 == null ? null : aButton_r122.iconPosition) != "right");
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate1(" ", aButton_r122 == null ? null : aButton_r122.text, " ");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", (aButton_r122 == null ? null : aButton_r122.iconPosition) == "right");
} }
function FlxUiDatatableComponent_tr_41_td_3_input_2_Template(rf, ctx) { if (rf & 1) {
    const _r135 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "input", 78);
    ɵngcc0.ɵɵlistener("change", function FlxUiDatatableComponent_tr_41_td_3_input_2_Template_input_change_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r135); const ctx_r134 = ɵngcc0.ɵɵnextContext(2); const data_r96 = ctx_r134.$implicit; const i_r97 = ctx_r134.index; const ctx_r133 = ɵngcc0.ɵɵnextContext(); return ctx_r133.addRemoveDeleteItem(data_r96[ctx_r133.multipleSelectKey], i_r97, $event == null ? null : $event.target == null ? null : $event.target.checked); });
    ɵngcc0.ɵɵelementEnd();
} }
function FlxUiDatatableComponent_tr_41_td_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "td", 69);
    ɵngcc0.ɵɵtemplate(1, FlxUiDatatableComponent_tr_41_td_3_span_1_Template, 7, 7, "span", 21);
    ɵngcc0.ɵɵtemplate(2, FlxUiDatatableComponent_tr_41_td_3_input_2_Template, 1, 0, "input", 70);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r100 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r100.actionButtons);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r100.enableMultipleSelection && (ctx_r100.service == null ? null : ctx_r100.service.multipleDeletion == null ? null : ctx_r100.service.multipleDeletion.length) > 0);
} }
function FlxUiDatatableComponent_tr_41_td_4_img_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "img", 82);
} if (rf & 2) {
    const dataKey_r136 = ɵngcc0.ɵɵnextContext().$implicit;
    const data_r96 = ɵngcc0.ɵɵnextContext().$implicit;
    const ctx_r138 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵclassMap("img-fall-back " + (ctx_r138.embedPictures == null ? null : ctx_r138.embedPictures.class));
    ɵngcc0.ɵɵproperty("src", (ctx_r138.embedPictures == null ? null : ctx_r138.embedPictures.rootFolder) + data_r96[dataKey_r136], ɵngcc0.ɵɵsanitizeUrl)("flx-ui-datatable-img-fallback", ctx_r138.embedPictures == null ? null : ctx_r138.embedPictures.fallbackUrl);
} }
function FlxUiDatatableComponent_tr_41_td_4_ng_template_2_section_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "section");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "limittext");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const dataKey_r136 = ɵngcc0.ɵɵnextContext(2).$implicit;
    const data_r96 = ɵngcc0.ɵɵnextContext().$implicit;
    const ctx_r143 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind2(2, 1, ctx_r143.textSubstring, data_r96[dataKey_r136]), " ");
} }
function FlxUiDatatableComponent_tr_41_td_4_ng_template_2_ng_template_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const dataKey_r136 = ɵngcc0.ɵɵnextContext(2).$implicit;
    const data_r96 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(data_r96[dataKey_r136]);
} }
function FlxUiDatatableComponent_tr_41_td_4_ng_template_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵtemplate(0, FlxUiDatatableComponent_tr_41_td_4_ng_template_2_section_0_Template, 3, 4, "section", 11);
    ɵngcc0.ɵɵpipe(1, "haslimittext");
    ɵngcc0.ɵɵtemplate(2, FlxUiDatatableComponent_tr_41_td_4_ng_template_2_ng_template_2_Template, 2, 1, "ng-template", null, 83, ɵngcc0.ɵɵtemplateRefExtractor);
} if (rf & 2) {
    const _r144 = ɵngcc0.ɵɵreference(3);
    const x_r137 = ɵngcc0.ɵɵnextContext().index;
    const ctx_r140 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("ngIf", (ctx_r140.textSubstring == null ? null : ctx_r140.textSubstring.atIndexes) && ɵngcc0.ɵɵpipeBind2(1, 2, ctx_r140.textSubstring == null ? null : ctx_r140.textSubstring.atIndexes, x_r137))("ngIfElse", _r144);
} }
function FlxUiDatatableComponent_tr_41_td_4_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "td");
    ɵngcc0.ɵɵtemplate(1, FlxUiDatatableComponent_tr_41_td_4_img_1_Template, 1, 4, "img", 80);
    ɵngcc0.ɵɵtemplate(2, FlxUiDatatableComponent_tr_41_td_4_ng_template_2_Template, 4, 5, "ng-template", null, 81, ɵngcc0.ɵɵtemplateRefExtractor);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const x_r137 = ctx.index;
    const _r139 = ɵngcc0.ɵɵreference(3);
    const ctx_r101 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵclassMap(ctx_r101.classes == null ? null : ctx_r101.classes.tableData);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r101.hasImageEmbeded() && x_r137 == (ctx_r101.embedPictures == null ? null : ctx_r101.embedPictures.index))("ngIfElse", _r139);
} }
function FlxUiDatatableComponent_tr_41_td_5_span_1_div_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 76);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const aButton_r153 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction3(2, _c1, (aButton_r153 == null ? null : aButton_r153.tooltipPosition) == "left", (aButton_r153 == null ? null : aButton_r153.tooltipPosition) == "bottom", (aButton_r153 == null ? null : aButton_r153.tooltipPosition) == "right"));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(aButton_r153 == null ? null : aButton_r153.tooltip);
} }
function FlxUiDatatableComponent_tr_41_td_5_span_1_span_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "span", 77);
} if (rf & 2) {
    const aButton_r153 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵclassMap(aButton_r153 == null ? null : aButton_r153.icon);
} }
function FlxUiDatatableComponent_tr_41_td_5_span_1_span_6_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "span");
} if (rf & 2) {
    const aButton_r153 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵclassMap(aButton_r153 == null ? null : aButton_r153.icon);
} }
function FlxUiDatatableComponent_tr_41_td_5_span_1_Template(rf, ctx) { if (rf & 1) {
    const _r163 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "span");
    ɵngcc0.ɵɵelementStart(1, "button", 71);
    ɵngcc0.ɵɵlistener("click", function FlxUiDatatableComponent_tr_41_td_5_span_1_Template_button_click_1_listener() { ɵngcc0.ɵɵrestoreView(_r163); const buttonIndex_r154 = ctx.index; const i_r97 = ɵngcc0.ɵɵnextContext(2).index; const ctx_r161 = ɵngcc0.ɵɵnextContext(); return ctx_r161.actionButtonClicked(i_r97, buttonIndex_r154); });
    ɵngcc0.ɵɵtemplate(2, FlxUiDatatableComponent_tr_41_td_5_span_1_div_2_Template, 2, 6, "div", 72);
    ɵngcc0.ɵɵtemplate(3, FlxUiDatatableComponent_tr_41_td_5_span_1_span_3_Template, 1, 2, "span", 73);
    ɵngcc0.ɵɵelementStart(4, "span", 74);
    ɵngcc0.ɵɵtext(5);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtemplate(6, FlxUiDatatableComponent_tr_41_td_5_span_1_span_6_Template, 1, 2, "span", 75);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const aButton_r153 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵclassMapInterpolate1("btn ", aButton_r153 == null ? null : aButton_r153.class, "");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", aButton_r153 == null ? null : aButton_r153.tooltip);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", !(aButton_r153 == null ? null : aButton_r153.iconPosition) || (aButton_r153 == null ? null : aButton_r153.iconPosition) != "right");
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate1(" ", aButton_r153 == null ? null : aButton_r153.text, " ");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", (aButton_r153 == null ? null : aButton_r153.iconPosition) == "right");
} }
function FlxUiDatatableComponent_tr_41_td_5_input_2_Template(rf, ctx) { if (rf & 1) {
    const _r166 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "input", 78);
    ɵngcc0.ɵɵlistener("change", function FlxUiDatatableComponent_tr_41_td_5_input_2_Template_input_change_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r166); const ctx_r165 = ɵngcc0.ɵɵnextContext(2); const data_r96 = ctx_r165.$implicit; const i_r97 = ctx_r165.index; const ctx_r164 = ɵngcc0.ɵɵnextContext(); return ctx_r164.addRemoveDeleteItem(data_r96[ctx_r164.multipleSelectKey], i_r97, $event == null ? null : $event.target == null ? null : $event.target.checked); });
    ɵngcc0.ɵɵelementEnd();
} }
function FlxUiDatatableComponent_tr_41_td_5_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "td", 84);
    ɵngcc0.ɵɵtemplate(1, FlxUiDatatableComponent_tr_41_td_5_span_1_Template, 7, 7, "span", 21);
    ɵngcc0.ɵɵtemplate(2, FlxUiDatatableComponent_tr_41_td_5_input_2_Template, 1, 0, "input", 70);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r102 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r102.actionButtons);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r102.enableMultipleSelection && (ctx_r102.service == null ? null : ctx_r102.service.multipleDeletion == null ? null : ctx_r102.service.multipleDeletion.length) > 0);
} }
function FlxUiDatatableComponent_tr_41_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "tr", 64);
    ɵngcc0.ɵɵtemplate(1, FlxUiDatatableComponent_tr_41_td_1_Template, 3, 2, "td", 65);
    ɵngcc0.ɵɵtemplate(2, FlxUiDatatableComponent_tr_41_td_2_Template, 2, 4, "td", 66);
    ɵngcc0.ɵɵtemplate(3, FlxUiDatatableComponent_tr_41_td_3_Template, 3, 2, "td", 65);
    ɵngcc0.ɵɵtemplate(4, FlxUiDatatableComponent_tr_41_td_4_Template, 4, 5, "td", 67);
    ɵngcc0.ɵɵtemplate(5, FlxUiDatatableComponent_tr_41_td_5_Template, 3, 2, "td", 68);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r53 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r53.hasActionButtons && ((ctx_r53.actionButtonsConfig == null ? null : ctx_r53.actionButtonsConfig.position) == "left" && (ctx_r53.actionButtonsConfig == null ? null : ctx_r53.actionButtonsConfig.beforeNumbers)));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", !ctx_r53.hideNumbers);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r53.hasActionButtons && ((ctx_r53.actionButtonsConfig == null ? null : ctx_r53.actionButtonsConfig.position) == "left" && !(ctx_r53.actionButtonsConfig == null ? null : ctx_r53.actionButtonsConfig.beforeNumbers)));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r53.dataKeys);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r53.hasActionButtons && (ctx_r53.actionButtonsConfig == null ? null : ctx_r53.actionButtonsConfig.position) != "left");
} }
function FlxUiDatatableComponent_tr_43_td_1_Template(rf, ctx) { if (rf & 1) {
    const _r169 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "td", 86);
    ɵngcc0.ɵɵelementStart(1, "span", 87);
    ɵngcc0.ɵɵtext(2, "No data found");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelement(3, "br");
    ɵngcc0.ɵɵelementStart(4, "button", 88);
    ɵngcc0.ɵɵlistener("click", function FlxUiDatatableComponent_tr_43_td_1_Template_button_click_4_listener() { ɵngcc0.ɵɵrestoreView(_r169); const ctx_r168 = ɵngcc0.ɵɵnextContext(2); return ctx_r168.reload(); });
    ɵngcc0.ɵɵelement(5, "span", 89);
    ɵngcc0.ɵɵtext(6, " Reload");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r167 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(4);
    ɵngcc0.ɵɵclassMapInterpolate1("btn btn-default ", ctx_r167.classes == null ? null : ctx_r167.classes.reloadbutton, "");
} }
function FlxUiDatatableComponent_tr_43_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "tr");
    ɵngcc0.ɵɵtemplate(1, FlxUiDatatableComponent_tr_43_td_1_Template, 7, 3, "td", 85);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r54 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r54.service == null ? null : ctx_r54.service.loadFinish);
} }
function FlxUiDatatableComponent_div_45_div_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 3);
    ɵngcc0.ɵɵelementStart(1, "div", 93);
    ɵngcc0.ɵɵtext(2, "Total pagination: ");
    ɵngcc0.ɵɵelementStart(3, "span");
    ɵngcc0.ɵɵtext(4);
    ɵngcc0.ɵɵpipe(5, "ceil");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(6, "div", 94);
    ɵngcc0.ɵɵtext(7, "# of items per pagination: ");
    ɵngcc0.ɵɵelementStart(8, "span");
    ɵngcc0.ɵɵtext(9);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(10, "div", 95);
    ɵngcc0.ɵɵtext(11, "Total items: ");
    ɵngcc0.ɵɵelementStart(12, "span");
    ɵngcc0.ɵɵtext(13);
    ɵngcc0.ɵɵpipe(14, "async");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r170 = ɵngcc0.ɵɵnextContext(2);
    var tmp_2_0 = null;
    ɵngcc0.ɵɵadvance(4);
    ɵngcc0.ɵɵtextInterpolate1(" ", ɵngcc0.ɵɵpipeBind2(5, 3, ctx_r170.service == null ? null : ctx_r170.service.totalItems, ctx_r170.limit), "");
    ɵngcc0.ɵɵadvance(5);
    ɵngcc0.ɵɵtextInterpolate(ctx_r170.limit);
    ɵngcc0.ɵɵadvance(4);
    ɵngcc0.ɵɵtextInterpolate((tmp_2_0 = ɵngcc0.ɵɵpipeBind1(14, 6, ctx_r170.service == null ? null : ctx_r170.service.flxData)) == null ? null : tmp_2_0.length);
} }
function FlxUiDatatableComponent_div_45_ng_template_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 3);
    ɵngcc0.ɵɵelementStart(1, "div", 96);
    ɵngcc0.ɵɵtext(2, "Total search result: ");
    ɵngcc0.ɵɵelementStart(3, "span");
    ɵngcc0.ɵɵtext(4);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r172 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(4);
    ɵngcc0.ɵɵtextInterpolate(ctx_r172.searchResultFound == null ? null : ctx_r172.searchResultFound.length);
} }
function FlxUiDatatableComponent_div_45_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 90);
    ɵngcc0.ɵɵtemplate(1, FlxUiDatatableComponent_div_45_div_1_Template, 15, 8, "div", 91);
    ɵngcc0.ɵɵtemplate(2, FlxUiDatatableComponent_div_45_ng_template_2_Template, 5, 1, "ng-template", null, 92, ɵngcc0.ɵɵtemplateRefExtractor);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const _r171 = ɵngcc0.ɵɵreference(3);
    const ctx_r55 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", !ctx_r55.isSearching)("ngIfElse", _r171);
} }
const _c2 = function (a0) { return { "flx-pagination-end": a0 }; };
class FlxUiDatatableService {
    /**
     * @param {?} http
     */
    constructor(http) {
        this.http = http;
        this.dataUrl = '';
        this.behavior = new BehaviorSubject([]);
        this.flxData = this.behavior.asObservable();
        this.pagination = [];
        this.totalItems = 0;
        this.dataOffset = 0;
        this.limit = 20;
        this.dataSrcKey = '';
        //Hold items selected for multiple select
        this.multipleDeletion = [];
        //Keep track if API call is completed
        this.loadFinish = false;
        this.lazyloadingConfig = {};
        this.processedData = {};
    }
    /**
     * @param {?} config
     * @return {?}
     */
    setLazyloadingConfig(config) {
        this.lazyloadingConfig = config;
    }
    /**
     *
     * @param {?} url User api rul
     * @return {?}
     */
    getData(url) {
        let /** @type {?} */ headers = new Headers();
        headers.append('Content-Type', 'application/x-www-form-urlencoded');
        return this.http.get(url, { headers: headers }).pipe(retry(5), map((response) => response.json()));
    }
    /**
     *
     * @param {?} url Service api url
     * @param {?} id Datatype id to export
     * @param {?} data Data to export
     * @return {?}
     */
    postData(url, id, data) {
        let /** @type {?} */ headers = new Headers();
        headers.append('Content-Type', 'application/json; charset=utf-8');
        return this.http.post(url + id, data, { headers: headers }).pipe(map((resp) => resp.json()));
    }
    /**
     *
     * @param {?} dataUrl Set dataurl
     * @return {?}
     */
    setDataUrl(dataUrl) {
        this.dataUrl = dataUrl;
    }
    /**
     * @return {?}
     */
    getDataUrl() {
        return this.dataUrl;
    }
    /**
     *
     * @param {?} data Register new data from user API
     * @return {?}
     */
    chageDataTable(data) {
        this.behavior.next(data);
    }
    /**
     *
     * @param {?} numberOfList Total number of list
     * @param {?} limit Pagination limit
     * @return {?}
     */
    createPagination(numberOfList, limit) {
        let /** @type {?} */ obj = [];
        let /** @type {?} */ counter = 1;
        for (let /** @type {?} */ i = 0; i < numberOfList; i += limit) {
            obj.push({ label: counter, value: i });
            counter++;
        }
        if (!this.isLazyLoadingEnabled) {
            obj.push({ label: 'All', value: 'all' });
        }
        return obj;
    }
    /**
     * @return {?}
     */
    isLazyLoadingEnabled() {
        return this.lazyloadingConfig.hasOwnProperty("apiOffsetKey") && this.lazyloadingConfig.apiOffsetKey;
    }
    /**
     * @param {?} dataUrl
     * @param {?=} setSelectPagination
     * @return {?}
     */
    loadFlxDataTableData(dataUrl, setSelectPagination = true) {
        this.loadFinish = false;
        this.loader = this.getData(dataUrl).subscribe((responseData) => {
            try {
                this.multipleDeletion = [];
                var /** @type {?} */ data = (this.dataSrcKey) ? responseData[this.dataSrcKey] : responseData;
                this.chageDataTable(data);
                if (this.isLazyLoadingEnabled()) {
                    this.totalItems = responseData.total;
                    // Handle 1 pagination out of zero problem 1/0  instead of 0/0 if no data is comming
                    if (data.length > 0) {
                        this.dataOffset = this.dataOffset + this.limit;
                    }
                }
                else {
                    this.totalItems = data.length;
                    this.dataOffset = 1;
                }
                if (setSelectPagination) {
                    if (this.isLazyLoadingEnabled()) {
                        this.pagination = this.createPagination(responseData.total, this.limit);
                    }
                    else {
                        this.pagination = this.createPagination(data.length, this.limit);
                    }
                }
                this.loadFinish = true;
            }
            catch (/** @type {?} */ e) {
                console.log('Error in reading data in ', e);
            }
        }, (e => {
            this.loadFinish = true;
        }));
    }
    /**
     * @return {?}
     */
    cancelLoading() {
        this.loader.unsubscribe();
    }
    /**
     * @param {?} srcKey
     * @return {?}
     */
    setDataSrcKey(srcKey) {
        this.dataSrcKey = srcKey;
    }
    /**
     * @return {?}
     */
    getDataLength() {
        return new Promise((resolve) => {
            this.flxData.subscribe((resp) => {
                resolve(resp.length);
            }, (e => {
                resolve(0);
            }));
        });
    }
}
FlxUiDatatableService.ɵfac = function FlxUiDatatableService_Factory(t) { return new (t || FlxUiDatatableService)(ɵngcc0.ɵɵinject(ɵngcc1.Http)); };
FlxUiDatatableService.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: FlxUiDatatableService, factory: FlxUiDatatableService.ɵfac });
/** @nocollapse */
FlxUiDatatableService.ctorParameters = () => [
    { type: Http }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FlxUiDatatableService, [{
        type: Injectable
    }], function () { return [{ type: ɵngcc1.Http }]; }, null); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class FlxUiDataTable {
    /**
     * @param {?} service
     */
    constructor(service) {
        this.service = service;
        this.behavior = new BehaviorSubject([]);
        this.flxDatatableData = this.behavior.asObservable();
        this.service.flxData.subscribe((resp) => {
            this.ChangeData(resp);
        });
    }
    /**
     *
     * @param {?} data Change table data with new data
     * @return {?}
     */
    ChangeData(data) {
        this.behavior.next(data);
    }
    /**
     * Reload data from api: void
     * @return {?}
     */
    reloadData() {
        this.service.loadFlxDataTableData(this.service.getDataUrl());
    }
    /**
     * @return {?}
     */
    abortRequest() {
        this.service.cancelLoading();
    }
}
FlxUiDataTable.ɵfac = function FlxUiDataTable_Factory(t) { return new (t || FlxUiDataTable)(ɵngcc0.ɵɵinject(FlxUiDatatableService)); };
FlxUiDataTable.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: FlxUiDataTable, factory: FlxUiDataTable.ɵfac });
/** @nocollapse */
FlxUiDataTable.ctorParameters = () => [
    { type: FlxUiDatatableService }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FlxUiDataTable, [{
        type: Injectable
    }], function () { return [{ type: FlxUiDatatableService }]; }, null); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class FlxUiDatatableComponent {
    /**
     * @param {?} __form
     * @param {?} service
     */
    constructor(__form, service) {
        this.__form = __form;
        this.service = service;
        this.classes = {};
        this.headers = [];
        this.lazyloadingConfig = {};
        this.embedPictures = {};
        this.dataKeys = [];
        this.enableDataExports = false;
        this.dataExportsConfig = {};
        this.searchKeys = [];
        this.dataSrcKey = '';
        this.hasActionButtons = false;
        this.hideNumbers = false;
        this.enableMultipleSelection = false;
        this.multipleSelectKey = '';
        this.hasAddButton = false;
        this.dataUrl = '';
        this.actionButtonStart = false;
        this.multipleSelectButton = { text: 'Selected', icon: '' };
        this.searchPlaceholder = 'Enter name to search';
        this.actionHeader = 'Actions';
        this.limit = 20;
        this.spinnerSrc = '';
        this.actionButtons = [];
        this.paginationButtons = { background: '#dddddd', textColor: '#335599' };
        this.tableHeader = { background: '#ffffff', textColor: '#335599' };
        this.searchButton = { background: '#cccccc', textColor: '#335599' };
        this.addButton = {};
        this.searchBar = { borderSize: '1px', borderColor: '#ccc', background: '#ffffff', textColor: '#000000' };
        this.actionButtonsConfig = { position: 'right', beforeNumbers: true };
        this.firstActionButtonClicked = new EventEmitter();
        this.secondActionButtonClicked = new EventEmitter();
        this.thirdActionButtonClicked = new EventEmitter();
        this.multipleSelectClicked = new EventEmitter();
        this.addButtonClicked = new EventEmitter();
        /**
         * 18/12/2018
         */
        this.textSubstring = {};
        this.conditionManipulator = {};
        this.isExportAll = false;
        this.tData = [];
        this.behavior = new BehaviorSubject([]);
        this.searchDataTempOffset = [];
        this.displayData = this.behavior.asObservable();
        this.offset = 1;
        this.searchResultFound = [];
    }
    /**
     * @return {?}
     */
    reload() {
        this.service.loadFlxDataTableData(this.reloadUrl, true);
    }
    /**
     *
     * @param {?} checked Export all selection
     * @return {?}
     */
    checkToExportOption(checked) {
        this.isExportAll = checked;
    }
    /**
     *
     * @param {?} exportType Export type: print|pdf|excel|word
     * @return {?}
     */
    exportDocumentsAs(exportType) {
        let /** @type {?} */ loading = /** @type {?} */ (document.getElementById("export_loading"));
        loading.style.display = 'block';
        let /** @type {?} */ headers = (!this.dataExportsConfig.dataColumns || this.dataExportsConfig.dataColumns.length < 1) ? this.dataKeys : this.dataExportsConfig.dataColumns;
        let /** @type {?} */ dataToExport = (!this.isExportAll) ? this.displayData : this.service.flxData;
        //Subscribe to data
        dataToExport.subscribe((data) => {
            let /** @type {?} */ arrayObj = [];
            //Loop and push data
            for (let /** @type {?} */ d of data) {
                let /** @type {?} */ obj = {};
                for (let /** @type {?} */ h = 0; h < headers.length; h++) {
                    obj[headers[h]] = d[headers[h]];
                }
                arrayObj.push(obj);
            }
            if (exportType == 'print') {
                try {
                    printJS({ printable: arrayObj, properties: headers, type: 'json' });
                    loading.style.display = 'none';
                }
                catch (/** @type {?} */ e) {
                    loading.style.display = 'none';
                    // console.log('PrintJS not found. Add `https://printjs-4de6.kxcdn.com/print.min.js` to your index.html or add as part of angular.json script') ;
                }
            }
            else {
                let /** @type {?} */ extension = (exportType == 'pdf') ? 'pdf' : (exportType == 'excel') ? 'xlsx' : 'docx';
                let /** @type {?} */ pageId = (exportType == 'pdf') ? 3 : (exportType == 'excel') ? 5 : 4;
                let /** @type {?} */ requestData = { "data": JSON.stringify(arrayObj) };
                this.service.postData('http://exporter.azurewebsites.net/api/export/ExportFromJSON/', pageId, requestData).subscribe((resp) => {
                    var /** @type {?} */ download = 'http://exporter.azurewebsites.net/api/export/GetFile/' + resp;
                    download += "?fileName=andrei&extension=" + extension;
                    window.location.href = download;
                    loading.style.display = 'none';
                }, (e => {
                    //console.log('file export error',e) ;
                }));
            }
        }).unsubscribe();
    }
    /**
     * @return {?}
     */
    hasImageEmbeded() {
        return this.embedPictures.hasOwnProperty("index");
    }
    /**
     * @return {?}
     */
    getImage() {
        console.log('eoeoe');
        //   let img = new Image() ;
        //   img.src = imageSrc ;
        //   img.onload = ((e)=>{
        //       return imageSrc ;
        //   }) ;
        //   img.onerror = ((e)=>{
        //     return this.embedPictures.fallbackUrl ;
        //   })
    }
    /**
     * @param {?} JSONData
     * @param {?} ReportTitle
     * @param {?} ShowLabel
     * @return {?}
     */
    JSONToCSVConvertor(JSONData, ReportTitle, ShowLabel) {
        //If JSONData is not an object then JSON.parse will parse the JSON string in an Object
        var /** @type {?} */ arrData = typeof JSONData != 'object' ? JSON.parse(JSONData) : JSONData;
        var /** @type {?} */ CSV = '';
        //Set Report title in first row or line
        CSV += ReportTitle + '\r\n\n';
        //This condition will generate the Label/Header
        if (ShowLabel) {
            var /** @type {?} */ row = "";
            //This loop will extract the label from 1st index of on array
            for (var /** @type {?} */ index in arrData[0]) {
                //Now convert each value to string and comma-seprated
                row += index + ',';
            }
            row = row.slice(0, -1);
            //append Label row with line break
            CSV += row + '\r\n';
        }
        //1st loop is to extract each row
        for (var /** @type {?} */ i = 0; i < arrData.length; i++) {
            var /** @type {?} */ row = "";
            //2nd loop will extract each column and convert it in string comma-seprated
            for (var /** @type {?} */ index in arrData[i]) {
                row += '"' + arrData[i][index] + '",';
            }
            row.slice(0, row.length - 1);
            //add a line break after each row
            CSV += row + '\r\n';
        }
        if (CSV == '') {
            alert("Invalid data");
            return;
        }
        //Generate a file name
        var /** @type {?} */ fileName = "MyReport_";
        //this will remove the blank-spaces from the title and replace it with an underscore
        fileName += ReportTitle.replace(/ /g, "_");
        //Initialize file format you want csv or xls
        var /** @type {?} */ uri = 'data:text/csv;charset=utf-8,' + escape(CSV);
        // Now the little tricky part.
        // you can use either>> window.open(uri);
        // but this will not work in some browsers
        // or you will not get the correct file extension
        //this trick will generate a temp <a /> tag
        var /** @type {?} */ link = document.createElement("a");
        link.href = uri;
        //set the visibility hidden so it will not effect on your web-layout
        link.style = "visibility:hidden";
        link.download = fileName + ".csv";
        //this part will append the anchor tag and remove it after automatic click
        document.body.appendChild(link);
        link.click();
        document.body.removeChild(link);
    }
    /**
     *
     * @param {?} newData
     * @return {?}
     */
    changeDisplayData(newData) {
        this.behavior.next(newData);
        this.setColumnWidths();
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        $(window).resize(function () {
            /* Same as before */
        }).resize();
        if (this.isLazyloadingEnabled()) {
            this.reloadUrl = this.dataUrl + '&' + this.lazyloadingConfig.apiOffsetKey + '=0&' + this.lazyloadingConfig.apiSearchKey + '=';
        }
        else {
            this.reloadUrl = this.dataUrl;
        }
        this.searchForm = this.__form.group({
            searchString: ['', Validators.required]
        });
        this.searchForm = this.__form.group({
            searchString: ['', Validators.required]
        });
        this.service.limit = this.limit;
        this.service.setLazyloadingConfig(this.lazyloadingConfig);
        this.service.setDataUrl(this.dataUrl);
        this.service.setDataSrcKey(this.dataSrcKey);
        let /** @type {?} */ url = (this.isLazyloadingEnabled()) ? this.dataUrl + '&' + this.lazyloadingConfig.apiOffsetKey + '=0&' + this.lazyloadingConfig.apiSearchKey + '=' : this.dataUrl;
        this.service.loadFlxDataTableData(url);
        this.service.flxData.subscribe((resp) => {
            this.tData = resp;
            let /** @type {?} */ obj = [];
            if (this.tData.length > this.limit) {
                for (let /** @type {?} */ i = 0; i < this.limit; i++) {
                    obj.push(this.tData[i]);
                }
                // this.service.dataOffset = this.limit;
            }
            else {
                for (let /** @type {?} */ i = 0; i < this.tData.length; i++) {
                    obj.push(this.tData[i]);
                }
                // this.service.dataOffset = obj.length;
            }
            this.searchDataTempOffset = obj;
            this.changeDisplayData(obj);
        });
    }
    /**
     * @param {?} values
     * @param {?} form
     * @return {?}
     */
    searchDataInApi(values, form) {
        this.service.chageDataTable([]);
        this.service.loadFlxDataTableData(this.dataUrl + '&' + this.lazyloadingConfig.apiOffsetKey + '=0&' + this.lazyloadingConfig.apiSearchKey + '=' + values.searchString);
    }
    /**
     * @return {?}
     */
    ngAfterViewInit() {
        let /** @type {?} */ eachColumnWidth = this.returColumnswidth();
        $(window).ready(function () {
            $(".flx-table-thead tr th").css({ 'width': eachColumnWidth + '%' });
        });
    }
    /**
     * @return {?}
     */
    setColumnWidths() {
        let /** @type {?} */ eachColumnWidth = this.returColumnswidth();
        $(window).ready(function () {
            $(".flx-table-data tbody tr td").css({ 'width': eachColumnWidth + '%' });
        });
    }
    /**
     * @return {?}
     */
    returColumnswidth() {
        let /** @type {?} */ hasNumbers = (!this.hideNumbers) ? 0 : 1;
        let /** @type {?} */ hasActionButtons = (!this.hasActionButtons) ? 0 : 1;
        let /** @type {?} */ numberOfColumns = (this.headers.length + 1) + hasActionButtons + hasNumbers;
        console.log('number_of_columns', numberOfColumns);
        let /** @type {?} */ eachColumnWidth = 100 / numberOfColumns;
        return eachColumnWidth;
    }
    /**
     * @param {?} index
     * @param {?} buttonIndex
     * @return {?}
     */
    actionButtonClicked(index, buttonIndex) {
        this.displayData.subscribe((data) => {
            if (buttonIndex == 0) {
                return this.firstActionButtonClicked.emit({ index: index, data: data[index] });
            }
            else if (buttonIndex == 1) {
                return this.secondActionButtonClicked.emit({ index: index, data: data[index] });
            }
            else {
                this.thirdActionButtonClicked.emit({ index: index, data: data[index] });
            }
        }).unsubscribe();
    }
    /**
     * @return {?}
     */
    addButtonClick() {
        this.addButtonClicked.emit();
    }
    /**
     * @return {?}
     */
    confirmDelete() {
        return this.multipleSelectClicked.emit(this.service.multipleDeletion);
    }
    /**
     * @param {?} checked
     * @return {?}
     */
    addRemove(checked) {
        if (checked) {
            this.displayData.subscribe((data) => {
                for (let /** @type {?} */ i of data) {
                    try {
                        this.service.multipleDeletion.push(i[this.multipleSelectKey]);
                    }
                    catch (/** @type {?} */ e) { }
                }
                // console.log(this.service.multipleDeletion) ;
            }).unsubscribe();
        }
        else {
            this.service.multipleDeletion = [];
        }
    }
    /**
     * @param {?} dataKeyvalue
     * @param {?} index
     * @param {?} selected
     * @return {?}
     */
    addRemoveDeleteItem(dataKeyvalue, index, selected) {
        if (!selected) {
            for (var /** @type {?} */ i = 0; i < this.service.multipleDeletion.length; i++) {
                if (dataKeyvalue == this.service.multipleDeletion[i]) {
                    this.service.multipleDeletion.splice(i, 1);
                    break;
                }
            }
        }
        else {
            this.displayData.subscribe((resp) => {
                this.service.multipleDeletion.push(resp[index][this.multipleSelectKey]);
            }).unsubscribe();
        }
        // console.log('left '+dataKeyvalue,this.service.multipleDeletion) ;
    }
    /**
     * @return {?}
     */
    getSearchColumns() {
        return (this.hasAddButton) ? (this.enableDataExports) ? 'col-md-6 search-container' : 'col-md-7 search-container' :
            (this.enableDataExports) ? 'col-md-7 search-container' : 'col-md-8 search-container';
    }
    /**
     * @return {?}
     */
    disablePrevtButton() {
        return Math.ceil(this.service.dataOffset / this.limit) <= 1;
    }
    /**
     * @return {?}
     */
    disableNextButton() {
        return Math.ceil(this.service.dataOffset / this.limit) == Math.ceil(this.service.totalItems / this.limit);
    }
    /**
     * @return {?}
     */
    isLazyloadingEnabled() {
        return this.lazyloadingConfig.hasOwnProperty("apiOffsetKey") && this.lazyloadingConfig.apiOffsetKey;
    }
    /**
     * @param {?} type
     * @return {?}
     */
    nextPrevItem(type) {
        if (this.isLazyloadingEnabled()) {
            this.service.loadFinish = false;
            this.service.getDataLength().then(dataLength => {
                this.service.chageDataTable([]);
                this.service.dataOffset = (type == 'prev') ? ((this.service.dataOffset - this.limit) - this.limit) : this.service.dataOffset;
                let /** @type {?} */ url = (this.isLazyloadingEnabled()) ? this.dataUrl + '&' + this.lazyloadingConfig.apiOffsetKey + '=' + this.service.dataOffset + '&' + this.lazyloadingConfig.apiSearchKey + '=' : this.dataUrl;
                this.service.loadFlxDataTableData(url);
            }).catch(e => {
                //console.log('error',e) ;
            });
            return;
        }
        // Paginate if lazyloading is disabled
        if (type == 'prev') {
            this.paginateDatatableRecord((this.service.dataOffset - this.limit) - this.limit);
        }
        else {
            if (this.service.dataOffset < this.limit) {
                this.paginateDatatableRecord(this.service.dataOffset + (this.limit - 1));
            }
            else {
                this.paginateDatatableRecord(this.service.dataOffset);
            }
        }
    }
    /**
     * @param {?=} searchString
     * @return {?}
     */
    filterData(searchString = '') {
        if (searchString == '') {
            this.isSearching = false;
        }
        else {
            this.isSearching = true;
        }
        this.changeDisplayData([]);
        this.service.flxData.subscribe((data) => {
            let /** @type {?} */ searchResults = [];
            //If no string provided. Register all the previous data to the dataset
            if (searchString.trim() == '') {
                this.changeDisplayData(this.searchDataTempOffset);
                return;
            }
            //Check if searchKeys are set else use dataKeys as searchKeys
            let /** @type {?} */ searchKeys = (this.searchKeys.length < 1) ? this.dataKeys : this.searchKeys;
            for (let /** @type {?} */ i = 0; i < data.length; i++) {
                //Variable to check if data is found
                let /** @type {?} */ found = -1;
                for (let /** @type {?} */ x = 0; x < searchKeys.length; x++) {
                    try {
                        if (data[i][String(searchKeys[x])].toLowerCase().indexOf(searchString.toLocaleLowerCase()) !== -1) {
                            found = i;
                            break;
                        }
                    }
                    catch (/** @type {?} */ e) { }
                }
                //If found push the index of the data to the searchResults variable
                if (found > -1) {
                    searchResults.push(data[found]);
                }
            }
            this.searchResultFound = searchResults;
            //Register the results to the dataset
            this.changeDisplayData(searchResults);
            this.setColumnWidths();
        }).unsubscribe();
    }
    /**
     * @param {?} value offset value
     * @return {?}
     */
    paginateDatatable(value) {
        // Check if lazy loading is enabled
        if (this.isLazyloadingEnabled()) {
            this.service.loadFinish = false;
            // Subscribe to get the data length
            this.service.getDataLength().then(() => {
                this.service.chageDataTable([]);
                // Check if all is selected to prevent NAN value
                if (value != 'all') {
                    this.service.dataOffset = parseInt(value);
                }
                // setup url
                let /** @type {?} */ url = (this.isLazyloadingEnabled()) ? this.dataUrl + '&' + this.lazyloadingConfig['apiOffsetKey'] + '=' + value + '&' + this.lazyloadingConfig.apiSearchKey + '=' : this.dataUrl;
                // paginate
                this.service.loadFlxDataTableData(url, false);
            }).catch(e => {
                // console.log('error',e) ;
            });
            return;
        }
        this.paginateDatatableRecord(value);
    }
    /**
     *
     * @param {?} value pagination number
     * Perform pagination to the dataset
     * @return {?}
     */
    paginateDatatableRecord(value) {
        if (this.lazyloadingConfig.hasOwnProperty("apiOffsetKey") && this.lazyloadingConfig['apiOffsetKey']) {
            this.service.loadFinish = false;
            this.service.getDataLength().then(dataLength => {
                this.service.chageDataTable([]);
                this.service.dataOffset = parseInt(value) + this.limit;
                this.service.loadFlxDataTableData(this.dataUrl + '&' + this.lazyloadingConfig.apiOffsetKey + '=' + value + '&' + this.lazyloadingConfig.apiSearchKey + '=');
                this.setColumnWidths();
            }).catch(e => {
                // console.log('error',e) ;
            });
            return;
        }
        let /** @type {?} */ num = parseInt(value);
        if (num <= 0) {
            this.offset = 1;
            this.service.dataOffset = this.limit;
        }
        else {
            if (value != 'all') {
                this.offset = num + 1;
                this.service.dataOffset = num + this.limit;
            }
            else {
                this.offset = 1;
            }
        }
        this.service.flxData.subscribe((data) => {
            if (value !== 'all') {
                let /** @type {?} */ paginateResult = [];
                for (let /** @type {?} */ i = value; i < (this.limit + parseInt(value)); i++) {
                    if (data[i]) {
                        paginateResult.push(data[i]);
                    }
                }
                if (paginateResult.length > 0) {
                    this.changeDisplayData(paginateResult);
                }
            }
            else {
                this.changeDisplayData(data);
                this.searchDataTempOffset = data;
            }
            this.setColumnWidths();
        }).unsubscribe();
    }
}
FlxUiDatatableComponent.ɵfac = function FlxUiDatatableComponent_Factory(t) { return new (t || FlxUiDatatableComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc2.FormBuilder), ɵngcc0.ɵɵdirectiveInject(FlxUiDatatableService)); };
FlxUiDatatableComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: FlxUiDatatableComponent, selectors: [["flx-ui-datatable"]], inputs: { classes: "classes", headers: "headers", lazyloadingConfig: "lazyloadingConfig", embedPictures: "embedPictures", dataKeys: "dataKeys", enableDataExports: "enableDataExports", dataExportsConfig: "dataExportsConfig", searchKeys: "searchKeys", dataSrcKey: "dataSrcKey", hasActionButtons: "hasActionButtons", hideNumbers: "hideNumbers", enableMultipleSelection: "enableMultipleSelection", multipleSelectKey: "multipleSelectKey", hasAddButton: "hasAddButton", dataUrl: "dataUrl", actionButtonStart: "actionButtonStart", multipleSelectButton: "multipleSelectButton", searchPlaceholder: "searchPlaceholder", actionHeader: "actionHeader", limit: "limit", spinnerSrc: "spinnerSrc", actionButtons: "actionButtons", paginationButtons: "paginationButtons", tableHeader: "tableHeader", searchButton: "searchButton", addButton: "addButton", searchBar: "searchBar", actionButtonsConfig: "actionButtonsConfig", textSubstring: "textSubstring", conditionManipulator: "conditionManipulator", showBottomInfo: "showBottomInfo" }, outputs: { firstActionButtonClicked: "firstActionButtonClicked", secondActionButtonClicked: "secondActionButtonClicked", thirdActionButtonClicked: "thirdActionButtonClicked", multipleSelectClicked: "multipleSelectClicked", addButtonClicked: "addButtonClicked" }, decls: 46, vars: 58, consts: [["id", "export_loading", 1, "col-md-12", "text-center", 2, "display", "none", "margin-bottom", "0.5em", "color", "#dddddd", "font-size", "13px", "font-weight", "bold"], [1, "col-md-12", "flx-ui-datatable-header"], [1, "container-fluid"], [1, "row"], [1, "col-xs-3", "col-sm-3", "pagination-select", "col-md-2", 2, "position", "relative", "z-index", "1"], [3, "change"], [3, "value", 4, "ngFor", "ngForOf"], [1, "col-xs-5", "col-sm-5", "col-md-2", "text-center", "flx-datatable-pagination", "rmpd", 2, "position", "relative", "z-index", "2"], ["mat-icon-button", "", 3, "ngClass", "disabled", "click"], [1, "fa", "fa-angle-double-left", "fa-1x"], [1, "flx-datatable-tooltip-prev", 3, "ngClass"], [4, "ngIf", "ngIfElse"], ["searching", ""], [1, "fa", "fa-angle-double-right", "fa-1x"], [1, "flx-datatable-tooltip-next", 3, "ngClass"], ["type", "text", "class", "form-control rmsh rmrd customclass", 3, "background", "color", "ngStyle", "placeholder", "keyup", 4, "ngIf"], [3, "ngSubmit", 4, "ngIf"], ["class", "col-md-1 text-right rmpd", 4, "ngIf"], ["class", "col-md-1 text-right rmpd export-cnt", 4, "ngIf"], [1, "col-md-12", "rmpd", "table-responsive", "flx-tabl-container"], [4, "ngIf"], [4, "ngFor", "ngForOf"], [1, "flxdatatablebodyscrollbar"], ["class", "flxuidatatablerow", "id", "flxdatatable_singlerow", 4, "ngFor", "ngForOf"], ["class", "col-md-12 rmpd flx-total-data", 4, "ngIf"], [3, "value"], ["type", "text", 1, "form-control", "rmsh", "rmrd", "customclass", 3, "ngStyle", "placeholder", "keyup"], [3, "ngSubmit"], ["srch", "ngForm"], [1, "input-group"], ["type", "text", "required", "", "name", "searchString", "ngModel", "", 3, "ngStyle", "placeholder"], [1, "input-group-addon"], ["type", "submit", 1, "btn", "btn-default", "btn-clear", "btn-md", 3, "disabled"], [1, "fa", "fa-search"], [1, "col-md-1", "text-right", "rmpd"], [3, "click"], [1, "fa", "fa-plus"], [1, "col-md-1", "text-right", "rmpd", "export-cnt"], [1, "dropdown"], ["type", "button", "data-toggle", "dropdown"], [1, "fa", "fa-angle-down"], [1, "dropdown-header"], ["type", "checkbox", 2, "position", "relative", "top", "0.3em", 3, "change"], [2, "font-size", "10px", "color", "#999"], [1, "divider"], ["class", "dropdown-submenu", 3, "click", 4, "ngFor", "ngForOf"], [1, "dropdown-submenu", 3, "click"], ["href", "javascript:void(0)", "class", "btn btn-default btn-sm form-control", 4, "ngIf"], ["href", "javascript:void(0)", "class", "btn btn-default btn-sm form-control", "style", "color:#ff0000", 4, "ngIf"], ["href", "javascript:void(0)", "class", "btn btn-default btn-sm form-control", "style", "color:#009900;", 4, "ngIf"], ["href", "javascript:void(0)", "class", "btn btn-default btn-sm form-control", "style", "color:#335599;", 4, "ngIf"], ["href", "javascript:void(0)", 1, "btn", "btn-default", "btn-sm", "form-control"], [1, "glyphicon", "glyphicon-print"], ["href", "javascript:void(0)", 1, "btn", "btn-default", "btn-sm", "form-control", 2, "color", "#ff0000"], [1, "glyphicon", "glyphicon-file"], ["href", "javascript:void(0)", 1, "btn", "btn-default", "btn-sm", "form-control", 2, "color", "#009900"], ["href", "javascript:void(0)", 1, "btn", "btn-default", "btn-sm", "form-control", 2, "color", "#335599"], ["type", "checkbox", 3, "checked", "change", 4, "ngIf"], ["class", "btn btn-danger btn-xs flx-multiple-deletion-button", 3, "click", 4, "ngIf"], ["type", "checkbox", 3, "checked", "change"], [1, "btn", "btn-danger", "btn-xs", "flx-multiple-deletion-button", 3, "click"], ["colspan", "20", 1, "text-center"], ["style", "margin-top:1em;margin-bottom:1em;", 3, "class", "src", 4, "ngIf"], [2, "margin-top", "1em", "margin-bottom", "1em", 3, "src"], ["id", "flxdatatable_singlerow", 1, "flxuidatatablerow"], ["class", "table-buttons", "scope", "row", 4, "ngIf"], ["style", "color: #999;", 3, "class", 4, "ngIf"], [3, "class", 4, "ngFor", "ngForOf"], ["class", "table-buttons", 4, "ngIf"], ["scope", "row", 1, "table-buttons"], ["type", "checkbox", "checked", "", 3, "change", 4, "ngIf"], ["type", "button", 3, "click"], ["class", "toltip", "class", "flx-tooltip", 3, "ngClass", 4, "ngIf"], ["class", "action-button-icon-left", 3, "class", 4, "ngIf"], [1, "button-text"], [3, "class", 4, "ngIf"], [1, "flx-tooltip", 3, "ngClass"], [1, "action-button-icon-left"], ["type", "checkbox", "checked", "", 3, "change"], [2, "color", "#999"], [3, "class", "src", "flx-ui-datatable-img-fallback", 4, "ngIf", "ngIfElse"], ["nopicture", ""], [3, "src", "flx-ui-datatable-img-fallback"], ["nolimittext", ""], [1, "table-buttons"], ["colspan", "10", "class", "text-center", 4, "ngIf"], ["colspan", "10", 1, "text-center"], [2, "color", "#ff0000", "font-size", "13px"], ["type", "button", "color", "primary", 2, "margin-top", "1em", 3, "click"], [1, "fa", "fa-refresh"], [1, "col-md-12", "rmpd", "flx-total-data"], ["class", "row", 4, "ngIf", "ngIfElse"], ["bottomInfoSearching", ""], [1, "col-md-4", "text-left", "rmpd"], [1, "col-md-4", "text-center", "rmpd"], [1, "col-md-4", "text-right", "rmpd"], [1, "col-md-12", "text-right", "rmpd"]], template: function FlxUiDatatableComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div");
        ɵngcc0.ɵɵelementStart(1, "div", 0);
        ɵngcc0.ɵɵtext(2, "Exporting...");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(3, "div", 1);
        ɵngcc0.ɵɵelementStart(4, "div", 2);
        ɵngcc0.ɵɵelementStart(5, "div", 3);
        ɵngcc0.ɵɵelementStart(6, "div", 4);
        ɵngcc0.ɵɵelementStart(7, "select", 5);
        ɵngcc0.ɵɵlistener("change", function FlxUiDatatableComponent_Template_select_change_7_listener($event) { return ctx.paginateDatatable($event == null ? null : $event.target == null ? null : $event.target.value); });
        ɵngcc0.ɵɵtemplate(8, FlxUiDatatableComponent_option_8_Template, 2, 2, "option", 6);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(9, "div", 7);
        ɵngcc0.ɵɵelementStart(10, "button", 8);
        ɵngcc0.ɵɵlistener("click", function FlxUiDatatableComponent_Template_button_click_10_listener() { return ctx.nextPrevItem("prev"); });
        ɵngcc0.ɵɵelement(11, "span", 9);
        ɵngcc0.ɵɵelementStart(12, "span", 10);
        ɵngcc0.ɵɵtext(13, "Previous");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(14, FlxUiDatatableComponent_span_14_Template, 4, 8, "span", 11);
        ɵngcc0.ɵɵtemplate(15, FlxUiDatatableComponent_ng_template_15_Template, 0, 0, "ng-template", null, 12, ɵngcc0.ɵɵtemplateRefExtractor);
        ɵngcc0.ɵɵelementStart(17, "button", 8);
        ɵngcc0.ɵɵlistener("click", function FlxUiDatatableComponent_Template_button_click_17_listener() { return ctx.nextPrevItem("next"); });
        ɵngcc0.ɵɵelement(18, "span", 13);
        ɵngcc0.ɵɵelementStart(19, "span", 14);
        ɵngcc0.ɵɵtext(20, "Next");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(21, "div");
        ɵngcc0.ɵɵtemplate(22, FlxUiDatatableComponent_input_22_Template, 1, 8, "input", 15);
        ɵngcc0.ɵɵtemplate(23, FlxUiDatatableComponent_form_23_Template, 7, 12, "form", 16);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(24, FlxUiDatatableComponent_div_24_Template, 4, 9, "div", 17);
        ɵngcc0.ɵɵtemplate(25, FlxUiDatatableComponent_div_25_Template, 16, 10, "div", 18);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(26, "div", 2);
        ɵngcc0.ɵɵelementStart(27, "div", 3);
        ɵngcc0.ɵɵelementStart(28, "div", 19);
        ɵngcc0.ɵɵelementStart(29, "table");
        ɵngcc0.ɵɵelementStart(30, "thead");
        ɵngcc0.ɵɵelementStart(31, "tr");
        ɵngcc0.ɵɵtemplate(32, FlxUiDatatableComponent_th_32_Template, 4, 3, "th", 20);
        ɵngcc0.ɵɵtemplate(33, FlxUiDatatableComponent_th_33_Template, 5, 0, "th", 20);
        ɵngcc0.ɵɵtemplate(34, FlxUiDatatableComponent_th_34_Template, 4, 3, "th", 20);
        ɵngcc0.ɵɵtemplate(35, FlxUiDatatableComponent_th_35_Template, 2, 1, "th", 21);
        ɵngcc0.ɵɵtemplate(36, FlxUiDatatableComponent_th_36_Template, 4, 3, "th", 20);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(37, "perfect-scrollbar");
        ɵngcc0.ɵɵelementStart(38, "table");
        ɵngcc0.ɵɵelementStart(39, "tbody", 22);
        ɵngcc0.ɵɵtemplate(40, FlxUiDatatableComponent_tr_40_Template, 3, 1, "tr", 20);
        ɵngcc0.ɵɵtemplate(41, FlxUiDatatableComponent_tr_41_Template, 6, 5, "tr", 23);
        ɵngcc0.ɵɵpipe(42, "async");
        ɵngcc0.ɵɵtemplate(43, FlxUiDatatableComponent_tr_43_Template, 2, 1, "tr", 20);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(44, "div", 3);
        ɵngcc0.ɵɵtemplate(45, FlxUiDatatableComponent_div_45_Template, 4, 2, "div", 24);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        const _r41 = ɵngcc0.ɵɵreference(16);
        ɵngcc0.ɵɵclassMapInterpolate1("col-md-12 flx-ui-datatable-main ", ctx.classes == null ? null : ctx.classes.maincontainer, "");
        ɵngcc0.ɵɵadvance(7);
        ɵngcc0.ɵɵclassMapInterpolate1("form-control rmsh rmrd ", ctx.classes == null ? null : ctx.classes.paginationselect, "");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.service == null ? null : ctx.service.pagination);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵclassMapInterpolate1("flx-ui-datatable-pagination-buttons ", ctx.classes == null ? null : ctx.classes.paginationButton, "");
        ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(50, _c2, ctx.disablePrevtButton()))("disabled", ctx.disablePrevtButton());
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(52, _c2, ctx.disablePrevtButton()));
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.isSearching)("ngIfElse", _r41);
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵclassMapInterpolate1("flx-ui-datatable-pagination-buttons ", ctx.classes == null ? null : ctx.classes.paginationButton, "");
        ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(54, _c2, ctx.disableNextButton()))("disabled", ctx.disableNextButton());
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(56, _c2, ctx.disableNextButton()));
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵclassMap("search-bar " + ctx.getSearchColumns());
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.isLazyloadingEnabled());
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.isLazyloadingEnabled());
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.hasAddButton);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.enableDataExports);
        ɵngcc0.ɵɵadvance(4);
        ɵngcc0.ɵɵclassMapInterpolate1("table ", ctx.classes == null ? null : ctx.classes.tableType, " th-table");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵclassMapInterpolate1("", ctx.classes == null ? null : ctx.classes.tableHeader, " flx-table-thead");
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.hasActionButtons && ((ctx.actionButtonsConfig == null ? null : ctx.actionButtonsConfig.position) == "left" && (ctx.actionButtonsConfig == null ? null : ctx.actionButtonsConfig.beforeNumbers)));
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.hideNumbers);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.hasActionButtons && ((ctx.actionButtonsConfig == null ? null : ctx.actionButtonsConfig.position) == "left" && !(ctx.actionButtonsConfig == null ? null : ctx.actionButtonsConfig.beforeNumbers)));
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.headers);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.hasActionButtons && (ctx.actionButtonsConfig == null ? null : ctx.actionButtonsConfig.position) == "right");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵclassMapInterpolate1("data-scroll-content ", ctx.classes == null ? null : ctx.classes.dataScrollContent, "");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵclassMapInterpolate1("table ", ctx.classes == null ? null : ctx.classes.tableType, " flx-table-data");
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", !(ctx.service == null ? null : ctx.service.loadFinish));
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngForOf", ɵngcc0.ɵɵpipeBind1(42, 48, ctx.displayData));
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", (ctx.tData == null ? null : ctx.tData.length) < 1);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.showBottomInfo);
    } }, directives: function () { return [ɵngcc3.NgForOf, ɵngcc3.NgClass, ɵngcc3.NgIf, ɵngcc4.PerfectScrollbarComponent, ɵngcc2.NgSelectOption, ɵngcc2.ɵangular_packages_forms_forms_x, ɵngcc3.NgStyle, ɵngcc2.ɵangular_packages_forms_forms_y, ɵngcc2.NgControlStatusGroup, ɵngcc2.NgForm, ɵngcc2.DefaultValueAccessor, ɵngcc2.RequiredValidator, ɵngcc2.NgControlStatus, ɵngcc2.NgModel, ImageFallBack]; }, pipes: function () { return [ɵngcc3.AsyncPipe, Ceil,
        HasLimitText,
        LimitText]; }, styles: [".flx-ui-datatable-main[_ngcontent-%COMP%]{background-color:#fff;padding-top:1em;padding-bottom:1em}.flx-ui-datatable-main[_ngcontent-%COMP%]   .btn-danger[_ngcontent-%COMP%]{background-color:#f50057;border:1px solid #f50057;box-shadow:0 3px 5px 1px #ddd;-moz-box-shadow:0 3px 5px 1px #ddd;-webkit-box-shadow:0 3px 5px 1px #ddd;-o-box-shadow:0 3px 5px 1px #ddd;-ms-box-shadow:0 3px 5px 1px #ddd;border-radius:3px;margin-left:.3em}.flx-ui-datatable-main[_ngcontent-%COMP%]   #flxdatatablebodyscrollbar[_ngcontent-%COMP%], .flx-ui-datatable-main[_ngcontent-%COMP%]   .table-responsive[_ngcontent-%COMP%], .flx-ui-datatable-main[_ngcontent-%COMP%]   .table-responsive[_ngcontent-%COMP%]   table[_ngcontent-%COMP%]{width:100%}.flx-ui-datatable-main[_ngcontent-%COMP%]   img.img-fall-back[_ngcontent-%COMP%]{width:30px;height:30px}.flx-ui-datatable-main[_ngcontent-%COMP%]   .pagination-select[_ngcontent-%COMP%]   input[type=text][_ngcontent-%COMP%], .flx-ui-datatable-main[_ngcontent-%COMP%]   .pagination-select[_ngcontent-%COMP%]   select[_ngcontent-%COMP%], .flx-ui-datatable-main[_ngcontent-%COMP%]   .search-bar[_ngcontent-%COMP%]   input[type=text][_ngcontent-%COMP%], .flx-ui-datatable-main[_ngcontent-%COMP%]   .search-bar[_ngcontent-%COMP%]   select[_ngcontent-%COMP%]{border-top:none!important;border-left:none!important;border-right:none!important;border-bottom:2px solid #ddd!important;border-radius:0!important;box-shadow:0 0 0 0 transparent!important;-moz-box-shadow:0 0 0 0 transparent!important;-webkit-box-shadow:0 0 0 0 transparent!important;-o-box-shadow:0 0 0 0 transparent!important;-ms-box-shadow:0 0 0 0 transparent!important}.flx-ui-datatable-main[_ngcontent-%COMP%]   .pagination-select[_ngcontent-%COMP%]   input[type=text][_ngcontent-%COMP%]:focus, .flx-ui-datatable-main[_ngcontent-%COMP%]   .pagination-select[_ngcontent-%COMP%]   select[_ngcontent-%COMP%]:focus, .flx-ui-datatable-main[_ngcontent-%COMP%]   .search-bar[_ngcontent-%COMP%]   input[type=text][_ngcontent-%COMP%]:focus, .flx-ui-datatable-main[_ngcontent-%COMP%]   .search-bar[_ngcontent-%COMP%]   select[_ngcontent-%COMP%]:focus{border-bottom-color:#359!important;transition:.5s;-moz-transition:.5s;-webkit-transition:.5s;-o-transition:.5s;-ms-transition:.5s}.flx-ui-datatable-main[_ngcontent-%COMP%]   .pagination-select[_ngcontent-%COMP%]   select[_ngcontent-%COMP%], .flx-ui-datatable-main[_ngcontent-%COMP%]   .search-bar[_ngcontent-%COMP%]   select[_ngcontent-%COMP%]{-webkit-appearance:none;appearance:none;-moz-appearance:none}.flx-ui-datatable-main[_ngcontent-%COMP%]   .dropdown-toggle[_ngcontent-%COMP%]::after{display:none}.flx-ui-datatable-main[_ngcontent-%COMP%]   .dropdown-menu-export[_ngcontent-%COMP%]{margin-left:-130px;padding-left:1em;padding-right:1em;width:100px!important;margin-top:1em;border:1px solid #ddd!important}.flx-ui-datatable-main[_ngcontent-%COMP%]   .dropdown-menu-export[_ngcontent-%COMP%]::before{content:\"\";width:auto;height:auto;position:absolute;top:-20px;right:3px;border:10px solid;border-color:transparent transparent #ddd}.flx-ui-datatable-main[_ngcontent-%COMP%]   .flx-datatable-pagination[_ngcontent-%COMP%]{padding-top:.5em}.flx-ui-datatable-main[_ngcontent-%COMP%]   .flx-datatable-pagination[_ngcontent-%COMP%]   button[_ngcontent-%COMP%]{width:35px!important;height:35px!important;border-radius:50em!important;border:none!important;box-shadow:0 3px 10px 0 #b3c4e6;-moz-box-shadow:0 3px 10px 0 #b3c4e6;-webkit-box-shadow:0 3px 10px 0 #b3c4e6;-o-box-shadow:0 3px 10px 0 #b3c4e6;-ms-box-shadow:0 3px 10px 0 #b3c4e6;background-color:#359;color:#fff;font-size:23px;font-weight:700;position:absolute;top:0}.flx-ui-datatable-main[_ngcontent-%COMP%]   .flx-datatable-pagination[_ngcontent-%COMP%]   button[_ngcontent-%COMP%]:first-child{left:0}.flx-ui-datatable-main[_ngcontent-%COMP%]   .flx-datatable-pagination[_ngcontent-%COMP%]   button[_ngcontent-%COMP%]:first-child   .flx-datatable-tooltip-prev[_ngcontent-%COMP%]{position:absolute;left:0;font-size:13px;font-weight:400;color:#fff;background-color:#359;padding-left:.3em;padding-right:.3em;border-radius:3px;margin-left:-57px;margin-top:.3em;box-shadow:0 3px 10px 0 #b3c4e6!important;-moz-box-shadow:0 3px 10px 0 #b3c4e6!important;-webkit-box-shadow:0 3px 10px 0 #b3c4e6!important;-o-box-shadow:0 3px 10px 0 #b3c4e6!important;-ms-box-shadow:0 3px 10px 0 #b3c4e6!important;visibility:hidden}.flx-ui-datatable-main[_ngcontent-%COMP%]   .flx-datatable-pagination[_ngcontent-%COMP%]   button[_ngcontent-%COMP%]:hover > span.flx-datatable-tooltip-next[_ngcontent-%COMP%], .flx-ui-datatable-main[_ngcontent-%COMP%]   .flx-datatable-pagination[_ngcontent-%COMP%]   button[_ngcontent-%COMP%]:hover > span.flx-datatable-tooltip-prev[_ngcontent-%COMP%]{visibility:visible}.flx-ui-datatable-main[_ngcontent-%COMP%]   .flx-datatable-pagination[_ngcontent-%COMP%]   .flx-pagination-end[_ngcontent-%COMP%]{background-color:#f50057!important;box-shadow:0 3px 10px 0 #ffc2d8!important;-moz-box-shadow:0 3px 10px 0 #ffc2d8!important;-webkit-box-shadow:0 3px 10px 0 #ffc2d8!important;-o-box-shadow:0 3px 10px 0 #ffc2d8!important;-ms-box-shadow:0 3px 10px 0 #ffc2d8!important;cursor:not-allowed}.flx-ui-datatable-main[_ngcontent-%COMP%]   .flx-datatable-pagination[_ngcontent-%COMP%]   .flx-pagination-end[_ngcontent-%COMP%]:hover > span.flx-datatable-tooltip-next[_ngcontent-%COMP%], .flx-ui-datatable-main[_ngcontent-%COMP%]   .flx-datatable-pagination[_ngcontent-%COMP%]   .flx-pagination-end[_ngcontent-%COMP%]:hover > span.flx-datatable-tooltip-prev[_ngcontent-%COMP%]{visibility:hidden}.flx-ui-datatable-main[_ngcontent-%COMP%]   .flx-datatable-pagination[_ngcontent-%COMP%]   button[_ngcontent-%COMP%]:last-child{right:0}.flx-ui-datatable-main[_ngcontent-%COMP%]   .flx-datatable-pagination[_ngcontent-%COMP%]   button[_ngcontent-%COMP%]:last-child   .flx-datatable-tooltip-next[_ngcontent-%COMP%]{position:absolute;left:0;font-size:13px;font-weight:400;color:#fff;background-color:#359;padding-left:.3em;padding-right:.3em;border-radius:3px;margin-left:35px;margin-top:.5em;box-shadow:0 3px 10px 0 #b3c4e6!important;-moz-box-shadow:0 3px 10px 0 #b3c4e6!important;-webkit-box-shadow:0 3px 10px 0 #b3c4e6!important;-o-box-shadow:0 3px 10px 0 #b3c4e6!important;-ms-box-shadow:0 3px 10px 0 #b3c4e6!important;visibility:hidden}.flx-ui-datatable-main[_ngcontent-%COMP%]   .export-cnt[_ngcontent-%COMP%]   button[_ngcontent-%COMP%]{border-radius:50em!important}.flx-ui-datatable-main[_ngcontent-%COMP%]   .flx-tabl-container[_ngcontent-%COMP%]   table.th-table[_ngcontent-%COMP%]{margin-top:.5em}.flx-ui-datatable-main[_ngcontent-%COMP%]   table[_ngcontent-%COMP%]   tbody[_ngcontent-%COMP%]   tr[_ngcontent-%COMP%]{padding-top:0!important}.flx-ui-datatable-main[_ngcontent-%COMP%]   table[_ngcontent-%COMP%]   tbody[_ngcontent-%COMP%]   tr[_ngcontent-%COMP%]   td[_ngcontent-%COMP%]{padding-top:.5em;border-top:1px solid #f0f0f0;border-bottom:1px solid #f0f0f0}.flx-ui-datatable-main[_ngcontent-%COMP%]   table[_ngcontent-%COMP%]   tbody[_ngcontent-%COMP%]   tr[_ngcontent-%COMP%]   td[_ngcontent-%COMP%]   button[_ngcontent-%COMP%]{margin-right:.3em;margin-left:0}.flx-ui-datatable-main[_ngcontent-%COMP%]   table[_ngcontent-%COMP%]   tbody[_ngcontent-%COMP%]   tr[_ngcontent-%COMP%]   td[_ngcontent-%COMP%]   button[_ngcontent-%COMP%]   div.flx-tooltip[_ngcontent-%COMP%]{position:absolute;background-color:rgba(32,27,27,.808);text-align:center;font-size:13px;color:#fff;border-radius:3px;box-shadow:0 3px 20px 0 #4b4949;-moz-box-shadow:0 3px 20px 0 #4b4949;-webkit-box-shadow:0 3px 20px 0 #4b4949;-o-box-shadow:0 3px 20px 0 #4b4949;-ms-box-shadow:0 3px 20px 0 #4b4949;margin-left:-2.5em;margin-top:-2.8em;visibility:hidden;width:80px;padding:.3em .5em}.flx-ui-datatable-main[_ngcontent-%COMP%]   table[_ngcontent-%COMP%]   tbody[_ngcontent-%COMP%]   tr[_ngcontent-%COMP%]   td[_ngcontent-%COMP%]   button[_ngcontent-%COMP%]   .flx-tooltip-left[_ngcontent-%COMP%]{margin-left:-95px!important;margin-top:-.3em!important}.flx-ui-datatable-main[_ngcontent-%COMP%]   table[_ngcontent-%COMP%]   tbody[_ngcontent-%COMP%]   tr[_ngcontent-%COMP%]   td[_ngcontent-%COMP%]   button[_ngcontent-%COMP%]   .flx-tooltip-bottom[_ngcontent-%COMP%]{margin-top:2.3em!important}.flx-ui-datatable-main[_ngcontent-%COMP%]   table[_ngcontent-%COMP%]   tbody[_ngcontent-%COMP%]   tr[_ngcontent-%COMP%]   td[_ngcontent-%COMP%]   button[_ngcontent-%COMP%]   .flx-tooltip-right[_ngcontent-%COMP%]{margin-left:28px!important;margin-top:-.3em!important}.flx-ui-datatable-main[_ngcontent-%COMP%]   table[_ngcontent-%COMP%]   tbody[_ngcontent-%COMP%]   tr[_ngcontent-%COMP%]   td[_ngcontent-%COMP%]   button[_ngcontent-%COMP%]:hover > div.flx-tooltip[_ngcontent-%COMP%]{transition:.3s;visibility:visible}.flx-ui-datatable-main[_ngcontent-%COMP%]   table[_ngcontent-%COMP%]   tbody[_ngcontent-%COMP%]   tr[_ngcontent-%COMP%]   td.table-buttons[_ngcontent-%COMP%]{padding-top:.2em;padding-bottom:.2em}.flx-ui-datatable-main[_ngcontent-%COMP%]   table[_ngcontent-%COMP%]   tbody[_ngcontent-%COMP%]   tr[_ngcontent-%COMP%]:nth-of-type(even){background-color:#f8f9fa}.flx-ui-datatable-main[_ngcontent-%COMP%]   table[_ngcontent-%COMP%]   tbody[_ngcontent-%COMP%]   tr[_ngcontent-%COMP%]:nth-of-type(odd){background-color:#fff}.flx-ui-datatable-main[_ngcontent-%COMP%]   .btn-danger[_ngcontent-%COMP%]:hover{background-color:#ff146b;border:1px solid #ff146b;box-shadow:0 3px 10px 1px #ff5fb6;-moz-box-shadow:0 3px 10px 1px #ff5fb6;-webkit-box-shadow:0 3px 10px 1px #ff5fb6;-o-box-shadow:0 3px 10px 1px #ff5fb6;-ms-box-shadow:0 3px 10px 1px #ff5fb6;transition:.5s;-moz-transition:.5s;-webkit-transition:.5s;-o-transition:.5s;-ms-transition:.5s}.flx-ui-datatable-main[_ngcontent-%COMP%]   .btn-danger[_ngcontent-%COMP%]:focus{background-color:#f50057;border:1px solid #f50057}.flx-ui-datatable-main[_ngcontent-%COMP%]   .btn-white[_ngcontent-%COMP%]{background-color:#fff}.flx-ui-datatable-main[_ngcontent-%COMP%]   .btn-dark[_ngcontent-%COMP%]{background-color:#222!important}.flx-ui-datatable-main[_ngcontent-%COMP%]   .btn-primary[_ngcontent-%COMP%]{background-color:#359;color:#fff;border:1px solid #359;box-shadow:0 3px 5px 1px #ddd;-moz-box-shadow:0 3px 5px 1px #ddd;-webkit-box-shadow:0 3px 5px 1px #ddd;-o-box-shadow:0 3px 5px 1px #ddd;-ms-box-shadow:0 3px 5px 1px #ddd;border-radius:3px}.flx-ui-datatable-main[_ngcontent-%COMP%]   .btn-primary[_ngcontent-%COMP%]:hover{background-color:#4769ad;border:1px solid #4769ad;box-shadow:0 3px 10px 0 #b3c4e6;-moz-box-shadow:0 3px 10px 0 #b3c4e6;-webkit-box-shadow:0 3px 10px 0 #b3c4e6;-o-box-shadow:0 3px 10px 0 #b3c4e6;-ms-box-shadow:0 3px 10px 0 #b3c4e6;transition:.5s;-moz-transition:.5s;-webkit-transition:.5s;-o-transition:.5s;-ms-transition:.5s}.flx-ui-datatable-main[_ngcontent-%COMP%]   .btn-primary[_ngcontent-%COMP%]:focus{background-color:#359;border:1px solid #359}.flx-ui-datatable-main[_ngcontent-%COMP%]   .btn-large[_ngcontent-%COMP%]{padding-top:1em!important;padding-bottom:1em!important}.flx-ui-datatable-main[_ngcontent-%COMP%]   .btn-medium[_ngcontent-%COMP%]{padding-top:.7em!important;padding-bottom:.7em!important}.flx-ui-datatable-main[_ngcontent-%COMP%]   .btn-success[_ngcontent-%COMP%]{box-shadow:0 3px 5px 1px #ddd;-moz-box-shadow:0 3px 5px 1px #ddd;-webkit-box-shadow:0 3px 5px 1px #ddd;-o-box-shadow:0 3px 5px 1px #ddd;-ms-box-shadow:0 3px 5px 1px #ddd;border-radius:3px;background-color:#5cb85c;border:1px solid #5cb85c}.flx-ui-datatable-main[_ngcontent-%COMP%]   .btn-success[_ngcontent-%COMP%]:hover{background-color:#70cc70;border:1px solid #70cc70;box-shadow:0 3px 10px 1px #9df99d;-moz-box-shadow:0 3px 10px 1px #9df99d;-webkit-box-shadow:0 3px 10px 1px #9df99d;-o-box-shadow:0 3px 10px 1px #9df99d;-ms-box-shadow:0 3px 10px 1px #9df99d;transition:.5s;-moz-transition:.5s;-webkit-transition:.5s;-o-transition:.5s;-ms-transition:.5s}.flx-ui-datatable-main[_ngcontent-%COMP%]   .btn-success[_ngcontent-%COMP%]:focus{background-color:#5cb85c;border:1px solid #5cb85c}.flx-ui-datatable-main[_ngcontent-%COMP%]   .btn-default[_ngcontent-%COMP%]{background-color:#fff;box-shadow:0 3px 5px 1px #eee;-moz-box-shadow:0 3px 5px 1px #eee;-webkit-box-shadow:0 3px 5px 1px #eee;-o-box-shadow:0 3px 5px 1px #eee;-ms-box-shadow:0 3px 5px 1px #eee;border-radius:3px;border:1px solid #ddd}.flx-ui-datatable-main[_ngcontent-%COMP%]   .btn-default[_ngcontent-%COMP%]:hover{background-color:#fff;border:1px solid #e7e7e7;box-shadow:0 3px 10px 1px #e2e2e2;-moz-box-shadow:0 3px 10px 1px #e2e2e2;-webkit-box-shadow:0 3px 10px 1px #e2e2e2;-o-box-shadow:0 3px 10px 1px #e2e2e2;-ms-box-shadow:0 3px 10px 1px #e2e2e2;transition:.5s;-moz-transition:.5s;-webkit-transition:.5s;-o-transition:.5s;-ms-transition:.5s}.flx-ui-datatable-main[_ngcontent-%COMP%]   .btn-default[_ngcontent-%COMP%]:focus{background-color:#fff;border:1px solid #ddd}.flx-ui-datatable-main[_ngcontent-%COMP%]   .btn-secondary[_ngcontent-%COMP%]{box-shadow:0 3px 5px 1px #ddd;-moz-box-shadow:0 3px 5px 1px #ddd;-webkit-box-shadow:0 3px 5px 1px #ddd;-o-box-shadow:0 3px 5px 1px #ddd;-ms-box-shadow:0 3px 5px 1px #ddd;border-radius:3px;background-color:#1e88e5;color:#fff}.flx-ui-datatable-main[_ngcontent-%COMP%]   .btn-secondary[_ngcontent-%COMP%]:hover{color:#fff;background-color:#2892ef;border:1px solid #2892ef;box-shadow:0 3px 10px 1px #55bfff;-moz-box-shadow:0 3px 10px 1px #55bfff;-webkit-box-shadow:0 3px 10px 1px #55bfff;-o-box-shadow:0 3px 10px 1px #55bfff;-ms-box-shadow:0 3px 10px 1px #55bfff;transition:.5s;-moz-transition:.5s;-webkit-transition:.5s;-o-transition:.5s;-ms-transition:.5s}.flx-ui-datatable-main[_ngcontent-%COMP%]   .btn-secondary[_ngcontent-%COMP%]:focus{color:#fff}.flx-ui-datatable-main[_ngcontent-%COMP%]   .pagination-button[_ngcontent-%COMP%]{background-color:#359;color:#fff}.flx-ui-datatable-main[_ngcontent-%COMP%]   .table-font[_ngcontent-%COMP%]{font-family:Khula,sans-serif!important}.flx-ui-datatable-main[_ngcontent-%COMP%]   .table-header-icon[_ngcontent-%COMP%]{position:absolute;right:.2em;width:80px;height:80px;font-size:50px;margin-top:-30px;border-radius:5px!important}.flx-ui-datatable-main[_ngcontent-%COMP%]   .table-title[_ngcontent-%COMP%]{background-color:#359;color:#fff;border-radius:2px;padding:1em;font-size:15px;font-weight:700;margin-bottom:1.5em;font-family:Roboto,sans-serif;box-shadow:0 1px 5px 1px #ddd;-moz-box-shadow:0 1px 5px 1px #ddd;-webkit-box-shadow:0 1px 5px 1px #ddd;-o-box-shadow:0 1px 5px 1px #ddd;-ms-box-shadow:0 1px 5px 1px #ddd}.flx-ui-datatable-main[_ngcontent-%COMP%]   .danger[_ngcontent-%COMP%]{background-color:#f50057;color:#fff}.flx-ui-datatable-main[_ngcontent-%COMP%]   .primary[_ngcontent-%COMP%]{background-color:#359;color:#fff}.flx-ui-datatable-main[_ngcontent-%COMP%]   .success[_ngcontent-%COMP%]{background-color:#5cb85c;color:#fff}.flx-ui-datatable-main[_ngcontent-%COMP%]   .default[_ngcontent-%COMP%]{background-color:#fff;color:#000}.flx-ui-datatable-main[_ngcontent-%COMP%]   .secondary[_ngcontent-%COMP%]{background-color:#1e88e5;color:#fff}.flx-ui-datatable-main[_ngcontent-%COMP%]   .btn-clear[_ngcontent-%COMP%]{border:none!important;box-shadow:none!important}.flx-ui-datatable-main[_ngcontent-%COMP%]   .input-group[_ngcontent-%COMP%], .flx-ui-datatable-main[_ngcontent-%COMP%]   .input-group[_ngcontent-%COMP%]   input[_ngcontent-%COMP%]{background-color:transparent!important}.flx-ui-datatable-main[_ngcontent-%COMP%]   .input-group-addon[_ngcontent-%COMP%]{border:none!important;padding:0!important;box-shadow:none!important;background-color:transparent!important}.flx-ui-datatable-main[_ngcontent-%COMP%]   .input-group-addon[_ngcontent-%COMP%]   button[_ngcontent-%COMP%]{border:1px solid transparent!important;box-shadow:none!important;border-top:none!important;border-bottom:none!important;background-color:transparent!important;border-radius:50em!important;color:#359;width:30px;height:30px}.flx-ui-datatable-main[_ngcontent-%COMP%]   .input-group-addon[_ngcontent-%COMP%]   button[_ngcontent-%COMP%]   i[_ngcontent-%COMP%]{font-size:18px}.flx-ui-datatable-main[_ngcontent-%COMP%]   .input-group-addon[_ngcontent-%COMP%]   button[_ngcontent-%COMP%]:disabled{background-color:transparent!important}.flx-ui-datatable-main[_ngcontent-%COMP%]   .input-group-addon[_ngcontent-%COMP%]   button[_ngcontent-%COMP%]:disabled   i[_ngcontent-%COMP%]{color:#f50057}.flx-ui-datatable-main[_ngcontent-%COMP%]   .data-scroll-content[_ngcontent-%COMP%]{max-width:100%;max-height:400px;margin-top:-1em}.flx-ui-datatable-main[_ngcontent-%COMP%]   .data-scroll-content[_ngcontent-%COMP%]   table[_ngcontent-%COMP%]   tbody[_ngcontent-%COMP%]   tr[_ngcontent-%COMP%]:first-child   td[_ngcontent-%COMP%], .flx-ui-datatable-main[_ngcontent-%COMP%]   .flx-table-thead[_ngcontent-%COMP%]   tr[_ngcontent-%COMP%]   th[_ngcontent-%COMP%]{border-top:none!important}"] });
/** @nocollapse */
FlxUiDatatableComponent.ctorParameters = () => [
    { type: FormBuilder },
    { type: FlxUiDatatableService }
];
FlxUiDatatableComponent.propDecorators = {
    classes: [{ type: Input }],
    headers: [{ type: Input }],
    lazyloadingConfig: [{ type: Input }],
    embedPictures: [{ type: Input }],
    dataKeys: [{ type: Input }],
    enableDataExports: [{ type: Input }],
    dataExportsConfig: [{ type: Input }],
    showBottomInfo: [{ type: Input }],
    searchKeys: [{ type: Input }],
    dataSrcKey: [{ type: Input }],
    hasActionButtons: [{ type: Input }],
    hideNumbers: [{ type: Input }],
    enableMultipleSelection: [{ type: Input }],
    multipleSelectKey: [{ type: Input }],
    hasAddButton: [{ type: Input }],
    dataUrl: [{ type: Input }],
    actionButtonStart: [{ type: Input }],
    multipleSelectButton: [{ type: Input }],
    searchPlaceholder: [{ type: Input }],
    actionHeader: [{ type: Input }],
    limit: [{ type: Input }],
    spinnerSrc: [{ type: Input }],
    actionButtons: [{ type: Input }],
    paginationButtons: [{ type: Input }],
    tableHeader: [{ type: Input }],
    searchButton: [{ type: Input }],
    addButton: [{ type: Input }],
    searchBar: [{ type: Input }],
    actionButtonsConfig: [{ type: Input }],
    firstActionButtonClicked: [{ type: Output }],
    secondActionButtonClicked: [{ type: Output }],
    thirdActionButtonClicked: [{ type: Output }],
    multipleSelectClicked: [{ type: Output }],
    addButtonClicked: [{ type: Output }],
    textSubstring: [{ type: Input }],
    conditionManipulator: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FlxUiDatatableComponent, [{
        type: Component,
        args: [{
                selector: 'flx-ui-datatable',
                template: `<div class="col-md-12 flx-ui-datatable-main {{ classes?.maincontainer }}">
    <div id="export_loading" class="col-md-12 text-center" style="display: none;margin-bottom:0.5em;color:#dddddd;font-size:13px;font-weight:bold;">Exporting...</div>
    <div class="col-md-12 flx-ui-datatable-header">
        <div class="container-fluid">
            <div class="row">
                <div class="col-xs-3 col-sm-3 pagination-select col-md-2" style="position:relative;z-index: 1;">
                    <select class="form-control rmsh rmrd {{ classes?.paginationselect }}" (change)="paginateDatatable($event?.target?.value)">
                        <option *ngFor="let paging of service?.pagination" [value]="paging?.value">{{ paging?.label }}</option>
                    </select>
                </div>
                <div class="col-xs-5 col-sm-5 col-md-2 text-center flx-datatable-pagination rmpd" style="position:relative;z-index: 2;">
                    <button mat-icon-button [ngClass]="{'flx-pagination-end': disablePrevtButton()}" (click)="nextPrevItem('prev')" [disabled]="disablePrevtButton()" class="flx-ui-datatable-pagination-buttons {{ classes?.paginationButton }}"><span class="fa fa-angle-double-left fa-1x"></span> <span class="flx-datatable-tooltip-prev" [ngClass]="{'flx-pagination-end': disablePrevtButton()}">Previous</span> </button>
                        <span *ngIf="!isSearching;else searching">{{ service?.dataOffset | ceil: limit }} / {{ service?.totalItems | ceil: limit }}</span>
                        <ng-template #searching>
                            
                        </ng-template>
                    <button mat-icon-button [ngClass]="{'flx-pagination-end': disableNextButton()}" (click)="nextPrevItem('next')" [disabled]="disableNextButton()" class="flx-ui-datatable-pagination-buttons {{ classes?.paginationButton }}"><span class="fa fa-angle-double-right fa-1x"></span> <span class="flx-datatable-tooltip-next" [ngClass]="{'flx-pagination-end': disableNextButton()}">Next</span></button>
                </div>
                <div [class]="'search-bar '+getSearchColumns()">
                    <input type="text" *ngIf="!isLazyloadingEnabled()" [style.background]="searchBar?.background" [style.color]="searchBar?.textColor" [ngStyle]="{border:searchBar?.borderSize +' solid '+ searchBar?.borderColor} " (keyup)="filterData($event?.target?.value)" class="form-control rmsh rmrd customclass" [placeholder]="searchPlaceholder">
                    <form (ngSubmit)="searchDataInApi(srch?.value,srch)" #srch="ngForm" *ngIf="isLazyloadingEnabled()">
                        <div class="input-group">
                            <input type="text" required name="searchString" ngModel [style.background]="searchBar?.background" [style.color]="searchBar?.textColor" [ngStyle]="{border:searchBar?.borderSize +' solid '+ searchBar?.borderColor} " class="form-control rmsh rmrd {{ classes?.searchbar }}" [placeholder]="searchPlaceholder">
                            <span class="input-group-addon">
                                <button [disabled]="!srch?.valid" type="submit" class="btn btn-default btn-clear btn-md">
                                    <i class="fa fa-search"></i>
                                </button>
                            </span>
                        </div>
                    </form>
                </div>
                <div class="col-md-1 text-right rmpd" *ngIf="hasAddButton">
                    <button (click)="addButtonClick()" class="{{ classes?.addButton }}" [style.background]="addButton?.background" [style.borderColor]="addButton?.background" [style.color]="addButton?.textColor"><span class="fa fa-plus"></span> Add</button>
                </div>
                <div class="col-md-1 text-right rmpd export-cnt" *ngIf="enableDataExports">
                    <div class="dropdown">
                        <button class="btn btn-default {{ classes?.exportButton }} dropdown-toggle" type="button" data-toggle="dropdown">
                            <i class="fa fa-angle-down"></i>
                        </button>
                        <ul class="dropdown-menu dropdown-menu-export {{ classes?.exportDropdown }}">
                            <li class="dropdown-header">{{ dataExportsConfig?.title }}. <input type="checkbox" (change)="checkToExportOption($event?.target?.checked)" style="position: relative;top:0.3em;"> <sup style="font-size:10px;color:#999;"> All</sup></li> 
                            <li class="divider"></li>
                            <li class="dropdown-submenu" *ngFor="let export of dataExportsConfig?.exportsTo" (click)="exportDocumentsAs(export)">
                                <a href="javascript:void(0)" class="btn btn-default btn-sm form-control" *ngIf="export=='print'"><i class="glyphicon glyphicon-print"></i> Print</a>
                                <a href="javascript:void(0)" class="btn btn-default btn-sm form-control" *ngIf="export=='pdf'" style="color:#ff0000"><i class="glyphicon glyphicon-file"></i> PDF</a>
                                <a href="javascript:void(0)" class="btn btn-default btn-sm form-control" *ngIf="export=='excel'" style="color:#009900;"><i class="glyphicon glyphicon-file"></i> Excel</a>            
                                <a href="javascript:void(0)" class="btn btn-default btn-sm form-control" *ngIf="export=='word'" style="color:#335599;"><i class="glyphicon glyphicon-file"></i> Word</a>                        
                            </li>
                            <li class="divider"></li>
                            <li class="dropdown-header">
                                <span  *ngIf="!isExportAll">{{ (displayData | async)?.length }}</span>
                                <span  *ngIf="isExportAll">{{ (service?.flxData | async)?.length }}</span> 
                            </li>
                        </ul>
                    </div>
                </div>
            </div>
        </div>
    </div>
    <div class="container-fluid">
        <div class="row">
            <div class="col-md-12 rmpd table-responsive flx-tabl-container">
                <table class="table {{ classes?.tableType }} th-table">
                    <thead class="{{ classes?.tableHeader }} flx-table-thead">
                        <tr>
                            <!-- Actions buttons header left before numbers -->
                            <th *ngIf="hasActionButtons && (actionButtonsConfig?.position=='left' && actionButtonsConfig?.beforeNumbers)">{{ actionHeader }} 
                                <input type="checkbox" [checked]="service?.multipleDeletion?.length>0" (change)="addRemove($event?.target?.checked)" *ngIf="enableMultipleSelection">
                                <button class="btn btn-danger btn-xs flx-multiple-deletion-button" *ngIf="enableMultipleSelection && service?.multipleDeletion?.length>0" (click)="confirmDelete()"><span [class]="multipleSelectButton?.icon"></span> {{ multipleSelectButton?.text }}</button>
                            </th>


                            <th *ngIf="!hideNumbers">N<sup><u>o</u></sup></th>

                            <!-- Actions buttons header left after numbers -->
                            <th *ngIf="hasActionButtons && (actionButtonsConfig?.position=='left' && !actionButtonsConfig?.beforeNumbers)">{{ actionHeader }} 
                                <input type="checkbox" [checked]="service?.multipleDeletion?.length>0" (change)="addRemove($event?.target?.checked)" *ngIf="enableMultipleSelection">
                                <button class="btn btn-danger btn-xs flx-multiple-deletion-button" *ngIf="enableMultipleSelection && service?.multipleDeletion?.length>0" (click)="confirmDelete()"><span [class]="multipleSelectButton?.icon"></span> {{ multipleSelectButton?.text }}</button>
                            </th>


                            <th *ngFor="let header of headers">{{ header }}</th>

                            <!-- Actions buttons header right -->
                            <th *ngIf="hasActionButtons  && (actionButtonsConfig?.position=='right')">{{ actionHeader }} 
                                <input type="checkbox" [checked]="service?.multipleDeletion?.length>0" (change)="addRemove($event?.target?.checked)" *ngIf="enableMultipleSelection">
                                <button class="btn btn-danger btn-xs flx-multiple-deletion-button" *ngIf="enableMultipleSelection && service?.multipleDeletion?.length>0" (click)="confirmDelete()"><span [class]="multipleSelectButton?.icon"></span> {{ multipleSelectButton?.text }}</button>
                            </th>
                        </tr>
                    </thead>
                </table>
                <perfect-scrollbar class="data-scroll-content {{ classes?.dataScrollContent }}">
                    <table class="table {{ classes?.tableType }} flx-table-data">                        
                        <tbody class="flxdatatablebodyscrollbar">
                            <tr *ngIf="!service?.loadFinish">                    
                                <td colspan="20" class="text-center">
                                    <img *ngIf="spinnerSrc" [class]="classes?.spinner" [src]="spinnerSrc" style="margin-top:1em;margin-bottom:1em;">
                                </td>
                            </tr>
                            <tr class="flxuidatatablerow" id="flxdatatable_singlerow" *ngFor="let data of displayData | async;let i=index">
                                
                                <!-- Buttons Left Before Numbers -->
                                <td class="table-buttons" *ngIf="hasActionButtons && (actionButtonsConfig?.position=='left' && actionButtonsConfig?.beforeNumbers)" scope="row">
                                    <span *ngFor="let aButton of actionButtons;let buttonIndex=index">
                                        <button type="button" (click)="actionButtonClicked(i,buttonIndex)" class="btn {{ aButton?.class }}">
                                            <div class="toltip" class="flx-tooltip" [ngClass]="{'flx-tooltip-left':aButton?.tooltipPosition=='left','flx-tooltip-bottom':aButton?.tooltipPosition=='bottom','flx-tooltip-right':aButton?.tooltipPosition=='right'}" *ngIf="aButton?.tooltip">{{ aButton?.tooltip }}</div>
                                            <span class="action-button-icon-left" [class]="aButton?.icon" *ngIf="!aButton?.iconPosition || aButton?.iconPosition!='right'"></span>
                                            <span class="button-text"> {{ aButton?.text }} </span>
                                            <span [class]="aButton?.icon" *ngIf="aButton?.iconPosition=='right'"></span>
                                        </button>
                                    </span> 
                                    <input type="checkbox" checked (change)="addRemoveDeleteItem(data[multipleSelectKey],i,$event?.target?.checked)" *ngIf="enableMultipleSelection && service?.multipleDeletion?.length>0">
                                </td>
                                
                                <!-- Numbers -->
                                <td class="{{ classes?.tableData }}" *ngIf="!hideNumbers" style="color: #999;">{{ offset+i }}</td>
                                
                                <!-- Buttons Left After Numbers -->
                                <td class="table-buttons" *ngIf="hasActionButtons && (actionButtonsConfig?.position=='left' && !actionButtonsConfig?.beforeNumbers)" scope="row">
                                    <span *ngFor="let aButton of actionButtons;let buttonIndex=index">
                                        <button type="button" (click)="actionButtonClicked(i,buttonIndex)" class="btn {{ aButton?.class }}">
                                            <div class="toltip" class="flx-tooltip" [ngClass]="{'flx-tooltip-left':aButton?.tooltipPosition=='left','flx-tooltip-bottom':aButton?.tooltipPosition=='bottom','flx-tooltip-right':aButton?.tooltipPosition=='right'}" *ngIf="aButton?.tooltip">{{ aButton?.tooltip }}</div>
                                            <span class="action-button-icon-left" [class]="aButton?.icon" *ngIf="!aButton?.iconPosition || aButton?.iconPosition!='right'"></span>
                                            <span class="button-text"> {{ aButton?.text }} </span>
                                            <span [class]="aButton?.icon" *ngIf="aButton?.iconPosition=='right'"></span>
                                        </button>
                                    </span> 
                                    <input type="checkbox" checked (change)="addRemoveDeleteItem(data[multipleSelectKey],i,$event?.target?.checked)" *ngIf="enableMultipleSelection && service?.multipleDeletion?.length>0">
                                </td>
                                
                                <!-- Main -->
                                <td class="{{ classes?.tableData }}" *ngFor="let dataKey of dataKeys;let x=index">
                                    <img *ngIf="hasImageEmbeded() && x==embedPictures?.index; else nopicture" [class]="'img-fall-back ' +embedPictures?.class" [src]="embedPictures?.rootFolder+data[dataKey]" [flx-ui-datatable-img-fallback]="embedPictures?.fallbackUrl" >
                                    <ng-template #nopicture>
                                        <section *ngIf="textSubstring?.atIndexes && (textSubstring?.atIndexes | haslimittext:x);else nolimittext">
                                            {{ textSubstring | limittext: data[dataKey] }}
                                        </section>
                                        <ng-template #nolimittext>
                                            <span>{{ data[dataKey] }}</span>
                                        </ng-template>
                                    </ng-template> 
                                </td>
                                <!-- Buttons -->
                                <td class="table-buttons" *ngIf="hasActionButtons && actionButtonsConfig?.position!='left'">
                                    <span *ngFor="let aButton of actionButtons;let buttonIndex=index">
                                        <button type="button" (click)="actionButtonClicked(i,buttonIndex)" class="btn {{ aButton?.class }}">
                                            <div class="toltip" class="flx-tooltip" [ngClass]="{'flx-tooltip-left':aButton?.tooltipPosition=='left','flx-tooltip-bottom':aButton?.tooltipPosition=='bottom','flx-tooltip-right':aButton?.tooltipPosition=='right'}" *ngIf="aButton?.tooltip">{{ aButton?.tooltip }}</div>
                                            <span class="action-button-icon-left" [class]="aButton?.icon" *ngIf="!aButton?.iconPosition || aButton?.iconPosition!='right'"></span>
                                            <span class="button-text"> {{ aButton?.text }} </span>
                                            <span [class]="aButton?.icon" *ngIf="aButton?.iconPosition=='right'"></span>
                                        </button>
                                    </span> 
                                    <input type="checkbox" checked (change)="addRemoveDeleteItem(data[multipleSelectKey],i,$event?.target?.checked)" *ngIf="enableMultipleSelection && service?.multipleDeletion?.length>0">
                                </td>
                            </tr>
                            <tr *ngIf="tData?.length<1">
                                <td colspan="10" class="text-center" *ngIf="service?.loadFinish">
                                    <span style="color:#ff0000;font-size:13px;">No data found</span> <br>
                                    <button type="button" style="margin-top:1em;" (click)="reload()" class="btn btn-default {{ classes?.reloadbutton }}" color="primary"><span class="fa fa-refresh"></span> Reload</button>
                                </td>
                            </tr>
                        </tbody>
                    </table>
                </perfect-scrollbar>
                <div class="row">
                    <div class="col-md-12 rmpd flx-total-data" *ngIf="showBottomInfo">
                        <div class="row" *ngIf="!isSearching;else bottomInfoSearching">
                            <div class="col-md-4 text-left rmpd">Total pagination: <span> {{ service?.totalItems | ceil: limit }}</span></div>
                            <div class="col-md-4 text-center rmpd"># of items per pagination: <span>{{ limit }}</span></div>
                            <div class="col-md-4 text-right rmpd">Total items: <span>{{ (service?.flxData | async)?.length }}</span></div>
                        </div>
                        <ng-template #bottomInfoSearching>
                            <div class="row">
                                <div class="col-md-12 text-right rmpd">Total search result: <span>{{ searchResultFound?.length }}</span></div>
                            </div>
                        </ng-template>
                    </div>
                </div>
            </div>
        </div>
    </div>
</div>`,
                styles: [`.flx-ui-datatable-main{background-color:#fff;padding-top:1em;padding-bottom:1em}.flx-ui-datatable-main .btn-danger{background-color:#f50057;border:1px solid #f50057;box-shadow:0 3px 5px 1px #ddd;-moz-box-shadow:0 3px 5px 1px #ddd;-webkit-box-shadow:0 3px 5px 1px #ddd;-o-box-shadow:0 3px 5px 1px #ddd;-ms-box-shadow:0 3px 5px 1px #ddd;border-radius:3px;margin-left:.3em}.flx-ui-datatable-main #flxdatatablebodyscrollbar,.flx-ui-datatable-main .table-responsive,.flx-ui-datatable-main .table-responsive table{width:100%}.flx-ui-datatable-main img.img-fall-back{width:30px;height:30px}.flx-ui-datatable-main .pagination-select input[type=text],.flx-ui-datatable-main .pagination-select select,.flx-ui-datatable-main .search-bar input[type=text],.flx-ui-datatable-main .search-bar select{border-top:none!important;border-left:none!important;border-right:none!important;border-bottom:2px solid #ddd!important;border-radius:0!important;box-shadow:0 0 0 0 transparent!important;-moz-box-shadow:0 0 0 0 transparent!important;-webkit-box-shadow:0 0 0 0 transparent!important;-o-box-shadow:0 0 0 0 transparent!important;-ms-box-shadow:0 0 0 0 transparent!important}.flx-ui-datatable-main .pagination-select input[type=text]:focus,.flx-ui-datatable-main .pagination-select select:focus,.flx-ui-datatable-main .search-bar input[type=text]:focus,.flx-ui-datatable-main .search-bar select:focus{border-bottom-color:#359!important;transition:.5s;-moz-transition:.5s;-webkit-transition:.5s;-o-transition:.5s;-ms-transition:.5s}.flx-ui-datatable-main .pagination-select select,.flx-ui-datatable-main .search-bar select{-webkit-appearance:none;appearance:none;-moz-appearance:none}.flx-ui-datatable-main .dropdown-toggle::after{display:none}.flx-ui-datatable-main .dropdown-menu-export{margin-left:-130px;padding-left:1em;padding-right:1em;width:100px!important;margin-top:1em;border:1px solid #ddd!important}.flx-ui-datatable-main .dropdown-menu-export::before{content:"";width:auto;height:auto;position:absolute;top:-20px;right:3px;border:10px solid;border-color:transparent transparent #ddd}.flx-ui-datatable-main .flx-datatable-pagination{padding-top:.5em}.flx-ui-datatable-main .flx-datatable-pagination button{width:35px!important;height:35px!important;border-radius:50em!important;border:none!important;box-shadow:0 3px 10px 0 #b3c4e6;-moz-box-shadow:0 3px 10px 0 #b3c4e6;-webkit-box-shadow:0 3px 10px 0 #b3c4e6;-o-box-shadow:0 3px 10px 0 #b3c4e6;-ms-box-shadow:0 3px 10px 0 #b3c4e6;background-color:#359;color:#fff;font-size:23px;font-weight:700;position:absolute;top:0}.flx-ui-datatable-main .flx-datatable-pagination button:first-child{left:0}.flx-ui-datatable-main .flx-datatable-pagination button:first-child .flx-datatable-tooltip-prev{position:absolute;left:0;font-size:13px;font-weight:400;color:#fff;background-color:#359;padding-left:.3em;padding-right:.3em;border-radius:3px;margin-left:-57px;margin-top:.3em;box-shadow:0 3px 10px 0 #b3c4e6!important;-moz-box-shadow:0 3px 10px 0 #b3c4e6!important;-webkit-box-shadow:0 3px 10px 0 #b3c4e6!important;-o-box-shadow:0 3px 10px 0 #b3c4e6!important;-ms-box-shadow:0 3px 10px 0 #b3c4e6!important;visibility:hidden}.flx-ui-datatable-main .flx-datatable-pagination button:hover>span.flx-datatable-tooltip-next,.flx-ui-datatable-main .flx-datatable-pagination button:hover>span.flx-datatable-tooltip-prev{visibility:visible}.flx-ui-datatable-main .flx-datatable-pagination .flx-pagination-end{background-color:#f50057!important;box-shadow:0 3px 10px 0 #ffc2d8!important;-moz-box-shadow:0 3px 10px 0 #ffc2d8!important;-webkit-box-shadow:0 3px 10px 0 #ffc2d8!important;-o-box-shadow:0 3px 10px 0 #ffc2d8!important;-ms-box-shadow:0 3px 10px 0 #ffc2d8!important;cursor:not-allowed}.flx-ui-datatable-main .flx-datatable-pagination .flx-pagination-end:hover>span.flx-datatable-tooltip-next,.flx-ui-datatable-main .flx-datatable-pagination .flx-pagination-end:hover>span.flx-datatable-tooltip-prev{visibility:hidden}.flx-ui-datatable-main .flx-datatable-pagination button:last-child{right:0}.flx-ui-datatable-main .flx-datatable-pagination button:last-child .flx-datatable-tooltip-next{position:absolute;left:0;font-size:13px;font-weight:400;color:#fff;background-color:#359;padding-left:.3em;padding-right:.3em;border-radius:3px;margin-left:35px;margin-top:.5em;box-shadow:0 3px 10px 0 #b3c4e6!important;-moz-box-shadow:0 3px 10px 0 #b3c4e6!important;-webkit-box-shadow:0 3px 10px 0 #b3c4e6!important;-o-box-shadow:0 3px 10px 0 #b3c4e6!important;-ms-box-shadow:0 3px 10px 0 #b3c4e6!important;visibility:hidden}.flx-ui-datatable-main .export-cnt button{border-radius:50em!important}.flx-ui-datatable-main .flx-tabl-container table.th-table{margin-top:.5em}.flx-ui-datatable-main table tbody tr{padding-top:0!important}.flx-ui-datatable-main table tbody tr td{padding-top:.5em;border-top:1px solid #f0f0f0;border-bottom:1px solid #f0f0f0}.flx-ui-datatable-main table tbody tr td button{margin-right:.3em;margin-left:0}.flx-ui-datatable-main table tbody tr td button div.flx-tooltip{position:absolute;background-color:rgba(32,27,27,.808);text-align:center;font-size:13px;color:#fff;border-radius:3px;box-shadow:0 3px 20px 0 #4b4949;-moz-box-shadow:0 3px 20px 0 #4b4949;-webkit-box-shadow:0 3px 20px 0 #4b4949;-o-box-shadow:0 3px 20px 0 #4b4949;-ms-box-shadow:0 3px 20px 0 #4b4949;margin-left:-2.5em;margin-top:-2.8em;visibility:hidden;width:80px;padding:.3em .5em}.flx-ui-datatable-main table tbody tr td button .flx-tooltip-left{margin-left:-95px!important;margin-top:-.3em!important}.flx-ui-datatable-main table tbody tr td button .flx-tooltip-bottom{margin-top:2.3em!important}.flx-ui-datatable-main table tbody tr td button .flx-tooltip-right{margin-left:28px!important;margin-top:-.3em!important}.flx-ui-datatable-main table tbody tr td button:hover>div.flx-tooltip{transition:.3s;visibility:visible}.flx-ui-datatable-main table tbody tr td.table-buttons{padding-top:.2em;padding-bottom:.2em}.flx-ui-datatable-main table tbody tr:nth-of-type(even){background-color:#f8f9fa}.flx-ui-datatable-main table tbody tr:nth-of-type(odd){background-color:#fff}.flx-ui-datatable-main .btn-danger:hover{background-color:#ff146b;border:1px solid #ff146b;box-shadow:0 3px 10px 1px #ff5fb6;-moz-box-shadow:0 3px 10px 1px #ff5fb6;-webkit-box-shadow:0 3px 10px 1px #ff5fb6;-o-box-shadow:0 3px 10px 1px #ff5fb6;-ms-box-shadow:0 3px 10px 1px #ff5fb6;transition:.5s;-moz-transition:.5s;-webkit-transition:.5s;-o-transition:.5s;-ms-transition:.5s}.flx-ui-datatable-main .btn-danger:focus{background-color:#f50057;border:1px solid #f50057}.flx-ui-datatable-main .btn-white{background-color:#fff}.flx-ui-datatable-main .btn-dark{background-color:#222!important}.flx-ui-datatable-main .btn-primary{background-color:#359;color:#fff;border:1px solid #359;box-shadow:0 3px 5px 1px #ddd;-moz-box-shadow:0 3px 5px 1px #ddd;-webkit-box-shadow:0 3px 5px 1px #ddd;-o-box-shadow:0 3px 5px 1px #ddd;-ms-box-shadow:0 3px 5px 1px #ddd;border-radius:3px}.flx-ui-datatable-main .btn-primary:hover{background-color:#4769ad;border:1px solid #4769ad;box-shadow:0 3px 10px 0 #b3c4e6;-moz-box-shadow:0 3px 10px 0 #b3c4e6;-webkit-box-shadow:0 3px 10px 0 #b3c4e6;-o-box-shadow:0 3px 10px 0 #b3c4e6;-ms-box-shadow:0 3px 10px 0 #b3c4e6;transition:.5s;-moz-transition:.5s;-webkit-transition:.5s;-o-transition:.5s;-ms-transition:.5s}.flx-ui-datatable-main .btn-primary:focus{background-color:#359;border:1px solid #359}.flx-ui-datatable-main .btn-large{padding-top:1em!important;padding-bottom:1em!important}.flx-ui-datatable-main .btn-medium{padding-top:.7em!important;padding-bottom:.7em!important}.flx-ui-datatable-main .btn-success{box-shadow:0 3px 5px 1px #ddd;-moz-box-shadow:0 3px 5px 1px #ddd;-webkit-box-shadow:0 3px 5px 1px #ddd;-o-box-shadow:0 3px 5px 1px #ddd;-ms-box-shadow:0 3px 5px 1px #ddd;border-radius:3px;background-color:#5cb85c;border:1px solid #5cb85c}.flx-ui-datatable-main .btn-success:hover{background-color:#70cc70;border:1px solid #70cc70;box-shadow:0 3px 10px 1px #9df99d;-moz-box-shadow:0 3px 10px 1px #9df99d;-webkit-box-shadow:0 3px 10px 1px #9df99d;-o-box-shadow:0 3px 10px 1px #9df99d;-ms-box-shadow:0 3px 10px 1px #9df99d;transition:.5s;-moz-transition:.5s;-webkit-transition:.5s;-o-transition:.5s;-ms-transition:.5s}.flx-ui-datatable-main .btn-success:focus{background-color:#5cb85c;border:1px solid #5cb85c}.flx-ui-datatable-main .btn-default{background-color:#fff;box-shadow:0 3px 5px 1px #eee;-moz-box-shadow:0 3px 5px 1px #eee;-webkit-box-shadow:0 3px 5px 1px #eee;-o-box-shadow:0 3px 5px 1px #eee;-ms-box-shadow:0 3px 5px 1px #eee;border-radius:3px;border:1px solid #ddd}.flx-ui-datatable-main .btn-default:hover{background-color:#fff;border:1px solid #e7e7e7;box-shadow:0 3px 10px 1px #e2e2e2;-moz-box-shadow:0 3px 10px 1px #e2e2e2;-webkit-box-shadow:0 3px 10px 1px #e2e2e2;-o-box-shadow:0 3px 10px 1px #e2e2e2;-ms-box-shadow:0 3px 10px 1px #e2e2e2;transition:.5s;-moz-transition:.5s;-webkit-transition:.5s;-o-transition:.5s;-ms-transition:.5s}.flx-ui-datatable-main .btn-default:focus{background-color:#fff;border:1px solid #ddd}.flx-ui-datatable-main .btn-secondary{box-shadow:0 3px 5px 1px #ddd;-moz-box-shadow:0 3px 5px 1px #ddd;-webkit-box-shadow:0 3px 5px 1px #ddd;-o-box-shadow:0 3px 5px 1px #ddd;-ms-box-shadow:0 3px 5px 1px #ddd;border-radius:3px;background-color:#1e88e5;color:#fff}.flx-ui-datatable-main .btn-secondary:hover{color:#fff;background-color:#2892ef;border:1px solid #2892ef;box-shadow:0 3px 10px 1px #55bfff;-moz-box-shadow:0 3px 10px 1px #55bfff;-webkit-box-shadow:0 3px 10px 1px #55bfff;-o-box-shadow:0 3px 10px 1px #55bfff;-ms-box-shadow:0 3px 10px 1px #55bfff;transition:.5s;-moz-transition:.5s;-webkit-transition:.5s;-o-transition:.5s;-ms-transition:.5s}.flx-ui-datatable-main .btn-secondary:focus{color:#fff}.flx-ui-datatable-main .pagination-button{background-color:#359;color:#fff}.flx-ui-datatable-main .table-font{font-family:Khula,sans-serif!important}.flx-ui-datatable-main .table-header-icon{position:absolute;right:.2em;width:80px;height:80px;font-size:50px;margin-top:-30px;border-radius:5px!important}.flx-ui-datatable-main .table-title{background-color:#359;color:#fff;border-radius:2px;padding:1em;font-size:15px;font-weight:700;margin-bottom:1.5em;font-family:Roboto,sans-serif;box-shadow:0 1px 5px 1px #ddd;-moz-box-shadow:0 1px 5px 1px #ddd;-webkit-box-shadow:0 1px 5px 1px #ddd;-o-box-shadow:0 1px 5px 1px #ddd;-ms-box-shadow:0 1px 5px 1px #ddd}.flx-ui-datatable-main .danger{background-color:#f50057;color:#fff}.flx-ui-datatable-main .primary{background-color:#359;color:#fff}.flx-ui-datatable-main .success{background-color:#5cb85c;color:#fff}.flx-ui-datatable-main .default{background-color:#fff;color:#000}.flx-ui-datatable-main .secondary{background-color:#1e88e5;color:#fff}.flx-ui-datatable-main .btn-clear{border:none!important;box-shadow:none!important}.flx-ui-datatable-main .input-group,.flx-ui-datatable-main .input-group input{background-color:transparent!important}.flx-ui-datatable-main .input-group-addon{border:none!important;padding:0!important;box-shadow:none!important;background-color:transparent!important}.flx-ui-datatable-main .input-group-addon button{border:1px solid transparent!important;box-shadow:none!important;border-top:none!important;border-bottom:none!important;background-color:transparent!important;border-radius:50em!important;color:#359;width:30px;height:30px}.flx-ui-datatable-main .input-group-addon button i{font-size:18px}.flx-ui-datatable-main .input-group-addon button:disabled{background-color:transparent!important}.flx-ui-datatable-main .input-group-addon button:disabled i{color:#f50057}.flx-ui-datatable-main .data-scroll-content{max-width:100%;max-height:400px;margin-top:-1em}.flx-ui-datatable-main .data-scroll-content table tbody tr:first-child td,.flx-ui-datatable-main .flx-table-thead tr th{border-top:none!important}`]
            }]
    }], function () { return [{ type: ɵngcc2.FormBuilder }, { type: FlxUiDatatableService }]; }, { classes: [{
            type: Input
        }], headers: [{
            type: Input
        }], lazyloadingConfig: [{
            type: Input
        }], embedPictures: [{
            type: Input
        }], dataKeys: [{
            type: Input
        }], enableDataExports: [{
            type: Input
        }], dataExportsConfig: [{
            type: Input
        }], searchKeys: [{
            type: Input
        }], dataSrcKey: [{
            type: Input
        }], hasActionButtons: [{
            type: Input
        }], hideNumbers: [{
            type: Input
        }], enableMultipleSelection: [{
            type: Input
        }], multipleSelectKey: [{
            type: Input
        }], hasAddButton: [{
            type: Input
        }], dataUrl: [{
            type: Input
        }], actionButtonStart: [{
            type: Input
        }], multipleSelectButton: [{
            type: Input
        }], searchPlaceholder: [{
            type: Input
        }], actionHeader: [{
            type: Input
        }], limit: [{
            type: Input
        }], spinnerSrc: [{
            type: Input
        }], actionButtons: [{
            type: Input
        }], paginationButtons: [{
            type: Input
        }], tableHeader: [{
            type: Input
        }], searchButton: [{
            type: Input
        }], addButton: [{
            type: Input
        }], searchBar: [{
            type: Input
        }], actionButtonsConfig: [{
            type: Input
        }], firstActionButtonClicked: [{
            type: Output
        }], secondActionButtonClicked: [{
            type: Output
        }], thirdActionButtonClicked: [{
            type: Output
        }], multipleSelectClicked: [{
            type: Output
        }], addButtonClicked: [{
            type: Output
        }], textSubstring: [{
            type: Input
        }], conditionManipulator: [{
            type: Input
        }], showBottomInfo: [{
            type: Input
        }] }); })();
class ImageFallBack {
    /**
     * @param {?} el
     */
    constructor(el) {
        this.isApplied = false;
        this.EVENT_TYPE = 'error';
        this.el = el.nativeElement;
        this.el.addEventListener(this.EVENT_TYPE, this.onError.bind(this));
    }
    /**
     * @return {?}
     */
    onError() {
        this.removeEvents();
        if (!this.isApplied) {
            this.isApplied = true;
            this.el.setAttribute('src', this.imgSrc);
        }
    }
    /**
     * @return {?}
     */
    removeEvents() {
        this.el.removeEventListener(this.EVENT_TYPE, this.onError);
    }
    /**
     * @return {?}
     */
    ngOnDestroy() {
        this.removeEvents();
    }
}
ImageFallBack.ɵfac = function ImageFallBack_Factory(t) { return new (t || ImageFallBack)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef)); };
ImageFallBack.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: ImageFallBack, selectors: [["", "flx-ui-datatable-img-fallback", ""]], inputs: { imgSrc: ["flx-ui-datatable-img-fallback", "imgSrc"] } });
/** @nocollapse */
ImageFallBack.ctorParameters = () => [
    { type: ElementRef }
];
ImageFallBack.propDecorators = {
    imgSrc: [{ type: Input, args: ['flx-ui-datatable-img-fallback',] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ImageFallBack, [{
        type: Directive,
        args: [{
                selector: '[flx-ui-datatable-img-fallback]'
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }]; }, { imgSrc: [{
            type: Input,
            args: ['flx-ui-datatable-img-fallback']
        }] }); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class Ceil {
    constructor() {
    }
    /**
     * @param {?} value
     * @param {?} limit
     * @return {?}
     */
    transform(value, limit) {
        return Math.ceil(value / limit);
    }
}
Ceil.ɵfac = function Ceil_Factory(t) { return new (t || Ceil)(); };
Ceil.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "ceil", type: Ceil, pure: true });
/** @nocollapse */
Ceil.ctorParameters = () => [];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(Ceil, [{
        type: Pipe,
        args: [{
                name: 'ceil'
            }]
    }], function () { return []; }, null); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class HasLimitText {
    /**
     * @param {?} service
     */
    constructor(service) {
        this.service = service;
    }
    /**
     * @param {?} array
     * @param {?} indexReached
     * @return {?}
     */
    transform(array, indexReached) {
        let /** @type {?} */ found;
        for (let /** @type {?} */ i of array) {
            if (i == indexReached) {
                found = true;
                break;
            }
        }
        return found;
    }
}
HasLimitText.ɵfac = function HasLimitText_Factory(t) { return new (t || HasLimitText)(ɵngcc0.ɵɵdirectiveInject(FlxUiDatatableService)); };
HasLimitText.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "haslimittext", type: HasLimitText, pure: true });
/** @nocollapse */
HasLimitText.ctorParameters = () => [
    { type: FlxUiDatatableService }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(HasLimitText, [{
        type: Pipe,
        args: [{
                name: 'haslimittext'
            }]
    }], function () { return [{ type: FlxUiDatatableService }]; }, null); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class LimitText {
    /**
     * @param {?} service
     */
    constructor(service) {
        this.service = service;
    }
    /**
     * @param {?} mainpulationConfig
     * @param {?} text
     * @return {?}
     */
    transform(mainpulationConfig, text) {
        let /** @type {?} */ from = (mainpulationConfig.hasOwnProperty("from")) ? mainpulationConfig['from'] : 0;
        let /** @type {?} */ to = (mainpulationConfig.hasOwnProperty("to")) ? mainpulationConfig["to"] : text.length;
        let /** @type {?} */ appendDots = (mainpulationConfig.hasOwnProperty("appendDots")) ? mainpulationConfig["appendDots"] : "";
        let /** @type {?} */ txt = "";
        if (text.length > to) {
            txt = text.substring(from, to) + appendDots;
        }
        else {
            txt = text;
        }
        return txt;
    }
}
LimitText.ɵfac = function LimitText_Factory(t) { return new (t || LimitText)(ɵngcc0.ɵɵdirectiveInject(FlxUiDatatableService)); };
LimitText.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "limittext", type: LimitText, pure: true });
/** @nocollapse */
LimitText.ctorParameters = () => [
    { type: FlxUiDatatableService }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(LimitText, [{
        type: Pipe,
        args: [{
                name: 'limittext'
            }]
    }], function () { return [{ type: FlxUiDatatableService }]; }, null); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
class ProcessedData {
    /**
     * @param {?} service
     */
    constructor(service) {
        this.service = service;
    }
    /**
     * @param {?} dataIndex
     * @return {?}
     */
    transform(dataIndex) {
        let /** @type {?} */ isProcessed = false;
        let /** @type {?} */ foundIndex = -1;
        for (let /** @type {?} */ i = 0; i < this.service.processedData.length; i++) {
            if (i == dataIndex) {
                isProcessed = true;
                foundIndex = i;
            }
        }
        if (foundIndex < 0) {
            this.service.processedData.push(dataIndex);
        }
        return isProcessed;
    }
}
ProcessedData.ɵfac = function ProcessedData_Factory(t) { return new (t || ProcessedData)(ɵngcc0.ɵɵdirectiveInject(FlxUiDatatableService)); };
ProcessedData.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "isprocessed", type: ProcessedData, pure: true });
/** @nocollapse */
ProcessedData.ctorParameters = () => [
    { type: FlxUiDatatableService }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ProcessedData, [{
        type: Pipe,
        args: [{
                name: 'isprocessed'
            }]
    }], function () { return [{ type: FlxUiDatatableService }]; }, null); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
const /** @type {?} */ DEFAULT_PERFECT_SCROLLBAR_CONFIG = {
    suppressScrollX: true
};
const ɵ0 = DEFAULT_PERFECT_SCROLLBAR_CONFIG;
class FlxUiDatatableModule {
}
FlxUiDatatableModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: FlxUiDatatableModule });
FlxUiDatatableModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function FlxUiDatatableModule_Factory(t) { return new (t || FlxUiDatatableModule)(); }, providers: [FlxUiDatatableService,
        {
            provide: PERFECT_SCROLLBAR_CONFIG,
            useValue: ɵ0
        }], imports: [[
            CommonModule, ReactiveFormsModule, FormsModule, HttpModule, PerfectScrollbarModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(FlxUiDatatableModule, { declarations: function () { return [FlxUiDatatableComponent,
        Ceil,
        ImageFallBack,
        HasLimitText,
        LimitText,
        ProcessedData]; }, imports: function () { return [CommonModule, ReactiveFormsModule, FormsModule, HttpModule, PerfectScrollbarModule]; }, exports: function () { return [FlxUiDatatableComponent,
        Ceil]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FlxUiDatatableModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule, ReactiveFormsModule, FormsModule, HttpModule, PerfectScrollbarModule
                ],
                declarations: [
                    FlxUiDatatableComponent, Ceil, ImageFallBack, HasLimitText, LimitText, ProcessedData
                ],
                exports: [FlxUiDatatableComponent, Ceil],
                providers: [FlxUiDatatableService,
                    {
                        provide: PERFECT_SCROLLBAR_CONFIG,
                        useValue: ɵ0
                    }]
            }]
    }], null, null); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */

export { FlxUiDataTable, FlxUiDatatableModule, FlxUiDatatableComponent as ɵb, ImageFallBack as ɵc, FlxUiDatatableService as ɵa, Ceil as ɵd, HasLimitText as ɵe, LimitText as ɵf, ProcessedData as ɵg };
