// devextreme-angular/esm5/ui/nested/context-toolbox.js
/*!
 * devextreme-angular
 * Version: 20.1.3
 * Build date: Fri Apr 24 2020
 *
 * Copyright (c) 2012 - 2020 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-angular
 */
import * as tslib_1 from "tslib";
/* tslint:disable:max-line-length */
import { Component, NgModule, Host, SkipSelf, Input } from '@angular/core';
import { NestedOptionHost, } from 'devextreme-angular/core';
import { NestedOption } from 'devextreme-angular/core';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from 'devextreme-angular/core';
var DxoContextToolboxComponent = /** @class */ (function (_super) {
    tslib_1.__extends(DxoContextToolboxComponent, _super);
    function DxoContextToolboxComponent(parentOptionHost, optionHost) {
        var _this = _super.call(this) || this;
        parentOptionHost.setNestedOption(_this);
        optionHost.setHost(_this, _this._fullOptionPath.bind(_this));
        return _this;
    }
    Object.defineProperty(DxoContextToolboxComponent.prototype, "category", {
        get: function () {
            return this._getOption('category');
        },
        set: function (value) {
            this._setOption('category', value);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(DxoContextToolboxComponent.prototype, "displayMode", {
        get: function () {
            return this._getOption('displayMode');
        },
        set: function (value) {
            this._setOption('displayMode', value);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(DxoContextToolboxComponent.prototype, "enabled", {
        get: function () {
            return this._getOption('enabled');
        },
        set: function (value) {
            this._setOption('enabled', value);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(DxoContextToolboxComponent.prototype, "shapes", {
        get: function () {
            return this._getOption('shapes');
        },
        set: function (value) {
            this._setOption('shapes', value);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(DxoContextToolboxComponent.prototype, "_optionPath", {
        get: function () {
            return 'contextToolbox';
        },
        enumerable: true,
        configurable: true
    });
    DxoContextToolboxComponent.prototype.ngOnInit = function () {
        this._addRecreatedComponent();
    };
    DxoContextToolboxComponent.prototype.ngOnDestroy = function () {
        this._addRemovedOption(this._getOptionPath());
    };
    DxoContextToolboxComponent.ctorParameters = function () { return [
        { type: NestedOptionHost, decorators: [{ type: SkipSelf }, { type: Host }] },
        { type: NestedOptionHost, decorators: [{ type: Host }] }
    ]; };
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", String),
        tslib_1.__metadata("design:paramtypes", [String])
    ], DxoContextToolboxComponent.prototype, "category", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", String),
        tslib_1.__metadata("design:paramtypes", [String])
    ], DxoContextToolboxComponent.prototype, "displayMode", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Boolean),
        tslib_1.__metadata("design:paramtypes", [Boolean])
    ], DxoContextToolboxComponent.prototype, "enabled", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Array),
        tslib_1.__metadata("design:paramtypes", [Array])
    ], DxoContextToolboxComponent.prototype, "shapes", null);
    DxoContextToolboxComponent = tslib_1.__decorate([ tslib_1.__param(0, SkipSelf()), tslib_1.__param(0, Host()),
        tslib_1.__param(1, Host()),
        tslib_1.__metadata("design:paramtypes", [NestedOptionHost,
            NestedOptionHost])
    ], DxoContextToolboxComponent);
DxoContextToolboxComponent.ɵfac = function DxoContextToolboxComponent_Factory(t) { return new (t || DxoContextToolboxComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.NestedOptionHost, 5), ɵngcc0.ɵɵdirectiveInject(ɵngcc1.NestedOptionHost, 1)); };
DxoContextToolboxComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: DxoContextToolboxComponent, selectors: [["dxo-context-toolbox"]], inputs: { category: "category", displayMode: "displayMode", enabled: "enabled", shapes: "shapes" }, features: [ɵngcc0.ɵɵProvidersFeature([NestedOptionHost]), ɵngcc0.ɵɵInheritDefinitionFeature], decls: 0, vars: 0, template: function DxoContextToolboxComponent_Template(rf, ctx) { }, styles: [""] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DxoContextToolboxComponent, [{
        type: Component,
        args: [{
                selector: 'dxo-context-toolbox',
                template: '',
                providers: [NestedOptionHost],
                styles: ['']
            }]
    }], function () { return [{ type: ɵngcc1.NestedOptionHost, decorators: [{
                type: SkipSelf
            }, {
                type: Host
            }] }, { type: ɵngcc1.NestedOptionHost, decorators: [{
                type: Host
            }] }]; }, { category: [{
            type: Input
        }], displayMode: [{
            type: Input
        }], enabled: [{
            type: Input
        }], shapes: [{
            type: Input
        }] }); })();
    return DxoContextToolboxComponent;
}(NestedOption));
export { DxoContextToolboxComponent };
var DxoContextToolboxModule = /** @class */ (function () {
    function DxoContextToolboxModule() {
    }
DxoContextToolboxModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: DxoContextToolboxModule });
DxoContextToolboxModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function DxoContextToolboxModule_Factory(t) { return new (t || DxoContextToolboxModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(DxoContextToolboxModule, { declarations: [DxoContextToolboxComponent], exports: [DxoContextToolboxComponent] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DxoContextToolboxModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    DxoContextToolboxComponent
                ],
                exports: [
                    DxoContextToolboxComponent
                ]
            }]
    }], function () { return []; }, null); })();
    return DxoContextToolboxModule;
}());
export { DxoContextToolboxModule };
