// devextreme-angular/esm5/ui/nested/history-toolbar.js
/*!
 * devextreme-angular
 * Version: 20.1.3
 * Build date: Fri Apr 24 2020
 *
 * Copyright (c) 2012 - 2020 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-angular
 */
import * as tslib_1 from "tslib";
/* tslint:disable:max-line-length */
import { Component, NgModule, Host, SkipSelf, Input, ContentChildren, forwardRef, QueryList } from '@angular/core';
import { NestedOptionHost, } from 'devextreme-angular/core';
import { NestedOption } from 'devextreme-angular/core';
import { DxiCommandComponent } from './command-dxi';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from 'devextreme-angular/core';
var DxoHistoryToolbarComponent = /** @class */ (function (_super) {
    tslib_1.__extends(DxoHistoryToolbarComponent, _super);
    function DxoHistoryToolbarComponent(parentOptionHost, optionHost) {
        var _this = _super.call(this) || this;
        parentOptionHost.setNestedOption(_this);
        optionHost.setHost(_this, _this._fullOptionPath.bind(_this));
        return _this;
    }
    Object.defineProperty(DxoHistoryToolbarComponent.prototype, "commands", {
        get: function () {
            return this._getOption('commands');
        },
        set: function (value) {
            this._setOption('commands', value);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(DxoHistoryToolbarComponent.prototype, "visible", {
        get: function () {
            return this._getOption('visible');
        },
        set: function (value) {
            this._setOption('visible', value);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(DxoHistoryToolbarComponent.prototype, "_optionPath", {
        get: function () {
            return 'historyToolbar';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(DxoHistoryToolbarComponent.prototype, "commandsChildren", {
        get: function () {
            return this._getOption('commands');
        },
        set: function (value) {
            this.setChildren('commands', value);
        },
        enumerable: true,
        configurable: true
    });
    DxoHistoryToolbarComponent.prototype.ngOnInit = function () {
        this._addRecreatedComponent();
    };
    DxoHistoryToolbarComponent.prototype.ngOnDestroy = function () {
        this._addRemovedOption(this._getOptionPath());
    };
    DxoHistoryToolbarComponent.ctorParameters = function () { return [
        { type: NestedOptionHost, decorators: [{ type: SkipSelf }, { type: Host }] },
        { type: NestedOptionHost, decorators: [{ type: Host }] }
    ]; };
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Array),
        tslib_1.__metadata("design:paramtypes", [Array])
    ], DxoHistoryToolbarComponent.prototype, "commands", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Boolean),
        tslib_1.__metadata("design:paramtypes", [Boolean])
    ], DxoHistoryToolbarComponent.prototype, "visible", null);
    tslib_1.__decorate([
        ContentChildren(forwardRef(function () { return DxiCommandComponent; })),
        tslib_1.__metadata("design:type", QueryList),
        tslib_1.__metadata("design:paramtypes", [Object])
    ], DxoHistoryToolbarComponent.prototype, "commandsChildren", null);
    DxoHistoryToolbarComponent = tslib_1.__decorate([ tslib_1.__param(0, SkipSelf()), tslib_1.__param(0, Host()),
        tslib_1.__param(1, Host()),
        tslib_1.__metadata("design:paramtypes", [NestedOptionHost,
            NestedOptionHost])
    ], DxoHistoryToolbarComponent);
DxoHistoryToolbarComponent.ɵfac = function DxoHistoryToolbarComponent_Factory(t) { return new (t || DxoHistoryToolbarComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.NestedOptionHost, 5), ɵngcc0.ɵɵdirectiveInject(ɵngcc1.NestedOptionHost, 1)); };
DxoHistoryToolbarComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: DxoHistoryToolbarComponent, selectors: [["dxo-history-toolbar"]], contentQueries: function DxoHistoryToolbarComponent_ContentQueries(rf, ctx, dirIndex) { if (rf & 1) {
        ɵngcc0.ɵɵcontentQuery(dirIndex, DxiCommandComponent, false);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.commandsChildren = _t);
    } }, inputs: { commands: "commands", visible: "visible" }, features: [ɵngcc0.ɵɵProvidersFeature([NestedOptionHost]), ɵngcc0.ɵɵInheritDefinitionFeature], decls: 0, vars: 0, template: function DxoHistoryToolbarComponent_Template(rf, ctx) { }, styles: [""] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DxoHistoryToolbarComponent, [{
        type: Component,
        args: [{
                selector: 'dxo-history-toolbar',
                template: '',
                providers: [NestedOptionHost],
                styles: ['']
            }]
    }], function () { return [{ type: ɵngcc1.NestedOptionHost, decorators: [{
                type: SkipSelf
            }, {
                type: Host
            }] }, { type: ɵngcc1.NestedOptionHost, decorators: [{
                type: Host
            }] }]; }, { commands: [{
            type: Input
        }], visible: [{
            type: Input
        }], commandsChildren: [{
            type: ContentChildren,
            args: [forwardRef(function () { return DxiCommandComponent; })]
        }] }); })();
    return DxoHistoryToolbarComponent;
}(NestedOption));
export { DxoHistoryToolbarComponent };
var DxoHistoryToolbarModule = /** @class */ (function () {
    function DxoHistoryToolbarModule() {
    }
DxoHistoryToolbarModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: DxoHistoryToolbarModule });
DxoHistoryToolbarModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function DxoHistoryToolbarModule_Factory(t) { return new (t || DxoHistoryToolbarModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(DxoHistoryToolbarModule, { declarations: [DxoHistoryToolbarComponent], exports: [DxoHistoryToolbarComponent] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DxoHistoryToolbarModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    DxoHistoryToolbarComponent
                ],
                exports: [
                    DxoHistoryToolbarComponent
                ]
            }]
    }], function () { return []; }, null); })();
    return DxoHistoryToolbarModule;
}());
export { DxoHistoryToolbarModule };
