// devextreme-angular/esm5/ui/nested/validation.js
/*!
 * devextreme-angular
 * Version: 20.1.3
 * Build date: Fri Apr 24 2020
 *
 * Copyright (c) 2012 - 2020 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-angular
 */
import * as tslib_1 from "tslib";
/* tslint:disable:max-line-length */
import { Component, NgModule, Host, SkipSelf, Input } from '@angular/core';
import { NestedOptionHost, } from 'devextreme-angular/core';
import { NestedOption } from 'devextreme-angular/core';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from 'devextreme-angular/core';
var DxoValidationComponent = /** @class */ (function (_super) {
    tslib_1.__extends(DxoValidationComponent, _super);
    function DxoValidationComponent(parentOptionHost, optionHost) {
        var _this = _super.call(this) || this;
        parentOptionHost.setNestedOption(_this);
        optionHost.setHost(_this, _this._fullOptionPath.bind(_this));
        return _this;
    }
    Object.defineProperty(DxoValidationComponent.prototype, "autoUpdateParentTasks", {
        get: function () {
            return this._getOption('autoUpdateParentTasks');
        },
        set: function (value) {
            this._setOption('autoUpdateParentTasks', value);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(DxoValidationComponent.prototype, "validateDependencies", {
        get: function () {
            return this._getOption('validateDependencies');
        },
        set: function (value) {
            this._setOption('validateDependencies', value);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(DxoValidationComponent.prototype, "_optionPath", {
        get: function () {
            return 'validation';
        },
        enumerable: true,
        configurable: true
    });
    DxoValidationComponent.prototype.ngOnInit = function () {
        this._addRecreatedComponent();
    };
    DxoValidationComponent.prototype.ngOnDestroy = function () {
        this._addRemovedOption(this._getOptionPath());
    };
    DxoValidationComponent.ctorParameters = function () { return [
        { type: NestedOptionHost, decorators: [{ type: SkipSelf }, { type: Host }] },
        { type: NestedOptionHost, decorators: [{ type: Host }] }
    ]; };
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Boolean),
        tslib_1.__metadata("design:paramtypes", [Boolean])
    ], DxoValidationComponent.prototype, "autoUpdateParentTasks", null);
    tslib_1.__decorate([
        Input(),
        tslib_1.__metadata("design:type", Boolean),
        tslib_1.__metadata("design:paramtypes", [Boolean])
    ], DxoValidationComponent.prototype, "validateDependencies", null);
    DxoValidationComponent = tslib_1.__decorate([ tslib_1.__param(0, SkipSelf()), tslib_1.__param(0, Host()),
        tslib_1.__param(1, Host()),
        tslib_1.__metadata("design:paramtypes", [NestedOptionHost,
            NestedOptionHost])
    ], DxoValidationComponent);
DxoValidationComponent.ɵfac = function DxoValidationComponent_Factory(t) { return new (t || DxoValidationComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.NestedOptionHost, 5), ɵngcc0.ɵɵdirectiveInject(ɵngcc1.NestedOptionHost, 1)); };
DxoValidationComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: DxoValidationComponent, selectors: [["dxo-validation"]], inputs: { autoUpdateParentTasks: "autoUpdateParentTasks", validateDependencies: "validateDependencies" }, features: [ɵngcc0.ɵɵProvidersFeature([NestedOptionHost]), ɵngcc0.ɵɵInheritDefinitionFeature], decls: 0, vars: 0, template: function DxoValidationComponent_Template(rf, ctx) { }, styles: [""] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DxoValidationComponent, [{
        type: Component,
        args: [{
                selector: 'dxo-validation',
                template: '',
                providers: [NestedOptionHost],
                styles: ['']
            }]
    }], function () { return [{ type: ɵngcc1.NestedOptionHost, decorators: [{
                type: SkipSelf
            }, {
                type: Host
            }] }, { type: ɵngcc1.NestedOptionHost, decorators: [{
                type: Host
            }] }]; }, { autoUpdateParentTasks: [{
            type: Input
        }], validateDependencies: [{
            type: Input
        }] }); })();
    return DxoValidationComponent;
}(NestedOption));
export { DxoValidationComponent };
var DxoValidationModule = /** @class */ (function () {
    function DxoValidationModule() {
    }
DxoValidationModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: DxoValidationModule });
DxoValidationModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function DxoValidationModule_Factory(t) { return new (t || DxoValidationModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(DxoValidationModule, { declarations: [DxoValidationComponent], exports: [DxoValidationComponent] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DxoValidationModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    DxoValidationComponent
                ],
                exports: [
                    DxoValidationComponent
                ]
            }]
    }], function () { return []; }, null); })();
    return DxoValidationModule;
}());
export { DxoValidationModule };
