// flx-ui-datatable/esm2015/lib/pipes/hasLimitText.pipe.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Pipe } from '@angular/core';
import { FlxUiDatatableService } from '../flx-ui-datatable.service';
import * as ɵngcc0 from '@angular/core';
export class HasLimitText {
    /**
     * @param {?} service
     */
    constructor(service) {
        this.service = service;
    }
    /**
     * @param {?} array
     * @param {?} indexReached
     * @return {?}
     */
    transform(array, indexReached) {
        let /** @type {?} */ found;
        for (let /** @type {?} */ i of array) {
            if (i == indexReached) {
                found = true;
                break;
            }
        }
        return found;
    }
}
HasLimitText.ɵfac = function HasLimitText_Factory(t) { return new (t || HasLimitText)(ɵngcc0.ɵɵdirectiveInject(FlxUiDatatableService)); };
HasLimitText.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "haslimittext", type: HasLimitText, pure: true });
/** @nocollapse */
HasLimitText.ctorParameters = () => [
    { type: FlxUiDatatableService }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(HasLimitText, [{
        type: Pipe,
        args: [{
                name: 'haslimittext'
            }]
    }], function () { return [{ type: FlxUiDatatableService }]; }, null); })();
function HasLimitText_tsickle_Closure_declarations() {
    /** @type {?} */
    HasLimitText.prototype.service;
}
