// flx-ui-datatable/esm2015/lib/pipes/limitText.pipe.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Pipe } from '@angular/core';
import { FlxUiDatatableService } from '../flx-ui-datatable.service';
import * as ɵngcc0 from '@angular/core';
export class LimitText {
    /**
     * @param {?} service
     */
    constructor(service) {
        this.service = service;
    }
    /**
     * @param {?} mainpulationConfig
     * @param {?} text
     * @return {?}
     */
    transform(mainpulationConfig, text) {
        let /** @type {?} */ from = (mainpulationConfig.hasOwnProperty("from")) ? mainpulationConfig['from'] : 0;
        let /** @type {?} */ to = (mainpulationConfig.hasOwnProperty("to")) ? mainpulationConfig["to"] : text.length;
        let /** @type {?} */ appendDots = (mainpulationConfig.hasOwnProperty("appendDots")) ? mainpulationConfig["appendDots"] : "";
        let /** @type {?} */ txt = "";
        if (text.length > to) {
            txt = text.substring(from, to) + appendDots;
        }
        else {
            txt = text;
        }
        return txt;
    }
}
LimitText.ɵfac = function LimitText_Factory(t) { return new (t || LimitText)(ɵngcc0.ɵɵdirectiveInject(FlxUiDatatableService)); };
LimitText.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "limittext", type: LimitText, pure: true });
/** @nocollapse */
LimitText.ctorParameters = () => [
    { type: FlxUiDatatableService }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(LimitText, [{
        type: Pipe,
        args: [{
                name: 'limittext'
            }]
    }], function () { return [{ type: FlxUiDatatableService }]; }, null); })();
function LimitText_tsickle_Closure_declarations() {
    /** @type {?} */
    LimitText.prototype.service;
}
