// flx-ui-datatable/esm2015/lib/pipes/processeddata.pipe.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Pipe } from '@angular/core';
import { FlxUiDatatableService } from '../flx-ui-datatable.service';
import * as ɵngcc0 from '@angular/core';
export class ProcessedData {
    /**
     * @param {?} service
     */
    constructor(service) {
        this.service = service;
    }
    /**
     * @param {?} dataIndex
     * @return {?}
     */
    transform(dataIndex) {
        let /** @type {?} */ isProcessed = false;
        let /** @type {?} */ foundIndex = -1;
        for (let /** @type {?} */ i = 0; i < this.service.processedData.length; i++) {
            if (i == dataIndex) {
                isProcessed = true;
                foundIndex = i;
            }
        }
        if (foundIndex < 0) {
            this.service.processedData.push(dataIndex);
        }
        return isProcessed;
    }
}
ProcessedData.ɵfac = function ProcessedData_Factory(t) { return new (t || ProcessedData)(ɵngcc0.ɵɵdirectiveInject(FlxUiDatatableService)); };
ProcessedData.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "isprocessed", type: ProcessedData, pure: true });
/** @nocollapse */
ProcessedData.ctorParameters = () => [
    { type: FlxUiDatatableService }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ProcessedData, [{
        type: Pipe,
        args: [{
                name: 'isprocessed'
            }]
    }], function () { return [{ type: FlxUiDatatableService }]; }, null); })();
function ProcessedData_tsickle_Closure_declarations() {
    /** @type {?} */
    ProcessedData.prototype.service;
}
