// pdfmake-wrapper/lib/definitions/table/table.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
import { StyleDefinition } from '../style-definition';
var Table = (function (_super) {
    __extends(Table, _super);
    function Table(_body) {
        var _this = _super.call(this) || this;
        _this._body = _body;
        _this.content.table = {};
        _this.content.table.body = _this._body;
        return _this;
    }
    Table.prototype.layout = function (layout) {
        this.content.layout = layout;
        return this;
    };
    Table.prototype.headerRows = function (rows) {
        this.content.table.headerRows = rows;
        return this;
    };
    Table.prototype.widths = function (widths) {
        this.content.table.widths = widths;
        return this;
    };
    Table.prototype.heights = function (heights) {
        this.content.table.heights = heights;
        return this;
    };
    Table.prototype.dontBreakRows = function (apply) {
        this.content.table.dontBreakRows = apply;
        return this;
    };
    Table.prototype.keepWithHeaderRows = function (rows) {
        this.content.table.keepWithHeaderRows = rows;
        return this;
    };
    return Table;
}(StyleDefinition));
export { Table };
