// devextreme-angular/bundles/devextreme-angular-ui-data-grid.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(require('@angular/core'),require('devextreme-angular/core'),require('@angular/platform-browser'),require('devextreme-angular/ui/nested'),exports, require('@angular/platform-browser'), require('@angular/core'), require('devextreme/bundles/dx.all'), require('devextreme/ui/data_grid'), require('devextreme-angular/core'), require('devextreme-angular/ui/nested')) :
    typeof define === 'function' && define.amd ? define('devextreme-angular/ui/data-grid', ['@angular/core','devextreme-angular/core','@angular/platform-browser','devextreme-angular/ui/nested','exports', '@angular/platform-browser', '@angular/core', 'devextreme/bundles/dx.all', 'devextreme/ui/data_grid', 'devextreme-angular/core', 'devextreme-angular/ui/nested'], factory) :
    (global = global || self, factory(global.ng.core,global.devextremeAngular.core,global.ng.platformBrowser,global.devextremeAngular.ui.nested,(global['devextreme-angular'] = global['devextreme-angular'] || {}, global['devextreme-angular'].ui = global['devextreme-angular'].ui || {}, global['devextreme-angular'].ui['data-grid'] = {}), global.ng.platformBrowser, global.ng.core, null, global.DxDataGrid, global['devextreme-angular'].core, global['devextreme-angular'].ui.nested));
}(this, (function (ɵngcc0,ɵngcc1,ɵngcc2,ɵngcc3,exports, platformBrowser, core, dx_all, DxDataGrid, core$1, nested) { 'use strict';

    DxDataGrid = DxDataGrid && Object.prototype.hasOwnProperty.call(DxDataGrid, 'default') ? DxDataGrid['default'] : DxDataGrid;

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation. All rights reserved.
    Licensed under the Apache License, Version 2.0 (the "License"); you may not use
    this file except in compliance with the License. You may obtain a copy of the
    License at http://www.apache.org/licenses/LICENSE-2.0

    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
    KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
    WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
    MERCHANTABLITY OR NON-INFRINGEMENT.

    See the Apache Version 2.0 License for specific language governing permissions
    and limitations under the License.
    ***************************************************************************** */
    /* global Reflect, Promise */

    var extendStatics = function(d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };

    function __extends(d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }

    var __assign = function() {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };

    function __rest(s, e) {
        var t = {};
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
            t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }

    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
        else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }

    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); }
    }

    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(metadataKey, metadataValue);
    }

    function __awaiter(thisArg, _arguments, P, generator) {
        function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
            function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
            function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }

    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f) throw new TypeError("Generator is already executing.");
            while (_) try {
                if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
                if (y = 0, t) op = [op[0] & 2, t.value];
                switch (op[0]) {
                    case 0: case 1: t = op; break;
                    case 4: _.label++; return { value: op[1], done: false };
                    case 5: _.label++; y = op[1]; op = [0]; continue;
                    case 7: op = _.ops.pop(); _.trys.pop(); continue;
                    default:
                        if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                        if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                        if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                        if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                        if (t[2]) _.ops.pop();
                        _.trys.pop(); continue;
                }
                op = body.call(thisArg, _);
            } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
            if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
        }
    }

    function __exportStar(m, exports) {
        for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
    }

    function __values(o) {
        var s = typeof Symbol === "function" && Symbol.iterator, m = s && o[s], i = 0;
        if (m) return m.call(o);
        if (o && typeof o.length === "number") return {
            next: function () {
                if (o && i >= o.length) o = void 0;
                return { value: o && o[i++], done: !o };
            }
        };
        throw new TypeError(s ? "Object is not iterable." : "Symbol.iterator is not defined.");
    }

    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m) return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done) ar.push(r.value);
        }
        catch (error) { e = { error: error }; }
        finally {
            try {
                if (r && !r.done && (m = i["return"])) m.call(i);
            }
            finally { if (e) throw e.error; }
        }
        return ar;
    }

    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }

    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    };

    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }

    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator) throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n]) i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try { step(g[n](v)); } catch (e) { settle(q[0][3], e); } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length) resume(q[0][0], q[0][1]); }
    }

    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }

    function __asyncValues(o) {
        if (!Symbol.asyncIterator) throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function(v) { resolve({ value: v, done: d }); }, reject); }
    }

    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
        return cooked;
    };

    function __importStar(mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
        result.default = mod;
        return result;
    }

    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }

    function __classPrivateFieldGet(receiver, privateMap) {
        if (!privateMap.has(receiver)) {
            throw new TypeError("attempted to get private field on non-instance");
        }
        return privateMap.get(receiver);
    }

    function __classPrivateFieldSet(receiver, privateMap, value) {
        if (!privateMap.has(receiver)) {
            throw new TypeError("attempted to set private field on non-instance");
        }
        privateMap.set(receiver, value);
        return value;
    }

    /*!
     * devextreme-angular
     * Version: 20.1.3
     * Build date: Fri Apr 24 2020
     *
     * Copyright (c) 2012 - 2020 Developer Express Inc. ALL RIGHTS RESERVED
     *
     * This software may be modified and distributed under the terms
     * of the MIT license. See the LICENSE file in the root of the project for details.
     *
     * https://github.com/DevExpress/devextreme-angular
     */
    /**
     * The DataGrid is a widget that represents data from a local or remote source in the form of a grid. This widget offers such basic features as sorting, grouping, filtering, as well as more advanced capabilities, like state storing, client-side exporting, master-detail interface, and many others.
     */
    var DxDataGridComponent = /** @class */ (function (_super) {
        __extends(DxDataGridComponent, _super);
        function DxDataGridComponent(elementRef, ngZone, templateHost, _watcherHelper, _idh, optionHost, transferState, platformId) {
            var _this = _super.call(this, elementRef, ngZone, templateHost, _watcherHelper, transferState, platformId) || this;
            _this._watcherHelper = _watcherHelper;
            _this._idh = _idh;
            _this._createEventEmitters([
                { subscribe: 'adaptiveDetailRowPreparing', emit: 'onAdaptiveDetailRowPreparing' },
                { subscribe: 'cellClick', emit: 'onCellClick' },
                { subscribe: 'cellDblClick', emit: 'onCellDblClick' },
                { subscribe: 'cellHoverChanged', emit: 'onCellHoverChanged' },
                { subscribe: 'cellPrepared', emit: 'onCellPrepared' },
                { subscribe: 'contentReady', emit: 'onContentReady' },
                { subscribe: 'contextMenuPreparing', emit: 'onContextMenuPreparing' },
                { subscribe: 'dataErrorOccurred', emit: 'onDataErrorOccurred' },
                { subscribe: 'disposing', emit: 'onDisposing' },
                { subscribe: 'editingStart', emit: 'onEditingStart' },
                { subscribe: 'editorPrepared', emit: 'onEditorPrepared' },
                { subscribe: 'editorPreparing', emit: 'onEditorPreparing' },
                { subscribe: 'exported', emit: 'onExported' },
                { subscribe: 'exporting', emit: 'onExporting' },
                { subscribe: 'fileSaving', emit: 'onFileSaving' },
                { subscribe: 'focusedCellChanged', emit: 'onFocusedCellChanged' },
                { subscribe: 'focusedCellChanging', emit: 'onFocusedCellChanging' },
                { subscribe: 'focusedRowChanged', emit: 'onFocusedRowChanged' },
                { subscribe: 'focusedRowChanging', emit: 'onFocusedRowChanging' },
                { subscribe: 'initialized', emit: 'onInitialized' },
                { subscribe: 'initNewRow', emit: 'onInitNewRow' },
                { subscribe: 'keyDown', emit: 'onKeyDown' },
                { subscribe: 'optionChanged', emit: 'onOptionChanged' },
                { subscribe: 'rowClick', emit: 'onRowClick' },
                { subscribe: 'rowCollapsed', emit: 'onRowCollapsed' },
                { subscribe: 'rowCollapsing', emit: 'onRowCollapsing' },
                { subscribe: 'rowDblClick', emit: 'onRowDblClick' },
                { subscribe: 'rowExpanded', emit: 'onRowExpanded' },
                { subscribe: 'rowExpanding', emit: 'onRowExpanding' },
                { subscribe: 'rowInserted', emit: 'onRowInserted' },
                { subscribe: 'rowInserting', emit: 'onRowInserting' },
                { subscribe: 'rowPrepared', emit: 'onRowPrepared' },
                { subscribe: 'rowRemoved', emit: 'onRowRemoved' },
                { subscribe: 'rowRemoving', emit: 'onRowRemoving' },
                { subscribe: 'rowUpdated', emit: 'onRowUpdated' },
                { subscribe: 'rowUpdating', emit: 'onRowUpdating' },
                { subscribe: 'rowValidating', emit: 'onRowValidating' },
                { subscribe: 'selectionChanged', emit: 'onSelectionChanged' },
                { subscribe: 'toolbarPreparing', emit: 'onToolbarPreparing' },
                { emit: 'accessKeyChange' },
                { emit: 'activeStateEnabledChange' },
                { emit: 'allowColumnReorderingChange' },
                { emit: 'allowColumnResizingChange' },
                { emit: 'autoNavigateToFocusedRowChange' },
                { emit: 'cacheEnabledChange' },
                { emit: 'cellHintEnabledChange' },
                { emit: 'columnAutoWidthChange' },
                { emit: 'columnChooserChange' },
                { emit: 'columnFixingChange' },
                { emit: 'columnHidingEnabledChange' },
                { emit: 'columnMinWidthChange' },
                { emit: 'columnResizingModeChange' },
                { emit: 'columnsChange' },
                { emit: 'columnWidthChange' },
                { emit: 'customizeColumnsChange' },
                { emit: 'customizeExportDataChange' },
                { emit: 'dataSourceChange' },
                { emit: 'dateSerializationFormatChange' },
                { emit: 'disabledChange' },
                { emit: 'editingChange' },
                { emit: 'elementAttrChange' },
                { emit: 'errorRowEnabledChange' },
                { emit: 'exportChange' },
                { emit: 'filterBuilderChange' },
                { emit: 'filterBuilderPopupChange' },
                { emit: 'filterPanelChange' },
                { emit: 'filterRowChange' },
                { emit: 'filterSyncEnabledChange' },
                { emit: 'filterValueChange' },
                { emit: 'focusedColumnIndexChange' },
                { emit: 'focusedRowEnabledChange' },
                { emit: 'focusedRowIndexChange' },
                { emit: 'focusedRowKeyChange' },
                { emit: 'focusStateEnabledChange' },
                { emit: 'groupingChange' },
                { emit: 'groupPanelChange' },
                { emit: 'headerFilterChange' },
                { emit: 'heightChange' },
                { emit: 'highlightChangesChange' },
                { emit: 'hintChange' },
                { emit: 'hoverStateEnabledChange' },
                { emit: 'keyboardNavigationChange' },
                { emit: 'keyExprChange' },
                { emit: 'loadPanelChange' },
                { emit: 'masterDetailChange' },
                { emit: 'noDataTextChange' },
                { emit: 'pagerChange' },
                { emit: 'pagingChange' },
                { emit: 'remoteOperationsChange' },
                { emit: 'renderAsyncChange' },
                { emit: 'repaintChangesOnlyChange' },
                { emit: 'rowAlternationEnabledChange' },
                { emit: 'rowDraggingChange' },
                { emit: 'rowTemplateChange' },
                { emit: 'rtlEnabledChange' },
                { emit: 'scrollingChange' },
                { emit: 'searchPanelChange' },
                { emit: 'selectedRowKeysChange' },
                { emit: 'selectionChange' },
                { emit: 'selectionFilterChange' },
                { emit: 'showBordersChange' },
                { emit: 'showColumnHeadersChange' },
                { emit: 'showColumnLinesChange' },
                { emit: 'showRowLinesChange' },
                { emit: 'sortByGroupSummaryInfoChange' },
                { emit: 'sortingChange' },
                { emit: 'stateStoringChange' },
                { emit: 'summaryChange' },
                { emit: 'tabIndexChange' },
                { emit: 'twoWayBindingEnabledChange' },
                { emit: 'visibleChange' },
                { emit: 'widthChange' },
                { emit: 'wordWrapEnabledChange' }
            ]);
            _this._idh.setHost(_this);
            optionHost.setHost(_this);
            return _this;
        }
        Object.defineProperty(DxDataGridComponent.prototype, "accessKey", {
            /**
             * Specifies the shortcut key that sets focus on the widget.
             */
            get: function () {
                return this._getOption('accessKey');
            },
            set: function (value) {
                this._setOption('accessKey', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "activeStateEnabled", {
            /**
             * Specifies whether or not the widget changes its state when interacting with a user.
             */
            get: function () {
                return this._getOption('activeStateEnabled');
            },
            set: function (value) {
                this._setOption('activeStateEnabled', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "allowColumnReordering", {
            /**
             * Specifies whether a user can reorder columns.
             */
            get: function () {
                return this._getOption('allowColumnReordering');
            },
            set: function (value) {
                this._setOption('allowColumnReordering', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "allowColumnResizing", {
            /**
             * Specifies whether a user can resize columns.
             */
            get: function () {
                return this._getOption('allowColumnResizing');
            },
            set: function (value) {
                this._setOption('allowColumnResizing', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "autoNavigateToFocusedRow", {
            /**
             * Automatically scrolls to the focused row when the focusedRowKey is changed.
             */
            get: function () {
                return this._getOption('autoNavigateToFocusedRow');
            },
            set: function (value) {
                this._setOption('autoNavigateToFocusedRow', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "cacheEnabled", {
            /**
             * Specifies whether data should be cached.
             */
            get: function () {
                return this._getOption('cacheEnabled');
            },
            set: function (value) {
                this._setOption('cacheEnabled', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "cellHintEnabled", {
            /**
             * Enables a hint that appears when a user hovers the mouse pointer over a cell with truncated content.
             */
            get: function () {
                return this._getOption('cellHintEnabled');
            },
            set: function (value) {
                this._setOption('cellHintEnabled', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "columnAutoWidth", {
            /**
             * Specifies whether columns should adjust their widths to the content.
             */
            get: function () {
                return this._getOption('columnAutoWidth');
            },
            set: function (value) {
                this._setOption('columnAutoWidth', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "columnChooser", {
            /**
             * Configures the column chooser.
             */
            get: function () {
                return this._getOption('columnChooser');
            },
            set: function (value) {
                this._setOption('columnChooser', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "columnFixing", {
            /**
             * Configures column fixing.
             */
            get: function () {
                return this._getOption('columnFixing');
            },
            set: function (value) {
                this._setOption('columnFixing', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "columnHidingEnabled", {
            /**
             * Specifies whether the widget should hide columns to adapt to the screen or container size. Ignored if allowColumnResizing is true and columnResizingMode is "widget".
             */
            get: function () {
                return this._getOption('columnHidingEnabled');
            },
            set: function (value) {
                this._setOption('columnHidingEnabled', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "columnMinWidth", {
            /**
             * Specifies the minimum width of columns.
             */
            get: function () {
                return this._getOption('columnMinWidth');
            },
            set: function (value) {
                this._setOption('columnMinWidth', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "columnResizingMode", {
            /**
             * Specifies how the widget resizes columns. Applies only if allowColumnResizing is true.
             */
            get: function () {
                return this._getOption('columnResizingMode');
            },
            set: function (value) {
                this._setOption('columnResizingMode', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "columns", {
            /**
             * An array of grid columns.
             */
            get: function () {
                return this._getOption('columns');
            },
            set: function (value) {
                this._setOption('columns', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "columnWidth", {
            /**
             * Specifies the width for all data columns. Has a lower priority than the column.width option.
             */
            get: function () {
                return this._getOption('columnWidth');
            },
            set: function (value) {
                this._setOption('columnWidth', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "customizeColumns", {
            /**
             * Customizes columns after they are created.
             */
            get: function () {
                return this._getOption('customizeColumns');
            },
            set: function (value) {
                this._setOption('customizeColumns', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "customizeExportData", {
            /**
             * Since v20.1, we recommend ExcelJS-based export which does not use this option.
             */
            get: function () {
                return this._getOption('customizeExportData');
            },
            set: function (value) {
                this._setOption('customizeExportData', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "dataSource", {
            /**
             * Binds the widget to data.
             */
            get: function () {
                return this._getOption('dataSource');
            },
            set: function (value) {
                this._setOption('dataSource', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "dateSerializationFormat", {
            /**
             * Specifies the format in which date-time values should be sent to the server. Use it only if you do not specify the dataSource at design time.
             */
            get: function () {
                return this._getOption('dateSerializationFormat');
            },
            set: function (value) {
                this._setOption('dateSerializationFormat', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "disabled", {
            /**
             * Specifies whether the widget responds to user interaction.
             */
            get: function () {
                return this._getOption('disabled');
            },
            set: function (value) {
                this._setOption('disabled', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "editing", {
            /**
             * Configures editing.
             */
            get: function () {
                return this._getOption('editing');
            },
            set: function (value) {
                this._setOption('editing', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "elementAttr", {
            /**
             * Specifies the attributes to be attached to the widget's root element.
             */
            get: function () {
                return this._getOption('elementAttr');
            },
            set: function (value) {
                this._setOption('elementAttr', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "errorRowEnabled", {
            /**
             * Indicates whether to show the error row.
             */
            get: function () {
                return this._getOption('errorRowEnabled');
            },
            set: function (value) {
                this._setOption('errorRowEnabled', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "export", {
            /**
             * Configures client-side exporting.
             */
            get: function () {
                return this._getOption('export');
            },
            set: function (value) {
                this._setOption('export', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "filterBuilder", {
            /**
             * Configures the integrated filter builder.
             */
            get: function () {
                return this._getOption('filterBuilder');
            },
            set: function (value) {
                this._setOption('filterBuilder', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "filterBuilderPopup", {
            /**
             * Configures the popup in which the integrated filter builder is shown.
             */
            get: function () {
                return this._getOption('filterBuilderPopup');
            },
            set: function (value) {
                this._setOption('filterBuilderPopup', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "filterPanel", {
            /**
             * Configures the filter panel.
             */
            get: function () {
                return this._getOption('filterPanel');
            },
            set: function (value) {
                this._setOption('filterPanel', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "filterRow", {
            /**
             * Configures the filter row.
             */
            get: function () {
                return this._getOption('filterRow');
            },
            set: function (value) {
                this._setOption('filterRow', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "filterSyncEnabled", {
            /**
             * Specifies whether to synchronize the filter row, header filter, and filter builder. The synchronized filter expression is stored in the filterValue option.
             */
            get: function () {
                return this._getOption('filterSyncEnabled');
            },
            set: function (value) {
                this._setOption('filterSyncEnabled', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "filterValue", {
            /**
             * Specifies a filter expression.
             */
            get: function () {
                return this._getOption('filterValue');
            },
            set: function (value) {
                this._setOption('filterValue', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "focusedColumnIndex", {
            /**
             * The index of the column that contains the focused data cell. This index is taken from the columns array.
             */
            get: function () {
                return this._getOption('focusedColumnIndex');
            },
            set: function (value) {
                this._setOption('focusedColumnIndex', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "focusedRowEnabled", {
            /**
             * Specifies whether the focused row feature is enabled.
             */
            get: function () {
                return this._getOption('focusedRowEnabled');
            },
            set: function (value) {
                this._setOption('focusedRowEnabled', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "focusedRowIndex", {
            /**
             * Specifies or indicates the focused data row's index. Use this option when focusedRowEnabled is true.
             */
            get: function () {
                return this._getOption('focusedRowIndex');
            },
            set: function (value) {
                this._setOption('focusedRowIndex', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "focusedRowKey", {
            /**
             * Specifies initially or currently focused grid row's key. Use it when focusedRowEnabled is true.
             */
            get: function () {
                return this._getOption('focusedRowKey');
            },
            set: function (value) {
                this._setOption('focusedRowKey', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "focusStateEnabled", {
            /**
             * Specifies whether the widget can be focused using keyboard navigation.
             */
            get: function () {
                return this._getOption('focusStateEnabled');
            },
            set: function (value) {
                this._setOption('focusStateEnabled', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "grouping", {
            /**
             * Configures grouping.
             */
            get: function () {
                return this._getOption('grouping');
            },
            set: function (value) {
                this._setOption('grouping', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "groupPanel", {
            /**
             * Configures the group panel.
             */
            get: function () {
                return this._getOption('groupPanel');
            },
            set: function (value) {
                this._setOption('groupPanel', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "headerFilter", {
            /**
             * Configures the header filter feature.
             */
            get: function () {
                return this._getOption('headerFilter');
            },
            set: function (value) {
                this._setOption('headerFilter', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "height", {
            /**
             * Specifies the widget's height.
             */
            get: function () {
                return this._getOption('height');
            },
            set: function (value) {
                this._setOption('height', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "highlightChanges", {
            /**
             * Specifies whether to highlight rows and cells with edited data. repaintChangesOnly should be true.
             */
            get: function () {
                return this._getOption('highlightChanges');
            },
            set: function (value) {
                this._setOption('highlightChanges', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "hint", {
            /**
             * Specifies text for a hint that appears when a user pauses on the widget.
             */
            get: function () {
                return this._getOption('hint');
            },
            set: function (value) {
                this._setOption('hint', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "hoverStateEnabled", {
            /**
             * Specifies whether the widget changes its state when a user pauses on it.
             */
            get: function () {
                return this._getOption('hoverStateEnabled');
            },
            set: function (value) {
                this._setOption('hoverStateEnabled', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "keyboardNavigation", {
            /**
             * Configures keyboard navigation.
             */
            get: function () {
                return this._getOption('keyboardNavigation');
            },
            set: function (value) {
                this._setOption('keyboardNavigation', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "keyExpr", {
            /**
             * Specifies the key property (or properties) that provide(s) key values to access data items. Each key value must be unique. This option applies only if data is a simple array.
             */
            get: function () {
                return this._getOption('keyExpr');
            },
            set: function (value) {
                this._setOption('keyExpr', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "loadPanel", {
            /**
             * Configures the load panel.
             */
            get: function () {
                return this._getOption('loadPanel');
            },
            set: function (value) {
                this._setOption('loadPanel', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "masterDetail", {
            /**
             * Allows you to build a master-detail interface in the grid.
             */
            get: function () {
                return this._getOption('masterDetail');
            },
            set: function (value) {
                this._setOption('masterDetail', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "noDataText", {
            /**
             * Specifies text shown when the widget does not display any data.
             */
            get: function () {
                return this._getOption('noDataText');
            },
            set: function (value) {
                this._setOption('noDataText', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "pager", {
            /**
             * Configures the pager.
             */
            get: function () {
                return this._getOption('pager');
            },
            set: function (value) {
                this._setOption('pager', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "paging", {
            /**
             * Configures paging.
             */
            get: function () {
                return this._getOption('paging');
            },
            set: function (value) {
                this._setOption('paging', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "remoteOperations", {
            /**
             * Notifies the DataGrid of the server's data processing operations.
             */
            get: function () {
                return this._getOption('remoteOperations');
            },
            set: function (value) {
                this._setOption('remoteOperations', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "renderAsync", {
            /**
             * Specifies whether to render the filter row, command columns, and columns with showEditorAlways set to true after other elements.
             */
            get: function () {
                return this._getOption('renderAsync');
            },
            set: function (value) {
                this._setOption('renderAsync', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "repaintChangesOnly", {
            /**
             * Specifies whether to repaint only those cells whose data changed.
             */
            get: function () {
                return this._getOption('repaintChangesOnly');
            },
            set: function (value) {
                this._setOption('repaintChangesOnly', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "rowAlternationEnabled", {
            /**
             * Specifies whether rows should be shaded differently.
             */
            get: function () {
                return this._getOption('rowAlternationEnabled');
            },
            set: function (value) {
                this._setOption('rowAlternationEnabled', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "rowDragging", {
            /**
             * Configures row reordering using drag and drop gestures.
             */
            get: function () {
                return this._getOption('rowDragging');
            },
            set: function (value) {
                this._setOption('rowDragging', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "rowTemplate", {
            /**
             * Specifies a custom template for rows.
             */
            get: function () {
                return this._getOption('rowTemplate');
            },
            set: function (value) {
                this._setOption('rowTemplate', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "rtlEnabled", {
            /**
             * Switches the widget to a right-to-left representation.
             */
            get: function () {
                return this._getOption('rtlEnabled');
            },
            set: function (value) {
                this._setOption('rtlEnabled', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "scrolling", {
            /**
             * Configures scrolling.
             */
            get: function () {
                return this._getOption('scrolling');
            },
            set: function (value) {
                this._setOption('scrolling', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "searchPanel", {
            /**
             * Configures the search panel.
             */
            get: function () {
                return this._getOption('searchPanel');
            },
            set: function (value) {
                this._setOption('searchPanel', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "selectedRowKeys", {
            /**
             * Allows you to select rows or determine which rows are selected.
             */
            get: function () {
                return this._getOption('selectedRowKeys');
            },
            set: function (value) {
                this._setOption('selectedRowKeys', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "selection", {
            /**
             * Configures runtime selection.
             */
            get: function () {
                return this._getOption('selection');
            },
            set: function (value) {
                this._setOption('selection', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "selectionFilter", {
            /**
             * Specifies filters for the rows that must be selected initially. Applies only if selection.deferred is true.
             */
            get: function () {
                return this._getOption('selectionFilter');
            },
            set: function (value) {
                this._setOption('selectionFilter', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "showBorders", {
            /**
             * Specifies whether the outer borders of the widget are visible.
             */
            get: function () {
                return this._getOption('showBorders');
            },
            set: function (value) {
                this._setOption('showBorders', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "showColumnHeaders", {
            /**
             * Specifies whether column headers are visible.
             */
            get: function () {
                return this._getOption('showColumnHeaders');
            },
            set: function (value) {
                this._setOption('showColumnHeaders', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "showColumnLines", {
            /**
             * Specifies whether vertical lines that separate one column from another are visible.
             */
            get: function () {
                return this._getOption('showColumnLines');
            },
            set: function (value) {
                this._setOption('showColumnLines', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "showRowLines", {
            /**
             * Specifies whether horizontal lines that separate one row from another are visible.
             */
            get: function () {
                return this._getOption('showRowLines');
            },
            set: function (value) {
                this._setOption('showRowLines', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "sortByGroupSummaryInfo", {
            /**
             * Allows you to sort groups according to the values of group summary items.
             */
            get: function () {
                return this._getOption('sortByGroupSummaryInfo');
            },
            set: function (value) {
                this._setOption('sortByGroupSummaryInfo', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "sorting", {
            /**
             * Configures runtime sorting.
             */
            get: function () {
                return this._getOption('sorting');
            },
            set: function (value) {
                this._setOption('sorting', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "stateStoring", {
            /**
             * Configures state storing.
             */
            get: function () {
                return this._getOption('stateStoring');
            },
            set: function (value) {
                this._setOption('stateStoring', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "summary", {
            /**
             * Specifies the options of the grid summary.
             */
            get: function () {
                return this._getOption('summary');
            },
            set: function (value) {
                this._setOption('summary', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "tabIndex", {
            /**
             * Specifies the number of the element when the Tab key is used for navigating.
             */
            get: function () {
                return this._getOption('tabIndex');
            },
            set: function (value) {
                this._setOption('tabIndex', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "twoWayBindingEnabled", {
            /**
             * Specifies whether to enable two-way data binding.
             */
            get: function () {
                return this._getOption('twoWayBindingEnabled');
            },
            set: function (value) {
                this._setOption('twoWayBindingEnabled', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "visible", {
            /**
             * Specifies whether the widget is visible.
             */
            get: function () {
                return this._getOption('visible');
            },
            set: function (value) {
                this._setOption('visible', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "width", {
            /**
             * Specifies the widget's width.
             */
            get: function () {
                return this._getOption('width');
            },
            set: function (value) {
                this._setOption('width', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "wordWrapEnabled", {
            /**
             * Specifies whether text that does not fit into a column should be wrapped.
             */
            get: function () {
                return this._getOption('wordWrapEnabled');
            },
            set: function (value) {
                this._setOption('wordWrapEnabled', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "columnsChildren", {
            get: function () {
                return this._getOption('columns');
            },
            set: function (value) {
                this.setChildren('columns', value);
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(DxDataGridComponent.prototype, "sortByGroupSummaryInfoChildren", {
            get: function () {
                return this._getOption('sortByGroupSummaryInfo');
            },
            set: function (value) {
                this.setChildren('sortByGroupSummaryInfo', value);
            },
            enumerable: true,
            configurable: true
        });
        DxDataGridComponent.prototype._createInstance = function (element, options) {
            return new DxDataGrid(element, options);
        };
        DxDataGridComponent.prototype.ngOnDestroy = function () {
            this._destroyWidget();
        };
        DxDataGridComponent.prototype.ngOnChanges = function (changes) {
            _super.prototype.ngOnChanges.call(this, changes);
            this.setupChanges('columns', changes);
            this.setupChanges('dataSource', changes);
            this.setupChanges('keyExpr', changes);
            this.setupChanges('selectedRowKeys', changes);
            this.setupChanges('sortByGroupSummaryInfo', changes);
        };
        DxDataGridComponent.prototype.setupChanges = function (prop, changes) {
            if (!(prop in this._optionsToUpdate)) {
                this._idh.setup(prop, changes);
            }
        };
        DxDataGridComponent.prototype.ngDoCheck = function () {
            this._idh.doCheck('columns');
            this._idh.doCheck('dataSource');
            this._idh.doCheck('keyExpr');
            this._idh.doCheck('selectedRowKeys');
            this._idh.doCheck('sortByGroupSummaryInfo');
            this._watcherHelper.checkWatchers();
            _super.prototype.ngDoCheck.call(this);
            _super.prototype.clearChangedOptions.call(this);
        };
        DxDataGridComponent.prototype._setOption = function (name, value) {
            var isSetup = this._idh.setupSingle(name, value);
            var isChanged = this._idh.getChanges(name, value) !== null;
            if (isSetup || isChanged) {
                _super.prototype._setOption.call(this, name, value);
            }
        };
        DxDataGridComponent.ctorParameters = function () { return [
            { type: core.ElementRef },
            { type: core.NgZone },
            { type: core$1.DxTemplateHost },
            { type: core$1.WatcherHelper },
            { type: core$1.IterableDifferHelper },
            { type: core$1.NestedOptionHost },
            { type: platformBrowser.TransferState },
            { type: undefined, decorators: [{ type: core.Inject, args: [core.PLATFORM_ID,] }] }
        ]; };
        __decorate([
            core.Input(),
            __metadata("design:type", String),
            __metadata("design:paramtypes", [String])
        ], DxDataGridComponent.prototype, "accessKey", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Boolean),
            __metadata("design:paramtypes", [Boolean])
        ], DxDataGridComponent.prototype, "activeStateEnabled", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Boolean),
            __metadata("design:paramtypes", [Boolean])
        ], DxDataGridComponent.prototype, "allowColumnReordering", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Boolean),
            __metadata("design:paramtypes", [Boolean])
        ], DxDataGridComponent.prototype, "allowColumnResizing", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Boolean),
            __metadata("design:paramtypes", [Boolean])
        ], DxDataGridComponent.prototype, "autoNavigateToFocusedRow", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Boolean),
            __metadata("design:paramtypes", [Boolean])
        ], DxDataGridComponent.prototype, "cacheEnabled", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Boolean),
            __metadata("design:paramtypes", [Boolean])
        ], DxDataGridComponent.prototype, "cellHintEnabled", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Boolean),
            __metadata("design:paramtypes", [Boolean])
        ], DxDataGridComponent.prototype, "columnAutoWidth", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Object),
            __metadata("design:paramtypes", [Object])
        ], DxDataGridComponent.prototype, "columnChooser", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Object),
            __metadata("design:paramtypes", [Object])
        ], DxDataGridComponent.prototype, "columnFixing", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Boolean),
            __metadata("design:paramtypes", [Boolean])
        ], DxDataGridComponent.prototype, "columnHidingEnabled", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Number),
            __metadata("design:paramtypes", [Number])
        ], DxDataGridComponent.prototype, "columnMinWidth", null);
        __decorate([
            core.Input(),
            __metadata("design:type", String),
            __metadata("design:paramtypes", [String])
        ], DxDataGridComponent.prototype, "columnResizingMode", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Array),
            __metadata("design:paramtypes", [Array])
        ], DxDataGridComponent.prototype, "columns", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Number),
            __metadata("design:paramtypes", [Number])
        ], DxDataGridComponent.prototype, "columnWidth", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", [Function])
        ], DxDataGridComponent.prototype, "customizeColumns", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Function),
            __metadata("design:paramtypes", [Function])
        ], DxDataGridComponent.prototype, "customizeExportData", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Object),
            __metadata("design:paramtypes", [Object])
        ], DxDataGridComponent.prototype, "dataSource", null);
        __decorate([
            core.Input(),
            __metadata("design:type", String),
            __metadata("design:paramtypes", [String])
        ], DxDataGridComponent.prototype, "dateSerializationFormat", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Boolean),
            __metadata("design:paramtypes", [Boolean])
        ], DxDataGridComponent.prototype, "disabled", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Object),
            __metadata("design:paramtypes", [Object])
        ], DxDataGridComponent.prototype, "editing", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Object),
            __metadata("design:paramtypes", [Object])
        ], DxDataGridComponent.prototype, "elementAttr", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Boolean),
            __metadata("design:paramtypes", [Boolean])
        ], DxDataGridComponent.prototype, "errorRowEnabled", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Object),
            __metadata("design:paramtypes", [Object])
        ], DxDataGridComponent.prototype, "export", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Object),
            __metadata("design:paramtypes", [Object])
        ], DxDataGridComponent.prototype, "filterBuilder", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Object),
            __metadata("design:paramtypes", [Object])
        ], DxDataGridComponent.prototype, "filterBuilderPopup", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Object),
            __metadata("design:paramtypes", [Object])
        ], DxDataGridComponent.prototype, "filterPanel", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Object),
            __metadata("design:paramtypes", [Object])
        ], DxDataGridComponent.prototype, "filterRow", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Object),
            __metadata("design:paramtypes", [Object])
        ], DxDataGridComponent.prototype, "filterSyncEnabled", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Object),
            __metadata("design:paramtypes", [Object])
        ], DxDataGridComponent.prototype, "filterValue", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Number),
            __metadata("design:paramtypes", [Number])
        ], DxDataGridComponent.prototype, "focusedColumnIndex", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Boolean),
            __metadata("design:paramtypes", [Boolean])
        ], DxDataGridComponent.prototype, "focusedRowEnabled", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Number),
            __metadata("design:paramtypes", [Number])
        ], DxDataGridComponent.prototype, "focusedRowIndex", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Object),
            __metadata("design:paramtypes", [Object])
        ], DxDataGridComponent.prototype, "focusedRowKey", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Boolean),
            __metadata("design:paramtypes", [Boolean])
        ], DxDataGridComponent.prototype, "focusStateEnabled", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Object),
            __metadata("design:paramtypes", [Object])
        ], DxDataGridComponent.prototype, "grouping", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Object),
            __metadata("design:paramtypes", [Object])
        ], DxDataGridComponent.prototype, "groupPanel", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Object),
            __metadata("design:paramtypes", [Object])
        ], DxDataGridComponent.prototype, "headerFilter", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Object),
            __metadata("design:paramtypes", [Object])
        ], DxDataGridComponent.prototype, "height", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Boolean),
            __metadata("design:paramtypes", [Boolean])
        ], DxDataGridComponent.prototype, "highlightChanges", null);
        __decorate([
            core.Input(),
            __metadata("design:type", String),
            __metadata("design:paramtypes", [String])
        ], DxDataGridComponent.prototype, "hint", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Boolean),
            __metadata("design:paramtypes", [Boolean])
        ], DxDataGridComponent.prototype, "hoverStateEnabled", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Object),
            __metadata("design:paramtypes", [Object])
        ], DxDataGridComponent.prototype, "keyboardNavigation", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Object),
            __metadata("design:paramtypes", [Object])
        ], DxDataGridComponent.prototype, "keyExpr", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Object),
            __metadata("design:paramtypes", [Object])
        ], DxDataGridComponent.prototype, "loadPanel", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Object),
            __metadata("design:paramtypes", [Object])
        ], DxDataGridComponent.prototype, "masterDetail", null);
        __decorate([
            core.Input(),
            __metadata("design:type", String),
            __metadata("design:paramtypes", [String])
        ], DxDataGridComponent.prototype, "noDataText", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Object),
            __metadata("design:paramtypes", [Object])
        ], DxDataGridComponent.prototype, "pager", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Object),
            __metadata("design:paramtypes", [Object])
        ], DxDataGridComponent.prototype, "paging", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Object),
            __metadata("design:paramtypes", [Object])
        ], DxDataGridComponent.prototype, "remoteOperations", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Boolean),
            __metadata("design:paramtypes", [Boolean])
        ], DxDataGridComponent.prototype, "renderAsync", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Boolean),
            __metadata("design:paramtypes", [Boolean])
        ], DxDataGridComponent.prototype, "repaintChangesOnly", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Boolean),
            __metadata("design:paramtypes", [Boolean])
        ], DxDataGridComponent.prototype, "rowAlternationEnabled", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Object),
            __metadata("design:paramtypes", [Object])
        ], DxDataGridComponent.prototype, "rowDragging", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Object),
            __metadata("design:paramtypes", [Object])
        ], DxDataGridComponent.prototype, "rowTemplate", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Boolean),
            __metadata("design:paramtypes", [Boolean])
        ], DxDataGridComponent.prototype, "rtlEnabled", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Object),
            __metadata("design:paramtypes", [Object])
        ], DxDataGridComponent.prototype, "scrolling", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Object),
            __metadata("design:paramtypes", [Object])
        ], DxDataGridComponent.prototype, "searchPanel", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Array),
            __metadata("design:paramtypes", [Array])
        ], DxDataGridComponent.prototype, "selectedRowKeys", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Object),
            __metadata("design:paramtypes", [Object])
        ], DxDataGridComponent.prototype, "selection", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Object),
            __metadata("design:paramtypes", [Object])
        ], DxDataGridComponent.prototype, "selectionFilter", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Boolean),
            __metadata("design:paramtypes", [Boolean])
        ], DxDataGridComponent.prototype, "showBorders", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Boolean),
            __metadata("design:paramtypes", [Boolean])
        ], DxDataGridComponent.prototype, "showColumnHeaders", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Boolean),
            __metadata("design:paramtypes", [Boolean])
        ], DxDataGridComponent.prototype, "showColumnLines", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Boolean),
            __metadata("design:paramtypes", [Boolean])
        ], DxDataGridComponent.prototype, "showRowLines", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Array),
            __metadata("design:paramtypes", [Array])
        ], DxDataGridComponent.prototype, "sortByGroupSummaryInfo", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Object),
            __metadata("design:paramtypes", [Object])
        ], DxDataGridComponent.prototype, "sorting", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Object),
            __metadata("design:paramtypes", [Object])
        ], DxDataGridComponent.prototype, "stateStoring", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Object),
            __metadata("design:paramtypes", [Object])
        ], DxDataGridComponent.prototype, "summary", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Number),
            __metadata("design:paramtypes", [Number])
        ], DxDataGridComponent.prototype, "tabIndex", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Boolean),
            __metadata("design:paramtypes", [Boolean])
        ], DxDataGridComponent.prototype, "twoWayBindingEnabled", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Boolean),
            __metadata("design:paramtypes", [Boolean])
        ], DxDataGridComponent.prototype, "visible", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Object),
            __metadata("design:paramtypes", [Object])
        ], DxDataGridComponent.prototype, "width", null);
        __decorate([
            core.Input(),
            __metadata("design:type", Boolean),
            __metadata("design:paramtypes", [Boolean])
        ], DxDataGridComponent.prototype, "wordWrapEnabled", null);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "onAdaptiveDetailRowPreparing", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "onCellClick", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "onCellDblClick", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "onCellHoverChanged", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "onCellPrepared", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "onContentReady", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "onContextMenuPreparing", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "onDataErrorOccurred", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "onDisposing", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "onEditingStart", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "onEditorPrepared", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "onEditorPreparing", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "onExported", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "onExporting", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "onFileSaving", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "onFocusedCellChanged", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "onFocusedCellChanging", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "onFocusedRowChanged", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "onFocusedRowChanging", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "onInitialized", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "onInitNewRow", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "onKeyDown", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "onOptionChanged", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "onRowClick", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "onRowCollapsed", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "onRowCollapsing", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "onRowDblClick", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "onRowExpanded", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "onRowExpanding", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "onRowInserted", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "onRowInserting", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "onRowPrepared", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "onRowRemoved", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "onRowRemoving", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "onRowUpdated", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "onRowUpdating", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "onRowValidating", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "onSelectionChanged", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "onToolbarPreparing", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "accessKeyChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "activeStateEnabledChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "allowColumnReorderingChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "allowColumnResizingChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "autoNavigateToFocusedRowChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "cacheEnabledChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "cellHintEnabledChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "columnAutoWidthChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "columnChooserChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "columnFixingChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "columnHidingEnabledChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "columnMinWidthChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "columnResizingModeChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "columnsChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "columnWidthChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "customizeColumnsChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "customizeExportDataChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "dataSourceChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "dateSerializationFormatChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "disabledChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "editingChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "elementAttrChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "errorRowEnabledChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "exportChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "filterBuilderChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "filterBuilderPopupChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "filterPanelChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "filterRowChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "filterSyncEnabledChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "filterValueChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "focusedColumnIndexChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "focusedRowEnabledChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "focusedRowIndexChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "focusedRowKeyChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "focusStateEnabledChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "groupingChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "groupPanelChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "headerFilterChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "heightChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "highlightChangesChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "hintChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "hoverStateEnabledChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "keyboardNavigationChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "keyExprChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "loadPanelChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "masterDetailChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "noDataTextChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "pagerChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "pagingChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "remoteOperationsChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "renderAsyncChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "repaintChangesOnlyChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "rowAlternationEnabledChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "rowDraggingChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "rowTemplateChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "rtlEnabledChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "scrollingChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "searchPanelChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "selectedRowKeysChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "selectionChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "selectionFilterChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "showBordersChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "showColumnHeadersChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "showColumnLinesChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "showRowLinesChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "sortByGroupSummaryInfoChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "sortingChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "stateStoringChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "summaryChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "tabIndexChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "twoWayBindingEnabledChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "visibleChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "widthChange", void 0);
        __decorate([
            core.Output(),
            __metadata("design:type", core.EventEmitter)
        ], DxDataGridComponent.prototype, "wordWrapEnabledChange", void 0);
        __decorate([
            core.ContentChildren(nested.DxiColumnComponent),
            __metadata("design:type", core.QueryList),
            __metadata("design:paramtypes", [Object])
        ], DxDataGridComponent.prototype, "columnsChildren", null);
        __decorate([
            core.ContentChildren(nested.DxiSortByGroupSummaryInfoComponent),
            __metadata("design:type", core.QueryList),
            __metadata("design:paramtypes", [Object])
        ], DxDataGridComponent.prototype, "sortByGroupSummaryInfoChildren", null);
        DxDataGridComponent = __decorate([ __param(7, core.Inject(core.PLATFORM_ID)),
            __metadata("design:paramtypes", [core.ElementRef, core.NgZone, core$1.DxTemplateHost,
                core$1.WatcherHelper,
                core$1.IterableDifferHelper,
                core$1.NestedOptionHost,
                platformBrowser.TransferState, Object])
        ], DxDataGridComponent);
DxDataGridComponent.ɵfac = function DxDataGridComponent_Factory(t) { return new (t || DxDataGridComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.NgZone), ɵngcc0.ɵɵdirectiveInject(ɵngcc1.DxTemplateHost), ɵngcc0.ɵɵdirectiveInject(ɵngcc1.WatcherHelper), ɵngcc0.ɵɵdirectiveInject(ɵngcc1.IterableDifferHelper), ɵngcc0.ɵɵdirectiveInject(ɵngcc1.NestedOptionHost), ɵngcc0.ɵɵdirectiveInject(ɵngcc2.TransferState), ɵngcc0.ɵɵdirectiveInject(core.PLATFORM_ID)); };
DxDataGridComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: DxDataGridComponent, selectors: [["dx-data-grid"]], contentQueries: function DxDataGridComponent_ContentQueries(rf, ctx, dirIndex) { if (rf & 1) {
        ɵngcc0.ɵɵcontentQuery(dirIndex, nested.DxiColumnComponent, false);
        ɵngcc0.ɵɵcontentQuery(dirIndex, nested.DxiSortByGroupSummaryInfoComponent, false);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.columnsChildren = _t);
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.sortByGroupSummaryInfoChildren = _t);
    } }, inputs: { accessKey: "accessKey", activeStateEnabled: "activeStateEnabled", allowColumnReordering: "allowColumnReordering", allowColumnResizing: "allowColumnResizing", autoNavigateToFocusedRow: "autoNavigateToFocusedRow", cacheEnabled: "cacheEnabled", cellHintEnabled: "cellHintEnabled", columnAutoWidth: "columnAutoWidth", columnChooser: "columnChooser", columnFixing: "columnFixing", columnHidingEnabled: "columnHidingEnabled", columnMinWidth: "columnMinWidth", columnResizingMode: "columnResizingMode", columns: "columns", columnWidth: "columnWidth", customizeColumns: "customizeColumns", customizeExportData: "customizeExportData", dataSource: "dataSource", dateSerializationFormat: "dateSerializationFormat", disabled: "disabled", editing: "editing", elementAttr: "elementAttr", errorRowEnabled: "errorRowEnabled", export: "export", filterBuilder: "filterBuilder", filterBuilderPopup: "filterBuilderPopup", filterPanel: "filterPanel", filterRow: "filterRow", filterSyncEnabled: "filterSyncEnabled", filterValue: "filterValue", focusedColumnIndex: "focusedColumnIndex", focusedRowEnabled: "focusedRowEnabled", focusedRowIndex: "focusedRowIndex", focusedRowKey: "focusedRowKey", focusStateEnabled: "focusStateEnabled", grouping: "grouping", groupPanel: "groupPanel", headerFilter: "headerFilter", height: "height", highlightChanges: "highlightChanges", hint: "hint", hoverStateEnabled: "hoverStateEnabled", keyboardNavigation: "keyboardNavigation", keyExpr: "keyExpr", loadPanel: "loadPanel", masterDetail: "masterDetail", noDataText: "noDataText", pager: "pager", paging: "paging", remoteOperations: "remoteOperations", renderAsync: "renderAsync", repaintChangesOnly: "repaintChangesOnly", rowAlternationEnabled: "rowAlternationEnabled", rowDragging: "rowDragging", rowTemplate: "rowTemplate", rtlEnabled: "rtlEnabled", scrolling: "scrolling", searchPanel: "searchPanel", selectedRowKeys: "selectedRowKeys", selection: "selection", selectionFilter: "selectionFilter", showBorders: "showBorders", showColumnHeaders: "showColumnHeaders", showColumnLines: "showColumnLines", showRowLines: "showRowLines", sortByGroupSummaryInfo: "sortByGroupSummaryInfo", sorting: "sorting", stateStoring: "stateStoring", summary: "summary", tabIndex: "tabIndex", twoWayBindingEnabled: "twoWayBindingEnabled", visible: "visible", width: "width", wordWrapEnabled: "wordWrapEnabled" }, outputs: { onAdaptiveDetailRowPreparing: "onAdaptiveDetailRowPreparing", onCellClick: "onCellClick", onCellDblClick: "onCellDblClick", onCellHoverChanged: "onCellHoverChanged", onCellPrepared: "onCellPrepared", onContentReady: "onContentReady", onContextMenuPreparing: "onContextMenuPreparing", onDataErrorOccurred: "onDataErrorOccurred", onDisposing: "onDisposing", onEditingStart: "onEditingStart", onEditorPrepared: "onEditorPrepared", onEditorPreparing: "onEditorPreparing", onExported: "onExported", onExporting: "onExporting", onFileSaving: "onFileSaving", onFocusedCellChanged: "onFocusedCellChanged", onFocusedCellChanging: "onFocusedCellChanging", onFocusedRowChanged: "onFocusedRowChanged", onFocusedRowChanging: "onFocusedRowChanging", onInitialized: "onInitialized", onInitNewRow: "onInitNewRow", onKeyDown: "onKeyDown", onOptionChanged: "onOptionChanged", onRowClick: "onRowClick", onRowCollapsed: "onRowCollapsed", onRowCollapsing: "onRowCollapsing", onRowDblClick: "onRowDblClick", onRowExpanded: "onRowExpanded", onRowExpanding: "onRowExpanding", onRowInserted: "onRowInserted", onRowInserting: "onRowInserting", onRowPrepared: "onRowPrepared", onRowRemoved: "onRowRemoved", onRowRemoving: "onRowRemoving", onRowUpdated: "onRowUpdated", onRowUpdating: "onRowUpdating", onRowValidating: "onRowValidating", onSelectionChanged: "onSelectionChanged", onToolbarPreparing: "onToolbarPreparing", accessKeyChange: "accessKeyChange", activeStateEnabledChange: "activeStateEnabledChange", allowColumnReorderingChange: "allowColumnReorderingChange", allowColumnResizingChange: "allowColumnResizingChange", autoNavigateToFocusedRowChange: "autoNavigateToFocusedRowChange", cacheEnabledChange: "cacheEnabledChange", cellHintEnabledChange: "cellHintEnabledChange", columnAutoWidthChange: "columnAutoWidthChange", columnChooserChange: "columnChooserChange", columnFixingChange: "columnFixingChange", columnHidingEnabledChange: "columnHidingEnabledChange", columnMinWidthChange: "columnMinWidthChange", columnResizingModeChange: "columnResizingModeChange", columnsChange: "columnsChange", columnWidthChange: "columnWidthChange", customizeColumnsChange: "customizeColumnsChange", customizeExportDataChange: "customizeExportDataChange", dataSourceChange: "dataSourceChange", dateSerializationFormatChange: "dateSerializationFormatChange", disabledChange: "disabledChange", editingChange: "editingChange", elementAttrChange: "elementAttrChange", errorRowEnabledChange: "errorRowEnabledChange", exportChange: "exportChange", filterBuilderChange: "filterBuilderChange", filterBuilderPopupChange: "filterBuilderPopupChange", filterPanelChange: "filterPanelChange", filterRowChange: "filterRowChange", filterSyncEnabledChange: "filterSyncEnabledChange", filterValueChange: "filterValueChange", focusedColumnIndexChange: "focusedColumnIndexChange", focusedRowEnabledChange: "focusedRowEnabledChange", focusedRowIndexChange: "focusedRowIndexChange", focusedRowKeyChange: "focusedRowKeyChange", focusStateEnabledChange: "focusStateEnabledChange", groupingChange: "groupingChange", groupPanelChange: "groupPanelChange", headerFilterChange: "headerFilterChange", heightChange: "heightChange", highlightChangesChange: "highlightChangesChange", hintChange: "hintChange", hoverStateEnabledChange: "hoverStateEnabledChange", keyboardNavigationChange: "keyboardNavigationChange", keyExprChange: "keyExprChange", loadPanelChange: "loadPanelChange", masterDetailChange: "masterDetailChange", noDataTextChange: "noDataTextChange", pagerChange: "pagerChange", pagingChange: "pagingChange", remoteOperationsChange: "remoteOperationsChange", renderAsyncChange: "renderAsyncChange", repaintChangesOnlyChange: "repaintChangesOnlyChange", rowAlternationEnabledChange: "rowAlternationEnabledChange", rowDraggingChange: "rowDraggingChange", rowTemplateChange: "rowTemplateChange", rtlEnabledChange: "rtlEnabledChange", scrollingChange: "scrollingChange", searchPanelChange: "searchPanelChange", selectedRowKeysChange: "selectedRowKeysChange", selectionChange: "selectionChange", selectionFilterChange: "selectionFilterChange", showBordersChange: "showBordersChange", showColumnHeadersChange: "showColumnHeadersChange", showColumnLinesChange: "showColumnLinesChange", showRowLinesChange: "showRowLinesChange", sortByGroupSummaryInfoChange: "sortByGroupSummaryInfoChange", sortingChange: "sortingChange", stateStoringChange: "stateStoringChange", summaryChange: "summaryChange", tabIndexChange: "tabIndexChange", twoWayBindingEnabledChange: "twoWayBindingEnabledChange", visibleChange: "visibleChange", widthChange: "widthChange", wordWrapEnabledChange: "wordWrapEnabledChange" }, features: [ɵngcc0.ɵɵProvidersFeature([
            core$1.DxTemplateHost,
            core$1.WatcherHelper,
            core$1.NestedOptionHost,
            core$1.IterableDifferHelper
        ]), ɵngcc0.ɵɵInheritDefinitionFeature, ɵngcc0.ɵɵNgOnChangesFeature()], decls: 0, vars: 0, template: function DxDataGridComponent_Template(rf, ctx) { }, encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DxDataGridComponent, [{
        type: core.Component,
        args: [{
                selector: 'dx-data-grid',
                template: '',
                providers: [
                    core$1.DxTemplateHost,
                    core$1.WatcherHelper,
                    core$1.NestedOptionHost,
                    core$1.IterableDifferHelper
                ]
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }, { type: ɵngcc0.NgZone }, { type: ɵngcc1.DxTemplateHost }, { type: ɵngcc1.WatcherHelper }, { type: ɵngcc1.IterableDifferHelper }, { type: ɵngcc1.NestedOptionHost }, { type: ɵngcc2.TransferState }, { type: undefined, decorators: [{
                type: core.Inject,
                args: [core.PLATFORM_ID]
            }] }]; }, { accessKey: [{
            type: core.Input
        }], activeStateEnabled: [{
            type: core.Input
        }], allowColumnReordering: [{
            type: core.Input
        }], allowColumnResizing: [{
            type: core.Input
        }], autoNavigateToFocusedRow: [{
            type: core.Input
        }], cacheEnabled: [{
            type: core.Input
        }], cellHintEnabled: [{
            type: core.Input
        }], columnAutoWidth: [{
            type: core.Input
        }], columnChooser: [{
            type: core.Input
        }], columnFixing: [{
            type: core.Input
        }], columnHidingEnabled: [{
            type: core.Input
        }], columnMinWidth: [{
            type: core.Input
        }], columnResizingMode: [{
            type: core.Input
        }], columns: [{
            type: core.Input
        }], columnWidth: [{
            type: core.Input
        }], customizeColumns: [{
            type: core.Input
        }], customizeExportData: [{
            type: core.Input
        }], dataSource: [{
            type: core.Input
        }], dateSerializationFormat: [{
            type: core.Input
        }], disabled: [{
            type: core.Input
        }], editing: [{
            type: core.Input
        }], elementAttr: [{
            type: core.Input
        }], errorRowEnabled: [{
            type: core.Input
        }], export: [{
            type: core.Input
        }], filterBuilder: [{
            type: core.Input
        }], filterBuilderPopup: [{
            type: core.Input
        }], filterPanel: [{
            type: core.Input
        }], filterRow: [{
            type: core.Input
        }], filterSyncEnabled: [{
            type: core.Input
        }], filterValue: [{
            type: core.Input
        }], focusedColumnIndex: [{
            type: core.Input
        }], focusedRowEnabled: [{
            type: core.Input
        }], focusedRowIndex: [{
            type: core.Input
        }], focusedRowKey: [{
            type: core.Input
        }], focusStateEnabled: [{
            type: core.Input
        }], grouping: [{
            type: core.Input
        }], groupPanel: [{
            type: core.Input
        }], headerFilter: [{
            type: core.Input
        }], height: [{
            type: core.Input
        }], highlightChanges: [{
            type: core.Input
        }], hint: [{
            type: core.Input
        }], hoverStateEnabled: [{
            type: core.Input
        }], keyboardNavigation: [{
            type: core.Input
        }], keyExpr: [{
            type: core.Input
        }], loadPanel: [{
            type: core.Input
        }], masterDetail: [{
            type: core.Input
        }], noDataText: [{
            type: core.Input
        }], pager: [{
            type: core.Input
        }], paging: [{
            type: core.Input
        }], remoteOperations: [{
            type: core.Input
        }], renderAsync: [{
            type: core.Input
        }], repaintChangesOnly: [{
            type: core.Input
        }], rowAlternationEnabled: [{
            type: core.Input
        }], rowDragging: [{
            type: core.Input
        }], rowTemplate: [{
            type: core.Input
        }], rtlEnabled: [{
            type: core.Input
        }], scrolling: [{
            type: core.Input
        }], searchPanel: [{
            type: core.Input
        }], selectedRowKeys: [{
            type: core.Input
        }], selection: [{
            type: core.Input
        }], selectionFilter: [{
            type: core.Input
        }], showBorders: [{
            type: core.Input
        }], showColumnHeaders: [{
            type: core.Input
        }], showColumnLines: [{
            type: core.Input
        }], showRowLines: [{
            type: core.Input
        }], sortByGroupSummaryInfo: [{
            type: core.Input
        }], sorting: [{
            type: core.Input
        }], stateStoring: [{
            type: core.Input
        }], summary: [{
            type: core.Input
        }], tabIndex: [{
            type: core.Input
        }], twoWayBindingEnabled: [{
            type: core.Input
        }], visible: [{
            type: core.Input
        }], width: [{
            type: core.Input
        }], wordWrapEnabled: [{
            type: core.Input
        }], columnsChildren: [{
            type: core.ContentChildren,
            args: [nested.DxiColumnComponent]
        }], sortByGroupSummaryInfoChildren: [{
            type: core.ContentChildren,
            args: [nested.DxiSortByGroupSummaryInfoComponent]
        }], onAdaptiveDetailRowPreparing: [{
            type: core.Output
        }], onCellClick: [{
            type: core.Output
        }], onCellDblClick: [{
            type: core.Output
        }], onCellHoverChanged: [{
            type: core.Output
        }], onCellPrepared: [{
            type: core.Output
        }], onContentReady: [{
            type: core.Output
        }], onContextMenuPreparing: [{
            type: core.Output
        }], onDataErrorOccurred: [{
            type: core.Output
        }], onDisposing: [{
            type: core.Output
        }], onEditingStart: [{
            type: core.Output
        }], onEditorPrepared: [{
            type: core.Output
        }], onEditorPreparing: [{
            type: core.Output
        }], onExported: [{
            type: core.Output
        }], onExporting: [{
            type: core.Output
        }], onFileSaving: [{
            type: core.Output
        }], onFocusedCellChanged: [{
            type: core.Output
        }], onFocusedCellChanging: [{
            type: core.Output
        }], onFocusedRowChanged: [{
            type: core.Output
        }], onFocusedRowChanging: [{
            type: core.Output
        }], onInitialized: [{
            type: core.Output
        }], onInitNewRow: [{
            type: core.Output
        }], onKeyDown: [{
            type: core.Output
        }], onOptionChanged: [{
            type: core.Output
        }], onRowClick: [{
            type: core.Output
        }], onRowCollapsed: [{
            type: core.Output
        }], onRowCollapsing: [{
            type: core.Output
        }], onRowDblClick: [{
            type: core.Output
        }], onRowExpanded: [{
            type: core.Output
        }], onRowExpanding: [{
            type: core.Output
        }], onRowInserted: [{
            type: core.Output
        }], onRowInserting: [{
            type: core.Output
        }], onRowPrepared: [{
            type: core.Output
        }], onRowRemoved: [{
            type: core.Output
        }], onRowRemoving: [{
            type: core.Output
        }], onRowUpdated: [{
            type: core.Output
        }], onRowUpdating: [{
            type: core.Output
        }], onRowValidating: [{
            type: core.Output
        }], onSelectionChanged: [{
            type: core.Output
        }], onToolbarPreparing: [{
            type: core.Output
        }], accessKeyChange: [{
            type: core.Output
        }], activeStateEnabledChange: [{
            type: core.Output
        }], allowColumnReorderingChange: [{
            type: core.Output
        }], allowColumnResizingChange: [{
            type: core.Output
        }], autoNavigateToFocusedRowChange: [{
            type: core.Output
        }], cacheEnabledChange: [{
            type: core.Output
        }], cellHintEnabledChange: [{
            type: core.Output
        }], columnAutoWidthChange: [{
            type: core.Output
        }], columnChooserChange: [{
            type: core.Output
        }], columnFixingChange: [{
            type: core.Output
        }], columnHidingEnabledChange: [{
            type: core.Output
        }], columnMinWidthChange: [{
            type: core.Output
        }], columnResizingModeChange: [{
            type: core.Output
        }], columnsChange: [{
            type: core.Output
        }], columnWidthChange: [{
            type: core.Output
        }], customizeColumnsChange: [{
            type: core.Output
        }], customizeExportDataChange: [{
            type: core.Output
        }], dataSourceChange: [{
            type: core.Output
        }], dateSerializationFormatChange: [{
            type: core.Output
        }], disabledChange: [{
            type: core.Output
        }], editingChange: [{
            type: core.Output
        }], elementAttrChange: [{
            type: core.Output
        }], errorRowEnabledChange: [{
            type: core.Output
        }], exportChange: [{
            type: core.Output
        }], filterBuilderChange: [{
            type: core.Output
        }], filterBuilderPopupChange: [{
            type: core.Output
        }], filterPanelChange: [{
            type: core.Output
        }], filterRowChange: [{
            type: core.Output
        }], filterSyncEnabledChange: [{
            type: core.Output
        }], filterValueChange: [{
            type: core.Output
        }], focusedColumnIndexChange: [{
            type: core.Output
        }], focusedRowEnabledChange: [{
            type: core.Output
        }], focusedRowIndexChange: [{
            type: core.Output
        }], focusedRowKeyChange: [{
            type: core.Output
        }], focusStateEnabledChange: [{
            type: core.Output
        }], groupingChange: [{
            type: core.Output
        }], groupPanelChange: [{
            type: core.Output
        }], headerFilterChange: [{
            type: core.Output
        }], heightChange: [{
            type: core.Output
        }], highlightChangesChange: [{
            type: core.Output
        }], hintChange: [{
            type: core.Output
        }], hoverStateEnabledChange: [{
            type: core.Output
        }], keyboardNavigationChange: [{
            type: core.Output
        }], keyExprChange: [{
            type: core.Output
        }], loadPanelChange: [{
            type: core.Output
        }], masterDetailChange: [{
            type: core.Output
        }], noDataTextChange: [{
            type: core.Output
        }], pagerChange: [{
            type: core.Output
        }], pagingChange: [{
            type: core.Output
        }], remoteOperationsChange: [{
            type: core.Output
        }], renderAsyncChange: [{
            type: core.Output
        }], repaintChangesOnlyChange: [{
            type: core.Output
        }], rowAlternationEnabledChange: [{
            type: core.Output
        }], rowDraggingChange: [{
            type: core.Output
        }], rowTemplateChange: [{
            type: core.Output
        }], rtlEnabledChange: [{
            type: core.Output
        }], scrollingChange: [{
            type: core.Output
        }], searchPanelChange: [{
            type: core.Output
        }], selectedRowKeysChange: [{
            type: core.Output
        }], selectionChange: [{
            type: core.Output
        }], selectionFilterChange: [{
            type: core.Output
        }], showBordersChange: [{
            type: core.Output
        }], showColumnHeadersChange: [{
            type: core.Output
        }], showColumnLinesChange: [{
            type: core.Output
        }], showRowLinesChange: [{
            type: core.Output
        }], sortByGroupSummaryInfoChange: [{
            type: core.Output
        }], sortingChange: [{
            type: core.Output
        }], stateStoringChange: [{
            type: core.Output
        }], summaryChange: [{
            type: core.Output
        }], tabIndexChange: [{
            type: core.Output
        }], twoWayBindingEnabledChange: [{
            type: core.Output
        }], visibleChange: [{
            type: core.Output
        }], widthChange: [{
            type: core.Output
        }], wordWrapEnabledChange: [{
            type: core.Output
        }] }); })();
        return DxDataGridComponent;
    }(core$1.DxComponent));
    var DxDataGridModule = /** @class */ (function () {
        function DxDataGridModule() {
        }
DxDataGridModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: DxDataGridModule });
DxDataGridModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function DxDataGridModule_Factory(t) { return new (t || DxDataGridModule)(); }, imports: [[
            nested.DxoColumnChooserModule,
            nested.DxoColumnFixingModule,
            nested.DxoTextsModule,
            nested.DxiColumnModule,
            nested.DxiButtonModule,
            nested.DxoHeaderFilterModule,
            nested.DxoLookupModule,
            nested.DxoFormatModule,
            nested.DxoFormItemModule,
            nested.DxoLabelModule,
            nested.DxiValidationRuleModule,
            nested.DxoEditingModule,
            nested.DxoFormModule,
            nested.DxoColCountByScreenModule,
            nested.DxiItemModule,
            nested.DxoTabPanelOptionsModule,
            nested.DxiTabModule,
            nested.DxoButtonOptionsModule,
            nested.DxoPopupModule,
            nested.DxoAnimationModule,
            nested.DxoHideModule,
            nested.DxoShowModule,
            nested.DxoPositionModule,
            nested.DxoAtModule,
            nested.DxoBoundaryOffsetModule,
            nested.DxoCollisionModule,
            nested.DxoMyModule,
            nested.DxoOffsetModule,
            nested.DxiToolbarItemModule,
            nested.DxoExportModule,
            nested.DxoFilterBuilderModule,
            nested.DxiCustomOperationModule,
            nested.DxiFieldModule,
            nested.DxoFilterOperationDescriptionsModule,
            nested.DxoGroupOperationDescriptionsModule,
            nested.DxoFilterBuilderPopupModule,
            nested.DxoFilterPanelModule,
            nested.DxoFilterRowModule,
            nested.DxoOperationDescriptionsModule,
            nested.DxoGroupingModule,
            nested.DxoGroupPanelModule,
            nested.DxoKeyboardNavigationModule,
            nested.DxoLoadPanelModule,
            nested.DxoMasterDetailModule,
            nested.DxoPagerModule,
            nested.DxoPagingModule,
            nested.DxoRemoteOperationsModule,
            nested.DxoRowDraggingModule,
            nested.DxoCursorOffsetModule,
            nested.DxoScrollingModule,
            nested.DxoSearchPanelModule,
            nested.DxoSelectionModule,
            nested.DxiSortByGroupSummaryInfoModule,
            nested.DxoSortingModule,
            nested.DxoStateStoringModule,
            nested.DxoSummaryModule,
            nested.DxiGroupItemModule,
            nested.DxoValueFormatModule,
            nested.DxiTotalItemModule,
            core$1.DxIntegrationModule,
            core$1.DxTemplateModule,
            platformBrowser.BrowserTransferStateModule
        ], ɵngcc3.DxoColumnChooserModule, ɵngcc3.DxoColumnFixingModule, ɵngcc3.DxoTextsModule, ɵngcc3.DxiColumnModule, ɵngcc3.DxiButtonModule, ɵngcc3.DxoHeaderFilterModule, ɵngcc3.DxoLookupModule, ɵngcc3.DxoFormatModule, ɵngcc3.DxoFormItemModule, ɵngcc3.DxoLabelModule, ɵngcc3.DxiValidationRuleModule, ɵngcc3.DxoEditingModule, ɵngcc3.DxoFormModule, ɵngcc3.DxoColCountByScreenModule, ɵngcc3.DxiItemModule, ɵngcc3.DxoTabPanelOptionsModule, ɵngcc3.DxiTabModule, ɵngcc3.DxoButtonOptionsModule, ɵngcc3.DxoPopupModule, ɵngcc3.DxoAnimationModule, ɵngcc3.DxoHideModule, ɵngcc3.DxoShowModule, ɵngcc3.DxoPositionModule, ɵngcc3.DxoAtModule, ɵngcc3.DxoBoundaryOffsetModule, ɵngcc3.DxoCollisionModule, ɵngcc3.DxoMyModule, ɵngcc3.DxoOffsetModule, ɵngcc3.DxiToolbarItemModule, ɵngcc3.DxoExportModule, ɵngcc3.DxoFilterBuilderModule, ɵngcc3.DxiCustomOperationModule, ɵngcc3.DxiFieldModule, ɵngcc3.DxoFilterOperationDescriptionsModule, ɵngcc3.DxoGroupOperationDescriptionsModule, ɵngcc3.DxoFilterBuilderPopupModule, ɵngcc3.DxoFilterPanelModule, ɵngcc3.DxoFilterRowModule, ɵngcc3.DxoOperationDescriptionsModule, ɵngcc3.DxoGroupingModule, ɵngcc3.DxoGroupPanelModule, ɵngcc3.DxoKeyboardNavigationModule, ɵngcc3.DxoLoadPanelModule, ɵngcc3.DxoMasterDetailModule, ɵngcc3.DxoPagerModule, ɵngcc3.DxoPagingModule, ɵngcc3.DxoRemoteOperationsModule, ɵngcc3.DxoRowDraggingModule, ɵngcc3.DxoCursorOffsetModule, ɵngcc3.DxoScrollingModule, ɵngcc3.DxoSearchPanelModule, ɵngcc3.DxoSelectionModule, ɵngcc3.DxiSortByGroupSummaryInfoModule, ɵngcc3.DxoSortingModule, ɵngcc3.DxoStateStoringModule, ɵngcc3.DxoSummaryModule, ɵngcc3.DxiGroupItemModule, ɵngcc3.DxoValueFormatModule, ɵngcc3.DxiTotalItemModule, ɵngcc1.DxTemplateModule] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(DxDataGridModule, { declarations: [DxDataGridComponent], imports: [ɵngcc3.DxoColumnChooserModule, ɵngcc3.DxoColumnFixingModule, ɵngcc3.DxoTextsModule, ɵngcc3.DxiColumnModule, ɵngcc3.DxiButtonModule, ɵngcc3.DxoHeaderFilterModule, ɵngcc3.DxoLookupModule, ɵngcc3.DxoFormatModule, ɵngcc3.DxoFormItemModule, ɵngcc3.DxoLabelModule, ɵngcc3.DxiValidationRuleModule, ɵngcc3.DxoEditingModule, ɵngcc3.DxoFormModule, ɵngcc3.DxoColCountByScreenModule, ɵngcc3.DxiItemModule, ɵngcc3.DxoTabPanelOptionsModule, ɵngcc3.DxiTabModule, ɵngcc3.DxoButtonOptionsModule, ɵngcc3.DxoPopupModule, ɵngcc3.DxoAnimationModule, ɵngcc3.DxoHideModule, ɵngcc3.DxoShowModule, ɵngcc3.DxoPositionModule, ɵngcc3.DxoAtModule, ɵngcc3.DxoBoundaryOffsetModule, ɵngcc3.DxoCollisionModule, ɵngcc3.DxoMyModule, ɵngcc3.DxoOffsetModule, ɵngcc3.DxiToolbarItemModule, ɵngcc3.DxoExportModule, ɵngcc3.DxoFilterBuilderModule, ɵngcc3.DxiCustomOperationModule, ɵngcc3.DxiFieldModule, ɵngcc3.DxoFilterOperationDescriptionsModule, ɵngcc3.DxoGroupOperationDescriptionsModule, ɵngcc3.DxoFilterBuilderPopupModule, ɵngcc3.DxoFilterPanelModule, ɵngcc3.DxoFilterRowModule, ɵngcc3.DxoOperationDescriptionsModule, ɵngcc3.DxoGroupingModule, ɵngcc3.DxoGroupPanelModule, ɵngcc3.DxoKeyboardNavigationModule, ɵngcc3.DxoLoadPanelModule, ɵngcc3.DxoMasterDetailModule, ɵngcc3.DxoPagerModule, ɵngcc3.DxoPagingModule, ɵngcc3.DxoRemoteOperationsModule, ɵngcc3.DxoRowDraggingModule, ɵngcc3.DxoCursorOffsetModule, ɵngcc3.DxoScrollingModule, ɵngcc3.DxoSearchPanelModule, ɵngcc3.DxoSelectionModule, ɵngcc3.DxiSortByGroupSummaryInfoModule, ɵngcc3.DxoSortingModule, ɵngcc3.DxoStateStoringModule, ɵngcc3.DxoSummaryModule, ɵngcc3.DxiGroupItemModule, ɵngcc3.DxoValueFormatModule, ɵngcc3.DxiTotalItemModule, ɵngcc1.DxIntegrationModule, ɵngcc1.DxTemplateModule, ɵngcc2.BrowserTransferStateModule], exports: [DxDataGridComponent, ɵngcc3.DxoColumnChooserModule, ɵngcc3.DxoColumnFixingModule, ɵngcc3.DxoTextsModule, ɵngcc3.DxiColumnModule, ɵngcc3.DxiButtonModule, ɵngcc3.DxoHeaderFilterModule, ɵngcc3.DxoLookupModule, ɵngcc3.DxoFormatModule, ɵngcc3.DxoFormItemModule, ɵngcc3.DxoLabelModule, ɵngcc3.DxiValidationRuleModule, ɵngcc3.DxoEditingModule, ɵngcc3.DxoFormModule, ɵngcc3.DxoColCountByScreenModule, ɵngcc3.DxiItemModule, ɵngcc3.DxoTabPanelOptionsModule, ɵngcc3.DxiTabModule, ɵngcc3.DxoButtonOptionsModule, ɵngcc3.DxoPopupModule, ɵngcc3.DxoAnimationModule, ɵngcc3.DxoHideModule, ɵngcc3.DxoShowModule, ɵngcc3.DxoPositionModule, ɵngcc3.DxoAtModule, ɵngcc3.DxoBoundaryOffsetModule, ɵngcc3.DxoCollisionModule, ɵngcc3.DxoMyModule, ɵngcc3.DxoOffsetModule, ɵngcc3.DxiToolbarItemModule, ɵngcc3.DxoExportModule, ɵngcc3.DxoFilterBuilderModule, ɵngcc3.DxiCustomOperationModule, ɵngcc3.DxiFieldModule, ɵngcc3.DxoFilterOperationDescriptionsModule, ɵngcc3.DxoGroupOperationDescriptionsModule, ɵngcc3.DxoFilterBuilderPopupModule, ɵngcc3.DxoFilterPanelModule, ɵngcc3.DxoFilterRowModule, ɵngcc3.DxoOperationDescriptionsModule, ɵngcc3.DxoGroupingModule, ɵngcc3.DxoGroupPanelModule, ɵngcc3.DxoKeyboardNavigationModule, ɵngcc3.DxoLoadPanelModule, ɵngcc3.DxoMasterDetailModule, ɵngcc3.DxoPagerModule, ɵngcc3.DxoPagingModule, ɵngcc3.DxoRemoteOperationsModule, ɵngcc3.DxoRowDraggingModule, ɵngcc3.DxoCursorOffsetModule, ɵngcc3.DxoScrollingModule, ɵngcc3.DxoSearchPanelModule, ɵngcc3.DxoSelectionModule, ɵngcc3.DxiSortByGroupSummaryInfoModule, ɵngcc3.DxoSortingModule, ɵngcc3.DxoStateStoringModule, ɵngcc3.DxoSummaryModule, ɵngcc3.DxiGroupItemModule, ɵngcc3.DxoValueFormatModule, ɵngcc3.DxiTotalItemModule, ɵngcc1.DxTemplateModule] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DxDataGridModule, [{
        type: core.NgModule,
        args: [{
                imports: [
                    nested.DxoColumnChooserModule,
                    nested.DxoColumnFixingModule,
                    nested.DxoTextsModule,
                    nested.DxiColumnModule,
                    nested.DxiButtonModule,
                    nested.DxoHeaderFilterModule,
                    nested.DxoLookupModule,
                    nested.DxoFormatModule,
                    nested.DxoFormItemModule,
                    nested.DxoLabelModule,
                    nested.DxiValidationRuleModule,
                    nested.DxoEditingModule,
                    nested.DxoFormModule,
                    nested.DxoColCountByScreenModule,
                    nested.DxiItemModule,
                    nested.DxoTabPanelOptionsModule,
                    nested.DxiTabModule,
                    nested.DxoButtonOptionsModule,
                    nested.DxoPopupModule,
                    nested.DxoAnimationModule,
                    nested.DxoHideModule,
                    nested.DxoShowModule,
                    nested.DxoPositionModule,
                    nested.DxoAtModule,
                    nested.DxoBoundaryOffsetModule,
                    nested.DxoCollisionModule,
                    nested.DxoMyModule,
                    nested.DxoOffsetModule,
                    nested.DxiToolbarItemModule,
                    nested.DxoExportModule,
                    nested.DxoFilterBuilderModule,
                    nested.DxiCustomOperationModule,
                    nested.DxiFieldModule,
                    nested.DxoFilterOperationDescriptionsModule,
                    nested.DxoGroupOperationDescriptionsModule,
                    nested.DxoFilterBuilderPopupModule,
                    nested.DxoFilterPanelModule,
                    nested.DxoFilterRowModule,
                    nested.DxoOperationDescriptionsModule,
                    nested.DxoGroupingModule,
                    nested.DxoGroupPanelModule,
                    nested.DxoKeyboardNavigationModule,
                    nested.DxoLoadPanelModule,
                    nested.DxoMasterDetailModule,
                    nested.DxoPagerModule,
                    nested.DxoPagingModule,
                    nested.DxoRemoteOperationsModule,
                    nested.DxoRowDraggingModule,
                    nested.DxoCursorOffsetModule,
                    nested.DxoScrollingModule,
                    nested.DxoSearchPanelModule,
                    nested.DxoSelectionModule,
                    nested.DxiSortByGroupSummaryInfoModule,
                    nested.DxoSortingModule,
                    nested.DxoStateStoringModule,
                    nested.DxoSummaryModule,
                    nested.DxiGroupItemModule,
                    nested.DxoValueFormatModule,
                    nested.DxiTotalItemModule,
                    core$1.DxIntegrationModule,
                    core$1.DxTemplateModule,
                    platformBrowser.BrowserTransferStateModule
                ],
                declarations: [
                    DxDataGridComponent
                ],
                exports: [
                    DxDataGridComponent,
                    nested.DxoColumnChooserModule,
                    nested.DxoColumnFixingModule,
                    nested.DxoTextsModule,
                    nested.DxiColumnModule,
                    nested.DxiButtonModule,
                    nested.DxoHeaderFilterModule,
                    nested.DxoLookupModule,
                    nested.DxoFormatModule,
                    nested.DxoFormItemModule,
                    nested.DxoLabelModule,
                    nested.DxiValidationRuleModule,
                    nested.DxoEditingModule,
                    nested.DxoFormModule,
                    nested.DxoColCountByScreenModule,
                    nested.DxiItemModule,
                    nested.DxoTabPanelOptionsModule,
                    nested.DxiTabModule,
                    nested.DxoButtonOptionsModule,
                    nested.DxoPopupModule,
                    nested.DxoAnimationModule,
                    nested.DxoHideModule,
                    nested.DxoShowModule,
                    nested.DxoPositionModule,
                    nested.DxoAtModule,
                    nested.DxoBoundaryOffsetModule,
                    nested.DxoCollisionModule,
                    nested.DxoMyModule,
                    nested.DxoOffsetModule,
                    nested.DxiToolbarItemModule,
                    nested.DxoExportModule,
                    nested.DxoFilterBuilderModule,
                    nested.DxiCustomOperationModule,
                    nested.DxiFieldModule,
                    nested.DxoFilterOperationDescriptionsModule,
                    nested.DxoGroupOperationDescriptionsModule,
                    nested.DxoFilterBuilderPopupModule,
                    nested.DxoFilterPanelModule,
                    nested.DxoFilterRowModule,
                    nested.DxoOperationDescriptionsModule,
                    nested.DxoGroupingModule,
                    nested.DxoGroupPanelModule,
                    nested.DxoKeyboardNavigationModule,
                    nested.DxoLoadPanelModule,
                    nested.DxoMasterDetailModule,
                    nested.DxoPagerModule,
                    nested.DxoPagingModule,
                    nested.DxoRemoteOperationsModule,
                    nested.DxoRowDraggingModule,
                    nested.DxoCursorOffsetModule,
                    nested.DxoScrollingModule,
                    nested.DxoSearchPanelModule,
                    nested.DxoSelectionModule,
                    nested.DxiSortByGroupSummaryInfoModule,
                    nested.DxoSortingModule,
                    nested.DxoStateStoringModule,
                    nested.DxoSummaryModule,
                    nested.DxiGroupItemModule,
                    nested.DxoValueFormatModule,
                    nested.DxiTotalItemModule,
                    core$1.DxTemplateModule
                ]
            }]
    }], function () { return []; }, null); })();
        return DxDataGridModule;
    }());

    exports.DxDataGridComponent = DxDataGridComponent;
    exports.DxDataGridModule = DxDataGridModule;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
