// flx-ui-datatable/esm5/lib/pipes/hasLimitText.pipe.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Pipe } from '@angular/core';
import { FlxUiDatatableService } from '../flx-ui-datatable.service';
import * as ɵngcc0 from '@angular/core';
var HasLimitText = /** @class */ (function () {
    function HasLimitText(service) {
        this.service = service;
    }
    /**
     * @param {?} array
     * @param {?} indexReached
     * @return {?}
     */
    HasLimitText.prototype.transform = /**
     * @param {?} array
     * @param {?} indexReached
     * @return {?}
     */
    function (array, indexReached) {
        var /** @type {?} */ found;
        try {
            for (var array_1 = tslib_1.__values(array), array_1_1 = array_1.next(); !array_1_1.done; array_1_1 = array_1.next()) {
                var i = array_1_1.value;
                if (i == indexReached) {
                    found = true;
                    break;
                }
            }
        }
        catch (e_1_1) { e_1 = { error: e_1_1 }; }
        finally {
            try {
                if (array_1_1 && !array_1_1.done && (_a = array_1.return)) _a.call(array_1);
            }
            finally { if (e_1) throw e_1.error; }
        }
        return found;
        var e_1, _a;
    };
    /** @nocollapse */
    HasLimitText.ctorParameters = function () { return [
        { type: FlxUiDatatableService }
    ]; };
HasLimitText.ɵfac = function HasLimitText_Factory(t) { return new (t || HasLimitText)(ɵngcc0.ɵɵdirectiveInject(FlxUiDatatableService)); };
HasLimitText.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "haslimittext", type: HasLimitText, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(HasLimitText, [{
        type: Pipe,
        args: [{
                name: 'haslimittext'
            }]
    }], function () { return [{ type: FlxUiDatatableService }]; }, null); })();
    return HasLimitText;
}());
export { HasLimitText };
function HasLimitText_tsickle_Closure_declarations() {
    /** @type {?} */
    HasLimitText.prototype.service;
}
