// flx-ui-datatable/esm5/lib/pipes/limitText.pipe.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Pipe } from '@angular/core';
import { FlxUiDatatableService } from '../flx-ui-datatable.service';
import * as ɵngcc0 from '@angular/core';
var LimitText = /** @class */ (function () {
    function LimitText(service) {
        this.service = service;
    }
    /**
     * @param {?} mainpulationConfig
     * @param {?} text
     * @return {?}
     */
    LimitText.prototype.transform = /**
     * @param {?} mainpulationConfig
     * @param {?} text
     * @return {?}
     */
    function (mainpulationConfig, text) {
        var /** @type {?} */ from = (mainpulationConfig.hasOwnProperty("from")) ? mainpulationConfig['from'] : 0;
        var /** @type {?} */ to = (mainpulationConfig.hasOwnProperty("to")) ? mainpulationConfig["to"] : text.length;
        var /** @type {?} */ appendDots = (mainpulationConfig.hasOwnProperty("appendDots")) ? mainpulationConfig["appendDots"] : "";
        var /** @type {?} */ txt = "";
        if (text.length > to) {
            txt = text.substring(from, to) + appendDots;
        }
        else {
            txt = text;
        }
        return txt;
    };
    /** @nocollapse */
    LimitText.ctorParameters = function () { return [
        { type: FlxUiDatatableService }
    ]; };
LimitText.ɵfac = function LimitText_Factory(t) { return new (t || LimitText)(ɵngcc0.ɵɵdirectiveInject(FlxUiDatatableService)); };
LimitText.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "limittext", type: LimitText, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(LimitText, [{
        type: Pipe,
        args: [{
                name: 'limittext'
            }]
    }], function () { return [{ type: FlxUiDatatableService }]; }, null); })();
    return LimitText;
}());
export { LimitText };
function LimitText_tsickle_Closure_declarations() {
    /** @type {?} */
    LimitText.prototype.service;
}
