// flx-ui-datatable/esm5/lib/pipes/processeddata.pipe.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
import { Pipe } from '@angular/core';
import { FlxUiDatatableService } from '../flx-ui-datatable.service';
import * as ɵngcc0 from '@angular/core';
var ProcessedData = /** @class */ (function () {
    function ProcessedData(service) {
        this.service = service;
    }
    /**
     * @param {?} dataIndex
     * @return {?}
     */
    ProcessedData.prototype.transform = /**
     * @param {?} dataIndex
     * @return {?}
     */
    function (dataIndex) {
        var /** @type {?} */ isProcessed = false;
        var /** @type {?} */ foundIndex = -1;
        for (var /** @type {?} */ i = 0; i < this.service.processedData.length; i++) {
            if (i == dataIndex) {
                isProcessed = true;
                foundIndex = i;
            }
        }
        if (foundIndex < 0) {
            this.service.processedData.push(dataIndex);
        }
        return isProcessed;
    };
    /** @nocollapse */
    ProcessedData.ctorParameters = function () { return [
        { type: FlxUiDatatableService }
    ]; };
ProcessedData.ɵfac = function ProcessedData_Factory(t) { return new (t || ProcessedData)(ɵngcc0.ɵɵdirectiveInject(FlxUiDatatableService)); };
ProcessedData.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "isprocessed", type: ProcessedData, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ProcessedData, [{
        type: Pipe,
        args: [{
                name: 'isprocessed'
            }]
    }], function () { return [{ type: FlxUiDatatableService }]; }, null); })();
    return ProcessedData;
}());
export { ProcessedData };
function ProcessedData_tsickle_Closure_declarations() {
    /** @type {?} */
    ProcessedData.prototype.service;
}
