// pdfmake-wrapper/lib/definitions/qr.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
import { StyleDefinition } from './style-definition';
var QR = (function (_super) {
    __extends(QR, _super);
    function QR(_code) {
        var _this = _super.call(this) || this;
        _this._code = _code;
        _this.content.qr = _this._code;
        return _this;
    }
    QR.prototype.fit = function (fit) {
        this.content.fit = fit;
        return this;
    };
    QR.prototype.version = function (version) {
        this.content.version = version;
        return this;
    };
    QR.prototype.foreground = function (foreground) {
        this.content.foreground = foreground;
        return this;
    };
    QR.prototype.eccLevel = function (eccLevel) {
        this.content.eccLevel = eccLevel;
        return this;
    };
    QR.prototype.mode = function (mode) {
        this.content.mode = mode;
        return this;
    };
    QR.prototype.mask = function (mask) {
        this.content.mask = mask;
        return this;
    };
    return QR;
}(StyleDefinition));
export { QR };
